{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/bryanjjones/InSiTE/blob/master/Site_Integration_Analysis.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "id": "mVgZpJUPP-cb",
        "cellView": "form",
        "outputId": "d823b6a6-e03a-4fce-9260-88270d87280b",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Number of cores: 8\n",
            "Cloning into 'InSiTE'...\n",
            "remote: Enumerating objects: 1151, done.\u001b[K\n",
            "remote: Counting objects: 100% (182/182), done.\u001b[K\n",
            "remote: Compressing objects: 100% (171/171), done.\u001b[K\n",
            "remote: Total 1151 (delta 109), reused 23 (delta 4), pack-reused 969\u001b[K\n",
            "Receiving objects: 100% (1151/1151), 6.70 MiB | 11.23 MiB/s, done.\n",
            "Resolving deltas: 100% (775/775), done.\n",
            "Collecting weblogo (from -r InSiTE/requirements.txt (line 1))\n",
            "  Downloading weblogo-3.7.12-py3-none-any.whl (571 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m571.7/571.7 kB\u001b[0m \u001b[31m7.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting biopython (from -r InSiTE/requirements.txt (line 2))\n",
            "  Downloading biopython-1.81-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.1 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.1/3.1 MB\u001b[0m \u001b[31m58.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting colorama (from -r InSiTE/requirements.txt (line 3))\n",
            "  Downloading colorama-0.4.6-py2.py3-none-any.whl (25 kB)\n",
            "Collecting pysam (from -r InSiTE/requirements.txt (line 4))\n",
            "  Downloading pysam-0.22.0-cp310-cp310-manylinux_2_28_x86_64.whl (21.9 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m21.9/21.9 MB\u001b[0m \u001b[31m63.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting pybedtools (from -r InSiTE/requirements.txt (line 5))\n",
            "  Downloading pybedtools-0.9.1.tar.gz (12.5 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m12.5/12.5 MB\u001b[0m \u001b[31m81.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting twobitreader (from -r InSiTE/requirements.txt (line 6))\n",
            "  Downloading twobitreader-3.1.7.tar.gz (9.2 kB)\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting cutadapt (from -r InSiTE/requirements.txt (line 7))\n",
            "  Downloading cutadapt-4.5-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (596 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m597.0/597.0 kB\u001b[0m \u001b[31m41.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: numpy in /usr/local/lib/python3.10/dist-packages (from weblogo->-r InSiTE/requirements.txt (line 1)) (1.23.5)\n",
            "Requirement already satisfied: scipy in /usr/local/lib/python3.10/dist-packages (from weblogo->-r InSiTE/requirements.txt (line 1)) (1.11.3)\n",
            "Requirement already satisfied: setuptools in /usr/local/lib/python3.10/dist-packages (from weblogo->-r InSiTE/requirements.txt (line 1)) (67.7.2)\n",
            "Requirement already satisfied: six in /usr/local/lib/python3.10/dist-packages (from pybedtools->-r InSiTE/requirements.txt (line 5)) (1.16.0)\n",
            "Collecting dnaio>=0.10.0 (from cutadapt->-r InSiTE/requirements.txt (line 7))\n",
            "  Downloading dnaio-1.0.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (95 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m95.2/95.2 kB\u001b[0m \u001b[31m11.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting xopen>=1.6.0 (from cutadapt->-r InSiTE/requirements.txt (line 7))\n",
            "  Downloading xopen-1.8.0-py3-none-any.whl (16 kB)\n",
            "Collecting isal>=1.4.1 (from xopen>=1.6.0->cutadapt->-r InSiTE/requirements.txt (line 7))\n",
            "  Downloading isal-1.5.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (259 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m259.8/259.8 kB\u001b[0m \u001b[31m24.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hBuilding wheels for collected packages: pybedtools, twobitreader\n",
            "  Building wheel for pybedtools (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for pybedtools: filename=pybedtools-0.9.1-cp310-cp310-linux_x86_64.whl size=13997642 sha256=9a8d179b1ebfb92b73a6e833005163e5ab8018011e7eed025753c2a77e8b317d\n",
            "  Stored in directory: /root/.cache/pip/wheels/f5/79/76/c2cef660013cdf573214696aec4bc7ec352bf46141682230af\n",
            "  Building wheel for twobitreader (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for twobitreader: filename=twobitreader-3.1.7-py3-none-any.whl size=9581 sha256=5ea5c676d9df683503bb2c284d3f05e2214444bac99fd7d2a0be76a03305523c\n",
            "  Stored in directory: /root/.cache/pip/wheels/d1/4a/92/1d73e58f8e3e5eef4e59c8719a7c5c542f32804ec50dd33a58\n",
            "Successfully built pybedtools twobitreader\n",
            "Installing collected packages: twobitreader, pysam, isal, colorama, biopython, xopen, weblogo, pybedtools, dnaio, cutadapt\n",
            "Successfully installed biopython-1.81 colorama-0.4.6 cutadapt-4.5 dnaio-1.0.1 isal-1.5.2 pybedtools-0.9.1 pysam-0.22.0 twobitreader-3.1.7 weblogo-3.7.12 xopen-1.8.0\n",
            "/content/InSiTE\n",
            "--2023-11-07 18:28:07--  https://sourceforge.net/projects/bowtie-bio/files/bowtie2/2.5.1/bowtie2-2.5.1-linux-x86_64.zip\n",
            "Resolving sourceforge.net (sourceforge.net)... 104.18.37.111, 172.64.150.145, 2606:4700:4400::6812:256f, ...\n",
            "Connecting to sourceforge.net (sourceforge.net)|104.18.37.111|:443... connected.\n",
            "HTTP request sent, awaiting response... 301 Moved Permanently\n",
            "Location: https://sourceforge.net/projects/bowtie-bio/files/bowtie2/2.5.1/bowtie2-2.5.1-linux-x86_64.zip/ [following]\n",
            "--2023-11-07 18:28:07--  https://sourceforge.net/projects/bowtie-bio/files/bowtie2/2.5.1/bowtie2-2.5.1-linux-x86_64.zip/\n",
            "Reusing existing connection to sourceforge.net:443.\n",
            "HTTP request sent, awaiting response... 301 Moved Permanently\n",
            "Location: https://sourceforge.net/projects/bowtie-bio/files/bowtie2/2.5.1/bowtie2-2.5.1-linux-x86_64.zip/download [following]\n",
            "--2023-11-07 18:28:08--  https://sourceforge.net/projects/bowtie-bio/files/bowtie2/2.5.1/bowtie2-2.5.1-linux-x86_64.zip/download\n",
            "Reusing existing connection to sourceforge.net:443.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://downloads.sourceforge.net/project/bowtie-bio/bowtie2/2.5.1/bowtie2-2.5.1-linux-x86_64.zip?ts=gAAAAABlSoG4Ghhaxyf2nnd2cBjR6_463l_TpdcndQhKf1jyo3bu4OOrDznFF8nQ9m8k8TWJmGqQ2R9Ge3ldUKtuZp93GJvclQ%3D%3D&use_mirror=newcontinuum&r= [following]\n",
            "--2023-11-07 18:28:08--  https://downloads.sourceforge.net/project/bowtie-bio/bowtie2/2.5.1/bowtie2-2.5.1-linux-x86_64.zip?ts=gAAAAABlSoG4Ghhaxyf2nnd2cBjR6_463l_TpdcndQhKf1jyo3bu4OOrDznFF8nQ9m8k8TWJmGqQ2R9Ge3ldUKtuZp93GJvclQ%3D%3D&use_mirror=newcontinuum&r=\n",
            "Resolving downloads.sourceforge.net (downloads.sourceforge.net)... 204.68.111.105\n",
            "Connecting to downloads.sourceforge.net (downloads.sourceforge.net)|204.68.111.105|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://newcontinuum.dl.sourceforge.net/project/bowtie-bio/bowtie2/2.5.1/bowtie2-2.5.1-linux-x86_64.zip [following]\n",
            "--2023-11-07 18:28:08--  https://newcontinuum.dl.sourceforge.net/project/bowtie-bio/bowtie2/2.5.1/bowtie2-2.5.1-linux-x86_64.zip\n",
            "Resolving newcontinuum.dl.sourceforge.net (newcontinuum.dl.sourceforge.net)... 64.79.96.4, 2607:ff50:0:11::32\n",
            "Connecting to newcontinuum.dl.sourceforge.net (newcontinuum.dl.sourceforge.net)|64.79.96.4|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 80730015 (77M) [application/octet-stream]\n",
            "Saving to: ‘bins/bowtie2-2.5.1-linux-x86_64.zip’\n",
            "\n",
            "bowtie2-2.5.1-linux 100%[===================>]  76.99M   124KB/s    in 5m 25s  \n",
            "\n",
            "2023-11-07 18:33:35 (242 KB/s) - ‘bins/bowtie2-2.5.1-linux-x86_64.zip’ saved [80730015/80730015]\n",
            "\n",
            "Archive:  bins/bowtie2-2.5.1-linux-x86_64.zip\n",
            "   creating: bins/bowtie2-2.5.1-linux-x86_64/\n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/bowtie2-inspect-s-debug  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/LICENSE  \n",
            "   creating: bins/bowtie2-2.5.1-linux-x86_64/doc/\n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/doc/README  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/doc/manual.html  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/doc/style.css  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/bowtie2-align-s  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/MANUAL  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/bowtie2-build-s  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/NEWS  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/bowtie2-build-l-debug  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/AUTHORS  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/bowtie2-inspect-l  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/bowtie2-align-s-debug  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/bowtie2-align-l-debug  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/bowtie2-align-l  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/bowtie2-build-s-debug  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/bowtie2-build-l  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/README.md  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/bowtie2-inspect-s  \n",
            " extracting: bins/bowtie2-2.5.1-linux-x86_64/BOWTIE2_VERSION  \n",
            "   creating: bins/bowtie2-2.5.1-linux-x86_64/scripts/\n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/scripts/bowtie2-hbb.sh  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/scripts/make_hg18.sh  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/scripts/make_rn4.sh  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/scripts/make_e_coli.sh  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/scripts/make_canFam2.sh  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/scripts/make_c_elegans.sh  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/scripts/gen_2b_occ_lookup.pl  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/scripts/convert_quals.pl  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/scripts/make_m_musculus_ncbi37.sh  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/scripts/make_mm10.sh  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/scripts/make_mm9.sh  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/scripts/make_h_sapiens_ncbi37.sh  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/scripts/make_h_sapiens_ncbi36.sh  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/scripts/gen_solqual_lookup.pl  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/scripts/make_b_taurus_UMD3.sh  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/scripts/make_a_thaliana_tair.sh  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/scripts/gen_occ_lookup.pl  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/scripts/make_s_cerevisiae.sh  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/scripts/make_d_melanogaster.sh  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/scripts/make_hg19.sh  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/scripts/infer_fraglen.pl  \n",
            "   creating: bins/bowtie2-2.5.1-linux-x86_64/example/\n",
            "   creating: bins/bowtie2-2.5.1-linux-x86_64/example/reads/\n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/example/reads/combined_reads.bam  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/example/reads/simulate.pl  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/example/reads/reads_2.fq  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/example/reads/longreads.fq  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/example/reads/reads_1.fq  \n",
            "   creating: bins/bowtie2-2.5.1-linux-x86_64/example/index/\n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/example/index/lambda_virus.rev.2.bt2  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/example/index/lambda_virus.3.bt2  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/example/index/lambda_virus.4.bt2  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/example/index/lambda_virus.2.bt2  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/example/index/lambda_virus.rev.1.bt2  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/example/index/lambda_virus.1.bt2  \n",
            "   creating: bins/bowtie2-2.5.1-linux-x86_64/example/reference/\n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/example/reference/lambda_virus.fa  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/bowtie2-build  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/bowtie2-inspect  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/bowtie2  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/bowtie2-inspect-l-debug  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/MANUAL.markdown  \n",
            "  inflating: bins/bowtie2-2.5.1-linux-x86_64/TUTORIAL  \n",
            "--2023-11-07 18:33:38--  https://sourceforge.net/projects/bbmap/files/BBMap_39.03.tar.gz\n",
            "Resolving sourceforge.net (sourceforge.net)... 172.64.150.145, 104.18.37.111, 2606:4700:4400::6812:256f, ...\n",
            "Connecting to sourceforge.net (sourceforge.net)|172.64.150.145|:443... connected.\n",
            "HTTP request sent, awaiting response... 301 Moved Permanently\n",
            "Location: https://sourceforge.net/projects/bbmap/files/BBMap_39.03.tar.gz/ [following]\n",
            "--2023-11-07 18:33:38--  https://sourceforge.net/projects/bbmap/files/BBMap_39.03.tar.gz/\n",
            "Reusing existing connection to sourceforge.net:443.\n",
            "HTTP request sent, awaiting response... 301 Moved Permanently\n",
            "Location: https://sourceforge.net/projects/bbmap/files/BBMap_39.03.tar.gz/download [following]\n",
            "--2023-11-07 18:33:39--  https://sourceforge.net/projects/bbmap/files/BBMap_39.03.tar.gz/download\n",
            "Reusing existing connection to sourceforge.net:443.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://downloads.sourceforge.net/project/bbmap/BBMap_39.03.tar.gz?ts=gAAAAABlSoMDsp7-l_3LuZueR4AXxM4YFz1Y-bAzErElyQrDbFs7Hj0lvzXn_Kc3Xqd0VTgHckVGOUCBSRUCexv4kLZ-nhzzRw%3D%3D&use_mirror=gigenet&r= [following]\n",
            "--2023-11-07 18:33:39--  https://downloads.sourceforge.net/project/bbmap/BBMap_39.03.tar.gz?ts=gAAAAABlSoMDsp7-l_3LuZueR4AXxM4YFz1Y-bAzErElyQrDbFs7Hj0lvzXn_Kc3Xqd0VTgHckVGOUCBSRUCexv4kLZ-nhzzRw%3D%3D&use_mirror=gigenet&r=\n",
            "Resolving downloads.sourceforge.net (downloads.sourceforge.net)... 204.68.111.105\n",
            "Connecting to downloads.sourceforge.net (downloads.sourceforge.net)|204.68.111.105|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://gigenet.dl.sourceforge.net/project/bbmap/BBMap_39.03.tar.gz [following]\n",
            "--2023-11-07 18:33:39--  https://gigenet.dl.sourceforge.net/project/bbmap/BBMap_39.03.tar.gz\n",
            "Resolving gigenet.dl.sourceforge.net (gigenet.dl.sourceforge.net)... 69.65.16.142\n",
            "Connecting to gigenet.dl.sourceforge.net (gigenet.dl.sourceforge.net)|69.65.16.142|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 12081369 (12M) [application/x-gzip]\n",
            "Saving to: ‘BBMap_39.03.tar.gz’\n",
            "\n",
            "BBMap_39.03.tar.gz  100%[===================>]  11.52M  13.3MB/s    in 0.9s    \n",
            "\n",
            "2023-11-07 18:33:40 (13.3 MB/s) - ‘BBMap_39.03.tar.gz’ saved [12081369/12081369]\n",
            "\n",
            "bbmap/\n",
            "bbmap/.gitignore\n",
            "bbmap/addadapters.sh\n",
            "bbmap/addssu.sh\n",
            "bbmap/adjusthomopolymers.sh\n",
            "bbmap/alltoall.sh\n",
            "bbmap/analyzeaccession.sh\n",
            "bbmap/analyzegenes.sh\n",
            "bbmap/analyzesketchresults.sh\n",
            "bbmap/applyvariants.sh\n",
            "bbmap/a_sample_mt.sh\n",
            "bbmap/bbcms.sh\n",
            "bbmap/bbcountunique.sh\n",
            "bbmap/bbcrisprfinder.sh\n",
            "bbmap/bbduk.sh\n",
            "bbmap/bbest.sh\n",
            "bbmap/bbfakereads.sh\n",
            "bbmap/bbmap.sh\n",
            "bbmap/bbmapskimmer.sh\n",
            "bbmap/bbmask.sh\n",
            "bbmap/bbmerge-auto.sh\n",
            "bbmap/bbmerge.sh\n",
            "bbmap/bbnorm.sh\n",
            "bbmap/bbrealign.sh\n",
            "bbmap/bbrename.sh\n",
            "bbmap/bbsketch.sh\n",
            "bbmap/bbsort.sh\n",
            "bbmap/bbsplit.sh\n",
            "bbmap/bbsplitpairs.sh\n",
            "bbmap/bbstats.sh\n",
            "bbmap/bbversion.sh\n",
            "bbmap/bbwrap.sh\n",
            "bbmap/bitbucket-pipelines.yml\n",
            "bbmap/bloomfilter.sh\n",
            "bbmap/bloomfilterparser.sh\n",
            "bbmap/calcmem.sh\n",
            "bbmap/calctruequality.sh\n",
            "bbmap/callgenes.sh\n",
            "bbmap/callpeaks.sh\n",
            "bbmap/callvariants.sh\n",
            "bbmap/callvariants2.sh\n",
            "bbmap/checkstrand.sh\n",
            "bbmap/clumpify.sh\n",
            "bbmap/commonkmers.sh\n",
            "bbmap/comparegff.sh\n",
            "bbmap/comparesketch.sh\n",
            "bbmap/comparessu.sh\n",
            "bbmap/comparevcf.sh\n",
            "bbmap/config/\n",
            "bbmap/config/cluster16s.txt\n",
            "bbmap/config/filter16s.txt\n",
            "bbmap/config/histograms.txt\n",
            "bbmap/config/recalibrate.txt\n",
            "bbmap/config/rnaseq.txt\n",
            "bbmap/config/trimadapters.txt\n",
            "bbmap/consect.sh\n",
            "bbmap/consensus.sh\n",
            "bbmap/copyfile.sh\n",
            "bbmap/countbarcodes.sh\n",
            "bbmap/countbarcodes2.sh\n",
            "bbmap/countduplicates.sh\n",
            "bbmap/countgc.sh\n",
            "bbmap/countsharedlines.sh\n",
            "bbmap/crossblock.sh\n",
            "bbmap/crosscontaminate.sh\n",
            "bbmap/current/\n",
            "bbmap/current/align2/\n",
            "bbmap/current/align2/AbstractIndex.class\n",
            "bbmap/current/align2/AbstractIndex.java\n",
            "bbmap/current/align2/AbstractMapper.class\n",
            "bbmap/current/align2/AbstractMapper.java\n",
            "bbmap/current/align2/AbstractMapThread.class\n",
            "bbmap/current/align2/AbstractMapThread.java\n",
            "bbmap/current/align2/BandedAligner.class\n",
            "bbmap/current/align2/BandedAligner.java\n",
            "bbmap/current/align2/BandedAlignerConcrete.class\n",
            "bbmap/current/align2/BandedAlignerConcrete.java\n",
            "bbmap/current/align2/BandedAlignerJNI.class\n",
            "bbmap/current/align2/BandedAlignerJNI.java\n",
            "bbmap/current/align2/BBIndex.class\n",
            "bbmap/current/align2/BBIndex.java\n",
            "bbmap/current/align2/BBIndex5.class\n",
            "bbmap/current/align2/BBIndex5.java\n",
            "bbmap/current/align2/BBIndexAcc.class\n",
            "bbmap/current/align2/BBIndexAcc.java\n",
            "bbmap/current/align2/BBIndexPacBio.class\n",
            "bbmap/current/align2/BBIndexPacBio.java\n",
            "bbmap/current/align2/BBIndexPacBioSkimmer.class\n",
            "bbmap/current/align2/BBIndexPacBioSkimmer.java\n",
            "bbmap/current/align2/BBMap.class\n",
            "bbmap/current/align2/BBMap.java\n",
            "bbmap/current/align2/BBMap5.class\n",
            "bbmap/current/align2/BBMap5.java\n",
            "bbmap/current/align2/BBMapAcc.class\n",
            "bbmap/current/align2/BBMapAcc.java\n",
            "bbmap/current/align2/BBMapPacBio.class\n",
            "bbmap/current/align2/BBMapPacBio.java\n",
            "bbmap/current/align2/BBMapPacBioSkimmer.class\n",
            "bbmap/current/align2/BBMapPacBioSkimmer.java\n",
            "bbmap/current/align2/BBMapThread$1.class\n",
            "bbmap/current/align2/BBMapThread.class\n",
            "bbmap/current/align2/BBMapThread.java\n",
            "bbmap/current/align2/BBMapThread5$1.class\n",
            "bbmap/current/align2/BBMapThread5.class\n",
            "bbmap/current/align2/BBMapThread5.java\n",
            "bbmap/current/align2/BBMapThreadAcc$1.class\n",
            "bbmap/current/align2/BBMapThreadAcc.class\n",
            "bbmap/current/align2/BBMapThreadAcc.java\n",
            "bbmap/current/align2/BBMapThreadPacBio$1.class\n",
            "bbmap/current/align2/BBMapThreadPacBio.class\n",
            "bbmap/current/align2/BBMapThreadPacBio.java\n",
            "bbmap/current/align2/BBMapThreadPacBioSkimmer.class\n",
            "bbmap/current/align2/BBMapThreadPacBioSkimmer.java\n",
            "bbmap/current/align2/BBSplitter$1.class\n",
            "bbmap/current/align2/BBSplitter$SetCount.class\n",
            "bbmap/current/align2/BBSplitter.class\n",
            "bbmap/current/align2/BBSplitter.java\n",
            "bbmap/current/align2/BBWrap.class\n",
            "bbmap/current/align2/BBWrap.java\n",
            "bbmap/current/align2/Blacklist.class\n",
            "bbmap/current/align2/Blacklist.java\n",
            "bbmap/current/align2/Block.class\n",
            "bbmap/current/align2/Block.java\n",
            "bbmap/current/align2/ChromLoadThread.class\n",
            "bbmap/current/align2/ChromLoadThread.java\n",
            "bbmap/current/align2/CompareSamFiles$1.class\n",
            "bbmap/current/align2/CompareSamFiles.class\n",
            "bbmap/current/align2/CompareSamFiles.java\n",
            "bbmap/current/align2/CompressString.class\n",
            "bbmap/current/align2/CompressString.java\n",
            "bbmap/current/align2/GapTools$Range.class\n",
            "bbmap/current/align2/GapTools.class\n",
            "bbmap/current/align2/GapTools.java\n",
            "bbmap/current/align2/GradeSamFile$1.class\n",
            "bbmap/current/align2/GradeSamFile.class\n",
            "bbmap/current/align2/GradeSamFile.java\n",
            "bbmap/current/align2/Index.class\n",
            "bbmap/current/align2/Index.java\n",
            "bbmap/current/align2/IndexMaker4$BlockMaker$CountThread.class\n",
            "bbmap/current/align2/IndexMaker4$BlockMaker.class\n",
            "bbmap/current/align2/IndexMaker4.class\n",
            "bbmap/current/align2/IndexMaker4.java\n",
            "bbmap/current/align2/IndexMaker5$BlockMaker$CountThread.class\n",
            "bbmap/current/align2/IndexMaker5$BlockMaker.class\n",
            "bbmap/current/align2/IndexMaker5.class\n",
            "bbmap/current/align2/IndexMaker5.java\n",
            "bbmap/current/align2/KeyRing.class\n",
            "bbmap/current/align2/KeyRing.java\n",
            "bbmap/current/align2/MakeQualityHistogram.class\n",
            "bbmap/current/align2/MakeQualityHistogram.java\n",
            "bbmap/current/align2/MakeRocCurve$1.class\n",
            "bbmap/current/align2/MakeRocCurve.class\n",
            "bbmap/current/align2/MakeRocCurve.java\n",
            "bbmap/current/align2/MSA.class\n",
            "bbmap/current/align2/MSA.java\n",
            "bbmap/current/align2/MultiStateAligner10ts.class\n",
            "bbmap/current/align2/MultiStateAligner10ts.java\n",
            "bbmap/current/align2/MultiStateAligner11ts.class\n",
            "bbmap/current/align2/MultiStateAligner11ts.java\n",
            "bbmap/current/align2/MultiStateAligner11tsJNI.class\n",
            "bbmap/current/align2/MultiStateAligner11tsJNI.java\n",
            "bbmap/current/align2/MultiStateAligner9Flat.class\n",
            "bbmap/current/align2/MultiStateAligner9Flat.java\n",
            "bbmap/current/align2/MultiStateAligner9PacBio.class\n",
            "bbmap/current/align2/MultiStateAligner9PacBio.java\n",
            "bbmap/current/align2/MultiStateAligner9ts.class\n",
            "bbmap/current/align2/MultiStateAligner9ts.java\n",
            "bbmap/current/align2/MultiStateAligner9XFlat.class\n",
            "bbmap/current/align2/MultiStateAligner9XFlat.java\n",
            "bbmap/current/align2/NeedlemanWunsch.class\n",
            "bbmap/current/align2/NeedlemanWunsch.java\n",
            "bbmap/current/align2/PackedHeap.class\n",
            "bbmap/current/align2/PackedHeap.java\n",
            "bbmap/current/align2/Pointer.class\n",
            "bbmap/current/align2/Pointer.java\n",
            "bbmap/current/align2/PrintTime.class\n",
            "bbmap/current/align2/PrintTime.java\n",
            "bbmap/current/align2/Quad.class\n",
            "bbmap/current/align2/Quad.java\n",
            "bbmap/current/align2/Quad64.class\n",
            "bbmap/current/align2/Quad64.java\n",
            "bbmap/current/align2/Quad64Heap.class\n",
            "bbmap/current/align2/Quad64Heap.java\n",
            "bbmap/current/align2/QuadHeap.class\n",
            "bbmap/current/align2/QuadHeap.java\n",
            "bbmap/current/align2/QualityTools.class\n",
            "bbmap/current/align2/QualityTools.java\n",
            "bbmap/current/align2/RandomReads3$1.class\n",
            "bbmap/current/align2/RandomReads3.class\n",
            "bbmap/current/align2/RandomReads3.java\n",
            "bbmap/current/align2/ReformatBatchOutput.class\n",
            "bbmap/current/align2/ReformatBatchOutput.java\n",
            "bbmap/current/align2/ReformatBatchOutput2.class\n",
            "bbmap/current/align2/ReformatBatchOutput2.java\n",
            "bbmap/current/align2/RefToIndex.class\n",
            "bbmap/current/align2/RefToIndex.java\n",
            "bbmap/current/align2/Solver.class\n",
            "bbmap/current/align2/Solver.java\n",
            "bbmap/current/align2/SplitMappedReads.class\n",
            "bbmap/current/align2/SplitMappedReads.java\n",
            "bbmap/current/align2/TranslateColorspaceRead.class\n",
            "bbmap/current/align2/TranslateColorspaceRead.java\n",
            "bbmap/current/aligner/\n",
            "bbmap/current/aligner/Aligner.class\n",
            "bbmap/current/aligner/Aligner.java\n",
            "bbmap/current/aligner/Alignment.class\n",
            "bbmap/current/aligner/Alignment.java\n",
            "bbmap/current/aligner/AlignmentResult.class\n",
            "bbmap/current/aligner/AlignmentResult.java\n",
            "bbmap/current/aligner/AllToAll$ProcessThread.class\n",
            "bbmap/current/aligner/AllToAll.class\n",
            "bbmap/current/aligner/AllToAll.java\n",
            "bbmap/current/aligner/FlatAligner.class\n",
            "bbmap/current/aligner/FlatAligner.java\n",
            "bbmap/current/aligner/FlatAligner2.class\n",
            "bbmap/current/aligner/FlatAligner2.java\n",
            "bbmap/current/aligner/MultiStateAligner9PacBioAdapter.class\n",
            "bbmap/current/aligner/MultiStateAligner9PacBioAdapter.java\n",
            "bbmap/current/aligner/MultiStateAligner9PacBioAdapter2.class\n",
            "bbmap/current/aligner/MultiStateAligner9PacBioAdapter2.java\n",
            "bbmap/current/aligner/MultiStateAligner9PacBioAdapter3.class\n",
            "bbmap/current/aligner/MultiStateAligner9PacBioAdapter3.java\n",
            "bbmap/current/aligner/MultiStateAligner9PacBioAdapter_WithBarriers.class\n",
            "bbmap/current/aligner/MultiStateAligner9PacBioAdapter_WithBarriers.java\n",
            "bbmap/current/aligner/SingleStateAlignerFlat.class\n",
            "bbmap/current/aligner/SingleStateAlignerFlat.java\n",
            "bbmap/current/aligner/SingleStateAlignerFlat2.class\n",
            "bbmap/current/aligner/SingleStateAlignerFlat2.java\n",
            "bbmap/current/aligner/SingleStateAlignerFlat2Amino.class\n",
            "bbmap/current/aligner/SingleStateAlignerFlat2Amino.java\n",
            "bbmap/current/aligner/SingleStateAlignerFlat2_1D.class\n",
            "bbmap/current/aligner/SingleStateAlignerFlat2_1D.java\n",
            "bbmap/current/aligner/SingleStateAlignerFlat3.class\n",
            "bbmap/current/aligner/SingleStateAlignerFlat3.java\n",
            "bbmap/current/aligner/SingleStateAlignerFlatFloat.class\n",
            "bbmap/current/aligner/SingleStateAlignerFlatFloat.java\n",
            "bbmap/current/aligner/SingleStateAlignerPacBioAdapter.class\n",
            "bbmap/current/aligner/SingleStateAlignerPacBioAdapter.java\n",
            "bbmap/current/assemble/\n",
            "bbmap/current/assemble/AbstractBuildThread.class\n",
            "bbmap/current/assemble/AbstractBuildThread.java\n",
            "bbmap/current/assemble/AbstractExploreThread.class\n",
            "bbmap/current/assemble/AbstractExploreThread.java\n",
            "bbmap/current/assemble/AbstractProcessContigThread.class\n",
            "bbmap/current/assemble/AbstractProcessContigThread.java\n",
            "bbmap/current/assemble/AbstractRemoveThread$RemoveThread1.class\n",
            "bbmap/current/assemble/AbstractRemoveThread$RemoveThread2.class\n",
            "bbmap/current/assemble/AbstractRemoveThread.class\n",
            "bbmap/current/assemble/AbstractRemoveThread.java\n",
            "bbmap/current/assemble/AbstractShaveThread.class\n",
            "bbmap/current/assemble/AbstractShaveThread.java\n",
            "bbmap/current/assemble/BubblePopper.class\n",
            "bbmap/current/assemble/BubblePopper.java\n",
            "bbmap/current/assemble/Contig.class\n",
            "bbmap/current/assemble/Contig.java\n",
            "bbmap/current/assemble/Edge.class\n",
            "bbmap/current/assemble/Edge.java\n",
            "bbmap/current/assemble/ErrorTracker.class\n",
            "bbmap/current/assemble/ErrorTracker.java\n",
            "bbmap/current/assemble/KmerCompressor$BuildThread.class\n",
            "bbmap/current/assemble/KmerCompressor.class\n",
            "bbmap/current/assemble/KmerCompressor.java\n",
            "bbmap/current/assemble/Postfilter.class\n",
            "bbmap/current/assemble/Postfilter.java\n",
            "bbmap/current/assemble/Rollback.class\n",
            "bbmap/current/assemble/Rollback.java\n",
            "bbmap/current/assemble/ShaveObject.class\n",
            "bbmap/current/assemble/ShaveObject.java\n",
            "bbmap/current/assemble/Shaver.class\n",
            "bbmap/current/assemble/Shaver.java\n",
            "bbmap/current/assemble/Shaver1$ExploreThread.class\n",
            "bbmap/current/assemble/Shaver1$ShaveThread.class\n",
            "bbmap/current/assemble/Shaver1.class\n",
            "bbmap/current/assemble/Shaver1.java\n",
            "bbmap/current/assemble/Shaver2$ExploreThread.class\n",
            "bbmap/current/assemble/Shaver2$ShaveThread.class\n",
            "bbmap/current/assemble/Shaver2.class\n",
            "bbmap/current/assemble/Shaver2.java\n",
            "bbmap/current/assemble/TadPipe.class\n",
            "bbmap/current/assemble/TadPipe.java\n",
            "bbmap/current/assemble/Tadpole$DumpKmersThread.class\n",
            "bbmap/current/assemble/Tadpole$ExtendThread.class\n",
            "bbmap/current/assemble/Tadpole$MakeKhistThread.class\n",
            "bbmap/current/assemble/Tadpole.class\n",
            "bbmap/current/assemble/Tadpole.java\n",
            "bbmap/current/assemble/Tadpole1$BuildThread.class\n",
            "bbmap/current/assemble/Tadpole1$ProcessContigThread.class\n",
            "bbmap/current/assemble/Tadpole1.class\n",
            "bbmap/current/assemble/Tadpole1.java\n",
            "bbmap/current/assemble/Tadpole2$BuildThread.class\n",
            "bbmap/current/assemble/Tadpole2$ProcessContigThread.class\n",
            "bbmap/current/assemble/Tadpole2.class\n",
            "bbmap/current/assemble/Tadpole2.java\n",
            "bbmap/current/assemble/TadpoleWrapper$1.class\n",
            "bbmap/current/assemble/TadpoleWrapper$Record.class\n",
            "bbmap/current/assemble/TadpoleWrapper.class\n",
            "bbmap/current/assemble/TadpoleWrapper.java\n",
            "bbmap/current/barcode/\n",
            "bbmap/current/barcode/AnalyzeBarcodes.class\n",
            "bbmap/current/barcode/AnalyzeBarcodes.java\n",
            "bbmap/current/barcode/Barcode.class\n",
            "bbmap/current/barcode/Barcode.java\n",
            "bbmap/current/barcode/BarcodeMappingStats.class\n",
            "bbmap/current/barcode/BarcodeMappingStats.java\n",
            "bbmap/current/barcode/BarcodeStats.class\n",
            "bbmap/current/barcode/BarcodeStats.java\n",
            "bbmap/current/barcode/CorrelateBarcodes.class\n",
            "bbmap/current/barcode/CorrelateBarcodes.java\n",
            "bbmap/current/barcode/CountBarcodes.class\n",
            "bbmap/current/barcode/CountBarcodes.java\n",
            "bbmap/current/barcode/CountBarcodes2.class\n",
            "bbmap/current/barcode/CountBarcodes2.java\n",
            "bbmap/current/bbmin/\n",
            "bbmap/current/bbmin/LongHashSet.class\n",
            "bbmap/current/bbmin/LongHashSet.java\n",
            "bbmap/current/bbmin/LongList.class\n",
            "bbmap/current/bbmin/LongList.java\n",
            "bbmap/current/bbmin/Minimizer.class\n",
            "bbmap/current/bbmin/Minimizer.java\n",
            "bbmap/current/bloom/\n",
            "bbmap/current/bloom/BloomFilter.class\n",
            "bbmap/current/bloom/BloomFilter.java\n",
            "bbmap/current/bloom/BloomFilterCorrector.class\n",
            "bbmap/current/bloom/BloomFilterCorrector.java\n",
            "bbmap/current/bloom/BloomFilterCorrectorWrapper$ProcessThread.class\n",
            "bbmap/current/bloom/BloomFilterCorrectorWrapper.class\n",
            "bbmap/current/bloom/BloomFilterCorrectorWrapper.java\n",
            "bbmap/current/bloom/BloomFilterWrapper$ProcessThread.class\n",
            "bbmap/current/bloom/BloomFilterWrapper.class\n",
            "bbmap/current/bloom/BloomFilterWrapper.java\n",
            "bbmap/current/bloom/ErrorCorrect$1.class\n",
            "bbmap/current/bloom/ErrorCorrect.class\n",
            "bbmap/current/bloom/ErrorCorrect.java\n",
            "bbmap/current/bloom/IndexCounter$CountThread.class\n",
            "bbmap/current/bloom/IndexCounter.class\n",
            "bbmap/current/bloom/IndexCounter.java\n",
            "bbmap/current/bloom/KCountArray$AllocThread.class\n",
            "bbmap/current/bloom/KCountArray.class\n",
            "bbmap/current/bloom/KCountArray.java\n",
            "bbmap/current/bloom/KCountArray2.class\n",
            "bbmap/current/bloom/KCountArray2.java\n",
            "bbmap/current/bloom/KCountArray3.class\n",
            "bbmap/current/bloom/KCountArray3.java\n",
            "bbmap/current/bloom/KCountArray4.class\n",
            "bbmap/current/bloom/KCountArray4.java\n",
            "bbmap/current/bloom/KCountArray4MT$WriteThread.class\n",
            "bbmap/current/bloom/KCountArray4MT.class\n",
            "bbmap/current/bloom/KCountArray4MT.java\n",
            "bbmap/current/bloom/KCountArray5MT$WriteThread.class\n",
            "bbmap/current/bloom/KCountArray5MT.class\n",
            "bbmap/current/bloom/KCountArray5MT.java\n",
            "bbmap/current/bloom/KCountArray6MT$WriteThread.class\n",
            "bbmap/current/bloom/KCountArray6MT.class\n",
            "bbmap/current/bloom/KCountArray6MT.java\n",
            "bbmap/current/bloom/KCountArray7MT$WriteThread.class\n",
            "bbmap/current/bloom/KCountArray7MT.class\n",
            "bbmap/current/bloom/KCountArray7MT.java\n",
            "bbmap/current/bloom/KCountArray7MTA$CountUsedThread.class\n",
            "bbmap/current/bloom/KCountArray7MTA.class\n",
            "bbmap/current/bloom/KCountArray7MTA.java\n",
            "bbmap/current/bloom/KCountArray8MT$WriteThread.class\n",
            "bbmap/current/bloom/KCountArray8MT.class\n",
            "bbmap/current/bloom/KCountArray8MT.java\n",
            "bbmap/current/bloom/KmerCount3.class\n",
            "bbmap/current/bloom/KmerCount3.java\n",
            "bbmap/current/bloom/KmerCount4.class\n",
            "bbmap/current/bloom/KmerCount4.java\n",
            "bbmap/current/bloom/KmerCount5.class\n",
            "bbmap/current/bloom/KmerCount5.java\n",
            "bbmap/current/bloom/KmerCount6.class\n",
            "bbmap/current/bloom/KmerCount6.java\n",
            "bbmap/current/bloom/KmerCountAbstract.class\n",
            "bbmap/current/bloom/KmerCountAbstract.java\n",
            "bbmap/current/bloom/LargeKmerCount.class\n",
            "bbmap/current/bloom/LargeKmerCount.java\n",
            "bbmap/current/bloom/LargeKmerCount2.class\n",
            "bbmap/current/bloom/LargeKmerCount2.java\n",
            "bbmap/current/bloom/ParseBloomFilter.class\n",
            "bbmap/current/bloom/ParseBloomFilter.java\n",
            "bbmap/current/bloom/ReadCounter$1.class\n",
            "bbmap/current/bloom/ReadCounter$CountThread.class\n",
            "bbmap/current/bloom/ReadCounter.class\n",
            "bbmap/current/bloom/ReadCounter.java\n",
            "bbmap/current/bloom/TestLargeKmer.class\n",
            "bbmap/current/bloom/TestLargeKmer.java\n",
            "bbmap/current/cardinality/\n",
            "bbmap/current/cardinality/BBLog.class\n",
            "bbmap/current/cardinality/BBLog.java\n",
            "bbmap/current/cardinality/BBLog_simple.class\n",
            "bbmap/current/cardinality/BBLog_simple.java\n",
            "bbmap/current/cardinality/CardinalityTracker.class\n",
            "bbmap/current/cardinality/CardinalityTracker.java\n",
            "bbmap/current/cardinality/LogLog.class\n",
            "bbmap/current/cardinality/LogLog.java\n",
            "bbmap/current/cardinality/LogLog16.class\n",
            "bbmap/current/cardinality/LogLog16.java\n",
            "bbmap/current/cardinality/LogLog2.class\n",
            "bbmap/current/cardinality/LogLog2.java\n",
            "bbmap/current/cardinality/LogLog8.class\n",
            "bbmap/current/cardinality/LogLog8.java\n",
            "bbmap/current/cardinality/LogLog8_simple.class\n",
            "bbmap/current/cardinality/LogLog8_simple.java\n",
            "bbmap/current/cardinality/LogLogWrapper$LogLogThread.class\n",
            "bbmap/current/cardinality/LogLogWrapper.class\n",
            "bbmap/current/cardinality/LogLogWrapper.java\n",
            "bbmap/current/cardinality/LogLog_old$LogLogWrapper$LogLogThread.class\n",
            "bbmap/current/cardinality/LogLog_old$LogLogWrapper.class\n",
            "bbmap/current/cardinality/LogLog_old.class\n",
            "bbmap/current/cardinality/LogLog_old.java\n",
            "bbmap/current/cardinality/MultiLogLog.class\n",
            "bbmap/current/cardinality/MultiLogLog.java\n",
            "bbmap/current/clump/\n",
            "bbmap/current/clump/Clump.class\n",
            "bbmap/current/clump/Clump.java\n",
            "bbmap/current/clump/Clumpify.class\n",
            "bbmap/current/clump/Clumpify.java\n",
            "bbmap/current/clump/ClumpList$ClumpThread.class\n",
            "bbmap/current/clump/ClumpList$ProcessThread.class\n",
            "bbmap/current/clump/ClumpList.class\n",
            "bbmap/current/clump/ClumpList.java\n",
            "bbmap/current/clump/ClumpTools.class\n",
            "bbmap/current/clump/ClumpTools.java\n",
            "bbmap/current/clump/Hasher.class\n",
            "bbmap/current/clump/Hasher.java\n",
            "bbmap/current/clump/KmerComparator$HashThread.class\n",
            "bbmap/current/clump/KmerComparator.class\n",
            "bbmap/current/clump/KmerComparator.java\n",
            "bbmap/current/clump/KmerComparator2.class\n",
            "bbmap/current/clump/KmerComparator2.java\n",
            "bbmap/current/clump/KmerComparatorX.class\n",
            "bbmap/current/clump/KmerComparatorX.java\n",
            "bbmap/current/clump/KmerComparatorY.class\n",
            "bbmap/current/clump/KmerComparatorY.java\n",
            "bbmap/current/clump/KmerReduce$HashThread.class\n",
            "bbmap/current/clump/KmerReduce.class\n",
            "bbmap/current/clump/KmerReduce.java\n",
            "bbmap/current/clump/KmerSort$FetchThread1.class\n",
            "bbmap/current/clump/KmerSort$HashSplitThread.class\n",
            "bbmap/current/clump/KmerSort.class\n",
            "bbmap/current/clump/KmerSort.java\n",
            "bbmap/current/clump/KmerSort1.class\n",
            "bbmap/current/clump/KmerSort1.java\n",
            "bbmap/current/clump/KmerSort2.class\n",
            "bbmap/current/clump/KmerSort2.java\n",
            "bbmap/current/clump/KmerSort3$FetchThread3$FetchSubThread.class\n",
            "bbmap/current/clump/KmerSort3$FetchThread3.class\n",
            "bbmap/current/clump/KmerSort3.class\n",
            "bbmap/current/clump/KmerSort3.java\n",
            "bbmap/current/clump/KmerSplit$HashThread.class\n",
            "bbmap/current/clump/KmerSplit.class\n",
            "bbmap/current/clump/KmerSplit.java\n",
            "bbmap/current/clump/PivotSet$HashThread.class\n",
            "bbmap/current/clump/PivotSet.class\n",
            "bbmap/current/clump/PivotSet.java\n",
            "bbmap/current/clump/ReadKey.class\n",
            "bbmap/current/clump/ReadKey.java\n",
            "bbmap/current/clump/Splitter.class\n",
            "bbmap/current/clump/Splitter.java\n",
            "bbmap/current/clump/StreamToOutput.class\n",
            "bbmap/current/clump/StreamToOutput.java\n",
            "bbmap/current/cluster/\n",
            "bbmap/current/cluster/Cluster.class\n",
            "bbmap/current/cluster/Cluster.java\n",
            "bbmap/current/cluster/ClusterTools.class\n",
            "bbmap/current/cluster/ClusterTools.java\n",
            "bbmap/current/cluster/MergeReadHeaders.class\n",
            "bbmap/current/cluster/MergeReadHeaders.java\n",
            "bbmap/current/cluster/ReadTag.class\n",
            "bbmap/current/cluster/ReadTag.java\n",
            "bbmap/current/cluster/ReclusterByKmer$ClusterThread.class\n",
            "bbmap/current/cluster/ReclusterByKmer.class\n",
            "bbmap/current/cluster/ReclusterByKmer.java\n",
            "bbmap/current/consensus/\n",
            "bbmap/current/consensus/BaseGraph.class\n",
            "bbmap/current/consensus/BaseGraph.java\n",
            "bbmap/current/consensus/BaseGraphPart.class\n",
            "bbmap/current/consensus/BaseGraphPart.java\n",
            "bbmap/current/consensus/BaseNode.class\n",
            "bbmap/current/consensus/BaseNode.java\n",
            "bbmap/current/consensus/ConsensusMaker$ProcessThread.class\n",
            "bbmap/current/consensus/ConsensusMaker.class\n",
            "bbmap/current/consensus/ConsensusMaker.java\n",
            "bbmap/current/consensus/ConsensusObject.class\n",
            "bbmap/current/consensus/ConsensusObject.java\n",
            "bbmap/current/consensus/FixScaffoldGaps$ProcessThread.class\n",
            "bbmap/current/consensus/FixScaffoldGaps$Scaffold.class\n",
            "bbmap/current/consensus/FixScaffoldGaps.class\n",
            "bbmap/current/consensus/FixScaffoldGaps.java\n",
            "bbmap/current/consensus/Lilypad$Contig.class\n",
            "bbmap/current/consensus/Lilypad$Edge.class\n",
            "bbmap/current/consensus/Lilypad$ProcessThread.class\n",
            "bbmap/current/consensus/Lilypad.class\n",
            "bbmap/current/consensus/Lilypad.java\n",
            "bbmap/current/covid/\n",
            "bbmap/current/covid/SummarizeCoverage.class\n",
            "bbmap/current/covid/SummarizeCoverage.java\n",
            "bbmap/current/dna/\n",
            "bbmap/current/dna/AminoAcid.class\n",
            "bbmap/current/dna/AminoAcid.java\n",
            "bbmap/current/dna/ChromArrayMaker.class\n",
            "bbmap/current/dna/ChromArrayMaker.java\n",
            "bbmap/current/dna/ChromosomeArray.class\n",
            "bbmap/current/dna/ChromosomeArray.java\n",
            "bbmap/current/dna/ChromToFasta$1.class\n",
            "bbmap/current/dna/ChromToFasta.class\n",
            "bbmap/current/dna/ChromToFasta.java\n",
            "bbmap/current/dna/Data.class\n",
            "bbmap/current/dna/Data.java\n",
            "bbmap/current/dna/Exon.class\n",
            "bbmap/current/dna/Exon.java\n",
            "bbmap/current/dna/FastaToChromArrays2$1.class\n",
            "bbmap/current/dna/FastaToChromArrays2.class\n",
            "bbmap/current/dna/FastaToChromArrays2.java\n",
            "bbmap/current/dna/Gene.class\n",
            "bbmap/current/dna/Gene.java\n",
            "bbmap/current/dna/GeneSet.class\n",
            "bbmap/current/dna/GeneSet.java\n",
            "bbmap/current/dna/Matrix.class\n",
            "bbmap/current/dna/Matrix.java\n",
            "bbmap/current/dna/Motif.class\n",
            "bbmap/current/dna/Motif.java\n",
            "bbmap/current/dna/MotifMulti.class\n",
            "bbmap/current/dna/MotifMulti.java\n",
            "bbmap/current/dna/MotifProbsN.class\n",
            "bbmap/current/dna/MotifProbsN.java\n",
            "bbmap/current/dna/MotifSimple.class\n",
            "bbmap/current/dna/MotifSimple.java\n",
            "bbmap/current/dna/Scaffold.class\n",
            "bbmap/current/dna/Scaffold.java\n",
            "bbmap/current/dna/ScafLoc.class\n",
            "bbmap/current/dna/ScafLoc.java\n",
            "bbmap/current/driver/\n",
            "bbmap/current/driver/BBVersion.class\n",
            "bbmap/current/driver/BBVersion.java\n",
            "bbmap/current/driver/ClearRam.class\n",
            "bbmap/current/driver/ClearRam.java\n",
            "bbmap/current/driver/CollateSpikeIn.class\n",
            "bbmap/current/driver/CollateSpikeIn.java\n",
            "bbmap/current/driver/CompareReferenceGenomes.class\n",
            "bbmap/current/driver/CompareReferenceGenomes.java\n",
            "bbmap/current/driver/CompareSequences.class\n",
            "bbmap/current/driver/CompareSequences.java\n",
            "bbmap/current/driver/ConcatenateFiles.class\n",
            "bbmap/current/driver/ConcatenateFiles.java\n",
            "bbmap/current/driver/ConcatenateTextFiles$1.class\n",
            "bbmap/current/driver/ConcatenateTextFiles$WriteThread.class\n",
            "bbmap/current/driver/ConcatenateTextFiles.class\n",
            "bbmap/current/driver/ConcatenateTextFiles.java\n",
            "bbmap/current/driver/Concatenator.class\n",
            "bbmap/current/driver/Concatenator.java\n",
            "bbmap/current/driver/ConvertSamToAln.class\n",
            "bbmap/current/driver/ConvertSamToAln.java\n",
            "bbmap/current/driver/CorrelateIdentity.class\n",
            "bbmap/current/driver/CorrelateIdentity.java\n",
            "bbmap/current/driver/CountRNAs.class\n",
            "bbmap/current/driver/CountRNAs.java\n",
            "bbmap/current/driver/CountSharedLines.class\n",
            "bbmap/current/driver/CountSharedLines.java\n",
            "bbmap/current/driver/EstherFilter.class\n",
            "bbmap/current/driver/EstherFilter.java\n",
            "bbmap/current/driver/FilterAssemblySummary.class\n",
            "bbmap/current/driver/FilterAssemblySummary.java\n",
            "bbmap/current/driver/FilterLines.class\n",
            "bbmap/current/driver/FilterLines.java\n",
            "bbmap/current/driver/FilterReadsByName.class\n",
            "bbmap/current/driver/FilterReadsByName.java\n",
            "bbmap/current/driver/FindMotifs.class\n",
            "bbmap/current/driver/FindMotifs.java\n",
            "bbmap/current/driver/FixChr.class\n",
            "bbmap/current/driver/FixChr.java\n",
            "bbmap/current/driver/FixDumbFile.class\n",
            "bbmap/current/driver/FixDumbFile.java\n",
            "bbmap/current/driver/GenerateNoCallsFromCoverage.class\n",
            "bbmap/current/driver/GenerateNoCallsFromCoverage.java\n",
            "bbmap/current/driver/GetSequence.class\n",
            "bbmap/current/driver/GetSequence.java\n",
            "bbmap/current/driver/GetUniquePrefixes.class\n",
            "bbmap/current/driver/GetUniquePrefixes.java\n",
            "bbmap/current/driver/Grep.class\n",
            "bbmap/current/driver/Grep.java\n",
            "bbmap/current/driver/Life.class\n",
            "bbmap/current/driver/Life.java\n",
            "bbmap/current/driver/LineCount.class\n",
            "bbmap/current/driver/LineCount.java\n",
            "bbmap/current/driver/LoadReads.class\n",
            "bbmap/current/driver/LoadReads.java\n",
            "bbmap/current/driver/LookAtID.class\n",
            "bbmap/current/driver/LookAtID.java\n",
            "bbmap/current/driver/MakeTestScript.class\n",
            "bbmap/current/driver/MakeTestScript.java\n",
            "bbmap/current/driver/MakeTestScriptScoreOnly.class\n",
            "bbmap/current/driver/MakeTestScriptScoreOnly.java\n",
            "bbmap/current/driver/MeasureGene.class\n",
            "bbmap/current/driver/MeasureGene.java\n",
            "bbmap/current/driver/MergeBigelow.class\n",
            "bbmap/current/driver/MergeBigelow.java\n",
            "bbmap/current/driver/MergeCoverageOTU.class\n",
            "bbmap/current/driver/MergeCoverageOTU.java\n",
            "bbmap/current/driver/MergeTextFiles.class\n",
            "bbmap/current/driver/MergeTextFiles.java\n",
            "bbmap/current/driver/MergeTextFiles2.class\n",
            "bbmap/current/driver/MergeTextFiles2.java\n",
            "bbmap/current/driver/MoveFiles.class\n",
            "bbmap/current/driver/MoveFiles.java\n",
            "bbmap/current/driver/ParseCrossblockResults$ResultsLine.class\n",
            "bbmap/current/driver/ParseCrossblockResults.class\n",
            "bbmap/current/driver/ParseCrossblockResults.java\n",
            "bbmap/current/driver/PlotGC.class\n",
            "bbmap/current/driver/PlotGC.java\n",
            "bbmap/current/driver/PrintEnv.class\n",
            "bbmap/current/driver/PrintEnv.java\n",
            "bbmap/current/driver/ProcessFragMerging.class\n",
            "bbmap/current/driver/ProcessFragMerging.java\n",
            "bbmap/current/driver/ProcessSpeed.class\n",
            "bbmap/current/driver/ProcessSpeed.java\n",
            "bbmap/current/driver/ProcessSpeed2.class\n",
            "bbmap/current/driver/ProcessSpeed2.java\n",
            "bbmap/current/driver/ProcessWebcheck.class\n",
            "bbmap/current/driver/ProcessWebcheck.java\n",
            "bbmap/current/driver/ReduceSilva.class\n",
            "bbmap/current/driver/ReduceSilva.java\n",
            "bbmap/current/driver/RenameAndMux$MuxThread.class\n",
            "bbmap/current/driver/RenameAndMux.class\n",
            "bbmap/current/driver/RenameAndMux.java\n",
            "bbmap/current/driver/RenameByHeader.class\n",
            "bbmap/current/driver/RenameByHeader.java\n",
            "bbmap/current/driver/RenameNcbiToTid.class\n",
            "bbmap/current/driver/RenameNcbiToTid.java\n",
            "bbmap/current/driver/RenameRefseqFiles.class\n",
            "bbmap/current/driver/RenameRefseqFiles.java\n",
            "bbmap/current/driver/Sample.class\n",
            "bbmap/current/driver/Sample.java\n",
            "bbmap/current/driver/Search.class\n",
            "bbmap/current/driver/Search.java\n",
            "bbmap/current/driver/SelectReads.class\n",
            "bbmap/current/driver/SelectReads.java\n",
            "bbmap/current/driver/SniffSplices.class\n",
            "bbmap/current/driver/SniffSplices.java\n",
            "bbmap/current/driver/SummarizeContamReport$ComparatorA.class\n",
            "bbmap/current/driver/SummarizeContamReport$ComparatorB.class\n",
            "bbmap/current/driver/SummarizeContamReport$StringLongLong.class\n",
            "bbmap/current/driver/SummarizeContamReport.class\n",
            "bbmap/current/driver/SummarizeContamReport.java\n",
            "bbmap/current/driver/SummarizeCoverage.class\n",
            "bbmap/current/driver/SummarizeCoverage.java\n",
            "bbmap/current/driver/SummarizeCrossblock.class\n",
            "bbmap/current/driver/SummarizeCrossblock.java\n",
            "bbmap/current/driver/SummarizeMSDIN.class\n",
            "bbmap/current/driver/SummarizeMSDIN.java\n",
            "bbmap/current/driver/SummarizeQuast$Entry.class\n",
            "bbmap/current/driver/SummarizeQuast$QuastSummary.class\n",
            "bbmap/current/driver/SummarizeQuast.class\n",
            "bbmap/current/driver/SummarizeQuast.java\n",
            "bbmap/current/driver/SummarizeSealCrosstalk$Result.class\n",
            "bbmap/current/driver/SummarizeSealCrosstalk.class\n",
            "bbmap/current/driver/SummarizeSealCrosstalk.java\n",
            "bbmap/current/driver/SummarizeSealStats$SealSummary.class\n",
            "bbmap/current/driver/SummarizeSealStats.class\n",
            "bbmap/current/driver/SummarizeSealStats.java\n",
            "bbmap/current/driver/TestCompressionSpeed.class\n",
            "bbmap/current/driver/TestCompressionSpeed.java\n",
            "bbmap/current/driver/TestLockSpeed$1.class\n",
            "bbmap/current/driver/TestLockSpeed$AtomBox.class\n",
            "bbmap/current/driver/TestLockSpeed$CountBox.class\n",
            "bbmap/current/driver/TestLockSpeed$CountThread.class\n",
            "bbmap/current/driver/TestLockSpeed$LockBox.class\n",
            "bbmap/current/driver/TestLockSpeed$VolatileBox.class\n",
            "bbmap/current/driver/TestLockSpeed.class\n",
            "bbmap/current/driver/TestLockSpeed.java\n",
            "bbmap/current/driver/Translator.class\n",
            "bbmap/current/driver/Translator.java\n",
            "bbmap/current/driver/Translator2.class\n",
            "bbmap/current/driver/Translator2.java\n",
            "bbmap/current/driver/TransposeTextFile.class\n",
            "bbmap/current/driver/TransposeTextFile.java\n",
            "bbmap/current/driver/TrimSamFile.class\n",
            "bbmap/current/driver/TrimSamFile.java\n",
            "bbmap/current/fileIO/\n",
            "bbmap/current/fileIO/ArrayFile.class\n",
            "bbmap/current/fileIO/ArrayFile.java\n",
            "bbmap/current/fileIO/ByteFile.class\n",
            "bbmap/current/fileIO/ByteFile.java\n",
            "bbmap/current/fileIO/ByteFile1.class\n",
            "bbmap/current/fileIO/ByteFile1.java\n",
            "bbmap/current/fileIO/ByteFile2$BF1Thread.class\n",
            "bbmap/current/fileIO/ByteFile2.class\n",
            "bbmap/current/fileIO/ByteFile2.java\n",
            "bbmap/current/fileIO/ByteStreamWriter.class\n",
            "bbmap/current/fileIO/ByteStreamWriter.java\n",
            "bbmap/current/fileIO/ChainBlock.class\n",
            "bbmap/current/fileIO/ChainBlock.java\n",
            "bbmap/current/fileIO/ChainLine.class\n",
            "bbmap/current/fileIO/ChainLine.java\n",
            "bbmap/current/fileIO/CompressFiles.class\n",
            "bbmap/current/fileIO/CompressFiles.java\n",
            "bbmap/current/fileIO/CopyFile$1.class\n",
            "bbmap/current/fileIO/CopyFile.class\n",
            "bbmap/current/fileIO/CopyFile.java\n",
            "bbmap/current/fileIO/CopyFiles.class\n",
            "bbmap/current/fileIO/CopyFiles.java\n",
            "bbmap/current/fileIO/CopyFiles2.class\n",
            "bbmap/current/fileIO/CopyFiles2.java\n",
            "bbmap/current/fileIO/FileFormat.class\n",
            "bbmap/current/fileIO/FileFormat.java\n",
            "bbmap/current/fileIO/FindFiles.class\n",
            "bbmap/current/fileIO/FindFiles.java\n",
            "bbmap/current/fileIO/GenericTextFile.class\n",
            "bbmap/current/fileIO/GenericTextFile.java\n",
            "bbmap/current/fileIO/LoadThread.class\n",
            "bbmap/current/fileIO/LoadThread.java\n",
            "bbmap/current/fileIO/MatrixFile.class\n",
            "bbmap/current/fileIO/MatrixFile.java\n",
            "bbmap/current/fileIO/OpenFile.class\n",
            "bbmap/current/fileIO/OpenFile.java\n",
            "bbmap/current/fileIO/PipeThread.class\n",
            "bbmap/current/fileIO/PipeThread.java\n",
            "bbmap/current/fileIO/QuickFile.class\n",
            "bbmap/current/fileIO/QuickFile.java\n",
            "bbmap/current/fileIO/ReadWrite$1.class\n",
            "bbmap/current/fileIO/ReadWrite$WriteObjectThread.class\n",
            "bbmap/current/fileIO/ReadWrite$WriteStringThread.class\n",
            "bbmap/current/fileIO/ReadWrite.class\n",
            "bbmap/current/fileIO/ReadWrite.java\n",
            "bbmap/current/fileIO/SummaryFile$1.class\n",
            "bbmap/current/fileIO/SummaryFile.class\n",
            "bbmap/current/fileIO/SummaryFile.java\n",
            "bbmap/current/fileIO/TextFile.class\n",
            "bbmap/current/fileIO/TextFile.java\n",
            "bbmap/current/fileIO/TextStreamWriter.class\n",
            "bbmap/current/fileIO/TextStreamWriter.java\n",
            "bbmap/current/fun/\n",
            "bbmap/current/fun/Calc.class\n",
            "bbmap/current/fun/Calc.java\n",
            "bbmap/current/fun/Chance.class\n",
            "bbmap/current/fun/Chance.java\n",
            "bbmap/current/fun/Crunch\n",
            "bbmap/current/fun/Crunch.class\n",
            "bbmap/current/fun/Crunch.java\n",
            "bbmap/current/fun/DiskBench$ReadThread.class\n",
            "bbmap/current/fun/DiskBench$WriteThread.class\n",
            "bbmap/current/fun/DiskBench.class\n",
            "bbmap/current/fun/DiskBench.java\n",
            "bbmap/current/fun/FindPath$Edge.class\n",
            "bbmap/current/fun/FindPath$Node.class\n",
            "bbmap/current/fun/FindPath$Path.class\n",
            "bbmap/current/fun/FindPath.class\n",
            "bbmap/current/fun/FindPath.java\n",
            "bbmap/current/fun/Foo.class\n",
            "bbmap/current/fun/Foo.java\n",
            "bbmap/current/fun/Foo2.class\n",
            "bbmap/current/fun/Foo2.java\n",
            "bbmap/current/fun/Foo3$LongList.class\n",
            "bbmap/current/fun/Foo3$Pair.class\n",
            "bbmap/current/fun/Foo3.class\n",
            "bbmap/current/fun/Foo3.java\n",
            "bbmap/current/fun/Foo4$Pair.class\n",
            "bbmap/current/fun/Foo4.class\n",
            "bbmap/current/fun/Foo4.java\n",
            "bbmap/current/fun/Foo5$LongList.class\n",
            "bbmap/current/fun/Foo5.class\n",
            "bbmap/current/fun/Foo5.java\n",
            "bbmap/current/fun/Foo6.class\n",
            "bbmap/current/fun/Foo6.java\n",
            "bbmap/current/fun/Foo7.class\n",
            "bbmap/current/fun/Foo7.java\n",
            "bbmap/current/fun/Genetic$Bug.class\n",
            "bbmap/current/fun/Genetic.class\n",
            "bbmap/current/fun/Genetic.java\n",
            "bbmap/current/fun/Life.class\n",
            "bbmap/current/fun/Life.java\n",
            "bbmap/current/fun/MakeAdjacencyList$1.class\n",
            "bbmap/current/fun/MakeAdjacencyList.class\n",
            "bbmap/current/fun/MakeAdjacencyList.java\n",
            "bbmap/current/fun/Palindrome.class\n",
            "bbmap/current/fun/Palindrome.java\n",
            "bbmap/current/fun/ProbShared.class\n",
            "bbmap/current/fun/ProbShared.java\n",
            "bbmap/current/fun/ProbShared2.class\n",
            "bbmap/current/fun/ProbShared2.java\n",
            "bbmap/current/fun/ProbShared3.class\n",
            "bbmap/current/fun/ProbShared3.java\n",
            "bbmap/current/gff/\n",
            "bbmap/current/gff/CompareGff.class\n",
            "bbmap/current/gff/CompareGff.java\n",
            "bbmap/current/gff/CompareGff_old.class\n",
            "bbmap/current/gff/CompareGff_old.java\n",
            "bbmap/current/gff/CutGff$ProcessThread.class\n",
            "bbmap/current/gff/CutGff.class\n",
            "bbmap/current/gff/CutGff.java\n",
            "bbmap/current/gff/CutGff_ST.class\n",
            "bbmap/current/gff/CutGff_ST.java\n",
            "bbmap/current/gff/GbffFeature.class\n",
            "bbmap/current/gff/GbffFeature.java\n",
            "bbmap/current/gff/GbffFile.class\n",
            "bbmap/current/gff/GbffFile.java\n",
            "bbmap/current/gff/GbffLocus.class\n",
            "bbmap/current/gff/GbffLocus.java\n",
            "bbmap/current/gff/GffLine.class\n",
            "bbmap/current/gff/GffLine.java\n",
            "bbmap/current/gff/VcfToGff$1.class\n",
            "bbmap/current/gff/VcfToGff.class\n",
            "bbmap/current/gff/VcfToGff.java\n",
            "bbmap/current/hiseq/\n",
            "bbmap/current/hiseq/AnalyzeFlowCell.class\n",
            "bbmap/current/hiseq/AnalyzeFlowCell.java\n",
            "bbmap/current/hiseq/CycleTracker.class\n",
            "bbmap/current/hiseq/CycleTracker.java\n",
            "bbmap/current/hiseq/FlowCell.class\n",
            "bbmap/current/hiseq/FlowCell.java\n",
            "bbmap/current/hiseq/FlowcellCoordinate.class\n",
            "bbmap/current/hiseq/FlowcellCoordinate.java\n",
            "bbmap/current/hiseq/IlluminaHeaderParser.class\n",
            "bbmap/current/hiseq/IlluminaHeaderParser.java\n",
            "bbmap/current/hiseq/Lane.class\n",
            "bbmap/current/hiseq/Lane.java\n",
            "bbmap/current/hiseq/MicroTile.class\n",
            "bbmap/current/hiseq/MicroTile.java\n",
            "bbmap/current/hiseq/PlotFlowCell.class\n",
            "bbmap/current/hiseq/PlotFlowCell.java\n",
            "bbmap/current/hiseq/Tile.class\n",
            "bbmap/current/hiseq/Tile.java\n",
            "bbmap/current/hmm/\n",
            "bbmap/current/hmm/HMMSearchLine.class\n",
            "bbmap/current/hmm/HMMSearchLine.java\n",
            "bbmap/current/hmm/HMMSearchReport.class\n",
            "bbmap/current/hmm/HMMSearchReport.java\n",
            "bbmap/current/hmm/ProteinSummary.class\n",
            "bbmap/current/hmm/ProteinSummary.java\n",
            "bbmap/current/icecream/\n",
            "bbmap/current/icecream/IceCreamAligner.class\n",
            "bbmap/current/icecream/IceCreamAligner.java\n",
            "bbmap/current/icecream/IceCreamAlignerJava.class\n",
            "bbmap/current/icecream/IceCreamAlignerJava.java\n",
            "bbmap/current/icecream/IceCreamAlignerJNI.class\n",
            "bbmap/current/icecream/IceCreamAlignerJNI.java\n",
            "bbmap/current/icecream/IceCreamFinder$ProcessThread.class\n",
            "bbmap/current/icecream/IceCreamFinder.class\n",
            "bbmap/current/icecream/IceCreamFinder.java\n",
            "bbmap/current/icecream/IceCreamGrader.class\n",
            "bbmap/current/icecream/IceCreamGrader.java\n",
            "bbmap/current/icecream/IceCreamMaker$ProcessThread.class\n",
            "bbmap/current/icecream/IceCreamMaker.class\n",
            "bbmap/current/icecream/IceCreamMaker.java\n",
            "bbmap/current/icecream/PBHeader.class\n",
            "bbmap/current/icecream/PBHeader.java\n",
            "bbmap/current/icecream/PolymerTrimmer.class\n",
            "bbmap/current/icecream/PolymerTrimmer.java\n",
            "bbmap/current/icecream/ReadBuilder.class\n",
            "bbmap/current/icecream/ReadBuilder.java\n",
            "bbmap/current/icecream/ReformatPacBio$ProcessThread.class\n",
            "bbmap/current/icecream/ReformatPacBio.class\n",
            "bbmap/current/icecream/ReformatPacBio.java\n",
            "bbmap/current/icecream/ZMW.class\n",
            "bbmap/current/icecream/ZMW.java\n",
            "bbmap/current/icecream/ZMWStreamer.class\n",
            "bbmap/current/icecream/ZMWStreamer.java\n",
            "bbmap/current/jasper/\n",
            "bbmap/current/jasper/Comparison.class\n",
            "bbmap/current/jasper/Comparison.java\n",
            "bbmap/current/jasper/DenseSimilarityMatrix.class\n",
            "bbmap/current/jasper/DenseSimilarityMatrix.java\n",
            "bbmap/current/jasper/DenseTree.class\n",
            "bbmap/current/jasper/DenseTree.java\n",
            "bbmap/current/jasper/DenseTreeValidate.class\n",
            "bbmap/current/jasper/DenseTreeValidate.java\n",
            "bbmap/current/jasper/KmerPosition.class\n",
            "bbmap/current/jasper/KmerPosition.java\n",
            "bbmap/current/jasper/KmerPosition3.class\n",
            "bbmap/current/jasper/KmerPosition3.java\n",
            "bbmap/current/jasper/NCBIComparison.class\n",
            "bbmap/current/jasper/NCBIComparison.java\n",
            "bbmap/current/jasper/NCBISparseSimilarityMatrix.class\n",
            "bbmap/current/jasper/NCBISparseSimilarityMatrix.java\n",
            "bbmap/current/jasper/NCBISparseTree.class\n",
            "bbmap/current/jasper/NCBISparseTree.java\n",
            "bbmap/current/jasper/NCBISparseTreeValidate.class\n",
            "bbmap/current/jasper/NCBISparseTreeValidate.java\n",
            "bbmap/current/jasper/NCBITreeNode.class\n",
            "bbmap/current/jasper/NCBITreeNode.java\n",
            "bbmap/current/jasper/Organism.class\n",
            "bbmap/current/jasper/Organism.java\n",
            "bbmap/current/jasper/SimilarityMatrix.class\n",
            "bbmap/current/jasper/SimilarityMatrix.java\n",
            "bbmap/current/jasper/SparseSimilarityMatrix.class\n",
            "bbmap/current/jasper/SparseSimilarityMatrix.java\n",
            "bbmap/current/jasper/SparseTree.class\n",
            "bbmap/current/jasper/SparseTree.java\n",
            "bbmap/current/jasper/SparseTreeValidate.class\n",
            "bbmap/current/jasper/SparseTreeValidate.java\n",
            "bbmap/current/jasper/TaxCompare.class\n",
            "bbmap/current/jasper/TaxCompare.java\n",
            "bbmap/current/jasper/TreeNode.class\n",
            "bbmap/current/jasper/TreeNode.java\n",
            "bbmap/current/jgi/\n",
            "bbmap/current/jgi/AddAdapters.class\n",
            "bbmap/current/jgi/AddAdapters.java\n",
            "bbmap/current/jgi/AdjustHomopolymers.class\n",
            "bbmap/current/jgi/AdjustHomopolymers.java\n",
            "bbmap/current/jgi/AssemblyStats2$Triple.class\n",
            "bbmap/current/jgi/AssemblyStats2.class\n",
            "bbmap/current/jgi/AssemblyStats2.java\n",
            "bbmap/current/jgi/AssemblyStatsWrapper.class\n",
            "bbmap/current/jgi/AssemblyStatsWrapper.java\n",
            "bbmap/current/jgi/BBDuk$LoadThread.class\n",
            "bbmap/current/jgi/BBDuk$ProcessThread.class\n",
            "bbmap/current/jgi/BBDuk.class\n",
            "bbmap/current/jgi/BBDuk.java\n",
            "bbmap/current/jgi/BBDuk2$LoadThread.class\n",
            "bbmap/current/jgi/BBDuk2$ProcessThread.class\n",
            "bbmap/current/jgi/BBDuk2.class\n",
            "bbmap/current/jgi/BBDuk2.java\n",
            "bbmap/current/jgi/BBMask$MaskLowEntropyThread.class\n",
            "bbmap/current/jgi/BBMask$MaskRepeatThread.class\n",
            "bbmap/current/jgi/BBMask$MaskSamThread.class\n",
            "bbmap/current/jgi/BBMask.class\n",
            "bbmap/current/jgi/BBMask.java\n",
            "bbmap/current/jgi/BBMerge$MateThread.class\n",
            "bbmap/current/jgi/BBMerge.class\n",
            "bbmap/current/jgi/BBMerge.java\n",
            "bbmap/current/jgi/BBMergeOverlapper.class\n",
            "bbmap/current/jgi/BBMergeOverlapper.java\n",
            "bbmap/current/jgi/BBQC.class\n",
            "bbmap/current/jgi/BBQC.java\n",
            "bbmap/current/jgi/CalcTrueQuality$CountMatrixSet.class\n",
            "bbmap/current/jgi/CalcTrueQuality$GBMatrixSet.class\n",
            "bbmap/current/jgi/CalcTrueQuality$Worker.class\n",
            "bbmap/current/jgi/CalcTrueQuality.class\n",
            "bbmap/current/jgi/CalcTrueQuality.java\n",
            "bbmap/current/jgi/CalcUniqueness$Counter.class\n",
            "bbmap/current/jgi/CalcUniqueness.class\n",
            "bbmap/current/jgi/CalcUniqueness.java\n",
            "bbmap/current/jgi/CallPeaks$Peak.class\n",
            "bbmap/current/jgi/CallPeaks.class\n",
            "bbmap/current/jgi/CallPeaks.java\n",
            "bbmap/current/jgi/CheckStrand.class\n",
            "bbmap/current/jgi/CheckStrand.java\n",
            "bbmap/current/jgi/CheckStrand2$ProcessThread.class\n",
            "bbmap/current/jgi/CheckStrand2.class\n",
            "bbmap/current/jgi/CheckStrand2.java\n",
            "bbmap/current/jgi/Consect.class\n",
            "bbmap/current/jgi/Consect.java\n",
            "bbmap/current/jgi/CorrelateBarcodes.class\n",
            "bbmap/current/jgi/CorrelateBarcodes.java\n",
            "bbmap/current/jgi/CountBarcodes.class\n",
            "bbmap/current/jgi/CountBarcodes.java\n",
            "bbmap/current/jgi/CountBarcodes2.class\n",
            "bbmap/current/jgi/CountBarcodes2.java\n",
            "bbmap/current/jgi/CountDuplicates$ProcessThread.class\n",
            "bbmap/current/jgi/CountDuplicates.class\n",
            "bbmap/current/jgi/CountDuplicates.java\n",
            "bbmap/current/jgi/CountDuplicatesBuffered$ProcessThread.class\n",
            "bbmap/current/jgi/CountDuplicatesBuffered.class\n",
            "bbmap/current/jgi/CountDuplicatesBuffered.java\n",
            "bbmap/current/jgi/CountGC$1.class\n",
            "bbmap/current/jgi/CountGC.class\n",
            "bbmap/current/jgi/CountGC.java\n",
            "bbmap/current/jgi/CountUniqueness.class\n",
            "bbmap/current/jgi/CountUniqueness.java\n",
            "bbmap/current/jgi/CoveragePileup.class\n",
            "bbmap/current/jgi/CoveragePileup.java\n",
            "bbmap/current/jgi/CovStatsLine.class\n",
            "bbmap/current/jgi/CovStatsLine.java\n",
            "bbmap/current/jgi/CrisprFinder$1.class\n",
            "bbmap/current/jgi/CrisprFinder$Alignment.class\n",
            "bbmap/current/jgi/CrisprFinder$ProcessThread.class\n",
            "bbmap/current/jgi/CrisprFinder$SeqCount.class\n",
            "bbmap/current/jgi/CrisprFinder.class\n",
            "bbmap/current/jgi/CrisprFinder.java\n",
            "bbmap/current/jgi/CrisprFinder2$ProcessThread.class\n",
            "bbmap/current/jgi/CrisprFinder2.class\n",
            "bbmap/current/jgi/CrisprFinder2.java\n",
            "bbmap/current/jgi/CrisprFinder3$1.class\n",
            "bbmap/current/jgi/CrisprFinder3$Alignment.class\n",
            "bbmap/current/jgi/CrisprFinder3$ProcessThread.class\n",
            "bbmap/current/jgi/CrisprFinder3.class\n",
            "bbmap/current/jgi/CrisprFinder3.java\n",
            "bbmap/current/jgi/CrisprFinder4$1.class\n",
            "bbmap/current/jgi/CrisprFinder4$Alignment.class\n",
            "bbmap/current/jgi/CrisprFinder4$ProcessThread.class\n",
            "bbmap/current/jgi/CrisprFinder4.class\n",
            "bbmap/current/jgi/CrisprFinder4.java\n",
            "bbmap/current/jgi/CrisprFinder_brkn$ProcessThread.class\n",
            "bbmap/current/jgi/CrisprFinder_brkn$SeqCount.class\n",
            "bbmap/current/jgi/CrisprFinder_brkn.class\n",
            "bbmap/current/jgi/CrisprFinder_S$ProcessThread.class\n",
            "bbmap/current/jgi/CrisprFinder_S$SeqCount.class\n",
            "bbmap/current/jgi/CrisprFinder_S.class\n",
            "bbmap/current/jgi/CrossContaminate$Vessel.class\n",
            "bbmap/current/jgi/CrossContaminate.class\n",
            "bbmap/current/jgi/CrossContaminate.java\n",
            "bbmap/current/jgi/CutPrimers.class\n",
            "bbmap/current/jgi/CutPrimers.java\n",
            "bbmap/current/jgi/DecontaminateByNormalization.class\n",
            "bbmap/current/jgi/DecontaminateByNormalization.java\n",
            "bbmap/current/jgi/Dedupe$ClusterLengthComparator.class\n",
            "bbmap/current/jgi/Dedupe$ClusterThread.class\n",
            "bbmap/current/jgi/Dedupe$HashThread.class\n",
            "bbmap/current/jgi/Dedupe$MstThread.class\n",
            "bbmap/current/jgi/Dedupe$Overlap.class\n",
            "bbmap/current/jgi/Dedupe$Unit.class\n",
            "bbmap/current/jgi/Dedupe$UnitOffsetComparator.class\n",
            "bbmap/current/jgi/Dedupe.class\n",
            "bbmap/current/jgi/Dedupe.java\n",
            "bbmap/current/jgi/Dedupe2$ClusterLengthComparator.class\n",
            "bbmap/current/jgi/Dedupe2$ClusterThread.class\n",
            "bbmap/current/jgi/Dedupe2$HashThread.class\n",
            "bbmap/current/jgi/Dedupe2$MstThread.class\n",
            "bbmap/current/jgi/Dedupe2$Overlap.class\n",
            "bbmap/current/jgi/Dedupe2$Unit.class\n",
            "bbmap/current/jgi/Dedupe2$UnitOffsetComparator.class\n",
            "bbmap/current/jgi/Dedupe2.class\n",
            "bbmap/current/jgi/Dedupe2.java\n",
            "bbmap/current/jgi/DedupeByMapping$Quad.class\n",
            "bbmap/current/jgi/DedupeByMapping.class\n",
            "bbmap/current/jgi/DedupeByMapping.java\n",
            "bbmap/current/jgi/DedupeProtein$ClusterLengthComparator.class\n",
            "bbmap/current/jgi/DedupeProtein$ClusterThread.class\n",
            "bbmap/current/jgi/DedupeProtein$HashThread.class\n",
            "bbmap/current/jgi/DedupeProtein$MstThread.class\n",
            "bbmap/current/jgi/DedupeProtein$Overlap.class\n",
            "bbmap/current/jgi/DedupeProtein$Unit.class\n",
            "bbmap/current/jgi/DedupeProtein$UnitOffsetComparator.class\n",
            "bbmap/current/jgi/DedupeProtein.class\n",
            "bbmap/current/jgi/DedupeProtein.java\n",
            "bbmap/current/jgi/DemuxByName.class\n",
            "bbmap/current/jgi/DemuxByName.java\n",
            "bbmap/current/jgi/DemuxByName2.class\n",
            "bbmap/current/jgi/DemuxByName2.java\n",
            "bbmap/current/jgi/Difference.class\n",
            "bbmap/current/jgi/Difference.java\n",
            "bbmap/current/jgi/FakeReads.class\n",
            "bbmap/current/jgi/FakeReads.java\n",
            "bbmap/current/jgi/FilterByCoverage.class\n",
            "bbmap/current/jgi/FilterByCoverage.java\n",
            "bbmap/current/jgi/FilterBySequence$Code.class\n",
            "bbmap/current/jgi/FilterBySequence$LoadThread.class\n",
            "bbmap/current/jgi/FilterBySequence$ProcessThread.class\n",
            "bbmap/current/jgi/FilterBySequence.class\n",
            "bbmap/current/jgi/FilterBySequence.java\n",
            "bbmap/current/jgi/FilterReadsWithSubs.class\n",
            "bbmap/current/jgi/FilterReadsWithSubs.java\n",
            "bbmap/current/jgi/FindHiCJunctions.class\n",
            "bbmap/current/jgi/FindHiCJunctions.java\n",
            "bbmap/current/jgi/FindPrimers$ProcessThread.class\n",
            "bbmap/current/jgi/FindPrimers.class\n",
            "bbmap/current/jgi/FindPrimers.java\n",
            "bbmap/current/jgi/FindPrimers_old.class\n",
            "bbmap/current/jgi/FindPrimers_old.java\n",
            "bbmap/current/jgi/FindString.class\n",
            "bbmap/current/jgi/FindString.java\n",
            "bbmap/current/jgi/FungalRelease.class\n",
            "bbmap/current/jgi/FungalRelease.java\n",
            "bbmap/current/jgi/FuseSequence.class\n",
            "bbmap/current/jgi/FuseSequence.java\n",
            "bbmap/current/jgi/GatherKapaStats$KapaEntry.class\n",
            "bbmap/current/jgi/GatherKapaStats$Plate.class\n",
            "bbmap/current/jgi/GatherKapaStats$TagData.class\n",
            "bbmap/current/jgi/GatherKapaStats$Well.class\n",
            "bbmap/current/jgi/GatherKapaStats.class\n",
            "bbmap/current/jgi/GatherKapaStats.java\n",
            "bbmap/current/jgi/GetReads.class\n",
            "bbmap/current/jgi/GetReads.java\n",
            "bbmap/current/jgi/GradeMergedReads.class\n",
            "bbmap/current/jgi/GradeMergedReads.java\n",
            "bbmap/current/jgi/GreedyBarCodeFinder.class\n",
            "bbmap/current/jgi/GreedyBarCodeFinder.java\n",
            "bbmap/current/jgi/IdentityMatrix$ProcessThread.class\n",
            "bbmap/current/jgi/IdentityMatrix.class\n",
            "bbmap/current/jgi/IdentityMatrix.java\n",
            "bbmap/current/jgi/Info.class\n",
            "bbmap/current/jgi/Info.java\n",
            "bbmap/current/jgi/KeepBestCopy.class\n",
            "bbmap/current/jgi/KeepBestCopy.java\n",
            "bbmap/current/jgi/KExpand.class\n",
            "bbmap/current/jgi/KExpand.java\n",
            "bbmap/current/jgi/KmerCountExact$DumpKmersThread.class\n",
            "bbmap/current/jgi/KmerCountExact$MakeKhistThread.class\n",
            "bbmap/current/jgi/KmerCountExact.class\n",
            "bbmap/current/jgi/KmerCountExact.java\n",
            "bbmap/current/jgi/KmerCountMulti$ProcessThread.class\n",
            "bbmap/current/jgi/KmerCountMulti.class\n",
            "bbmap/current/jgi/KmerCountMulti.java\n",
            "bbmap/current/jgi/KmerCoverage$1.class\n",
            "bbmap/current/jgi/KmerCoverage$ProcessThread.class\n",
            "bbmap/current/jgi/KmerCoverage.class\n",
            "bbmap/current/jgi/KmerCoverage.java\n",
            "bbmap/current/jgi/KmerFilterSetMaker.class\n",
            "bbmap/current/jgi/KmerFilterSetMaker.java\n",
            "bbmap/current/jgi/KmerNormalize$1.class\n",
            "bbmap/current/jgi/KmerNormalize$ProcessThread.class\n",
            "bbmap/current/jgi/KmerNormalize.class\n",
            "bbmap/current/jgi/KmerNormalize.java\n",
            "bbmap/current/jgi/KmerSample.class\n",
            "bbmap/current/jgi/KmerSample.java\n",
            "bbmap/current/jgi/MakeChimeras.class\n",
            "bbmap/current/jgi/MakeChimeras.java\n",
            "bbmap/current/jgi/MakeContaminatedGenomes.class\n",
            "bbmap/current/jgi/MakeContaminatedGenomes.java\n",
            "bbmap/current/jgi/MakeLengthHistogram$1.class\n",
            "bbmap/current/jgi/MakeLengthHistogram.class\n",
            "bbmap/current/jgi/MakeLengthHistogram.java\n",
            "bbmap/current/jgi/MakePolymers.class\n",
            "bbmap/current/jgi/MakePolymers.java\n",
            "bbmap/current/jgi/MergeBarcodes.class\n",
            "bbmap/current/jgi/MergeBarcodes.java\n",
            "bbmap/current/jgi/MergeSam.class\n",
            "bbmap/current/jgi/MergeSam.java\n",
            "bbmap/current/jgi/MutateGenome$SmallVar.class\n",
            "bbmap/current/jgi/MutateGenome.class\n",
            "bbmap/current/jgi/MutateGenome.java\n",
            "bbmap/current/jgi/MutateGenome2$SmallVar.class\n",
            "bbmap/current/jgi/MutateGenome2.class\n",
            "bbmap/current/jgi/MutateGenome2.java\n",
            "bbmap/current/jgi/Orf.class\n",
            "bbmap/current/jgi/Orf.java\n",
            "bbmap/current/jgi/PartitionReads$Partition.class\n",
            "bbmap/current/jgi/PartitionReads.class\n",
            "bbmap/current/jgi/PartitionReads.java\n",
            "bbmap/current/jgi/PhylipToFasta.class\n",
            "bbmap/current/jgi/PhylipToFasta.java\n",
            "bbmap/current/jgi/RandomGenome.class\n",
            "bbmap/current/jgi/RandomGenome.java\n",
            "bbmap/current/jgi/ReadKmerDepthDistribution$1.class\n",
            "bbmap/current/jgi/ReadKmerDepthDistribution$ProcessThread.class\n",
            "bbmap/current/jgi/ReadKmerDepthDistribution.class\n",
            "bbmap/current/jgi/ReadKmerDepthDistribution.java\n",
            "bbmap/current/jgi/RedirectTest.class\n",
            "bbmap/current/jgi/RedirectTest.java\n",
            "bbmap/current/jgi/ReformatReads.class\n",
            "bbmap/current/jgi/ReformatReads.java\n",
            "bbmap/current/jgi/RemapQuality.class\n",
            "bbmap/current/jgi/RemapQuality.java\n",
            "bbmap/current/jgi/RemoveBadBarcodes.class\n",
            "bbmap/current/jgi/RemoveBadBarcodes.java\n",
            "bbmap/current/jgi/RenameReads.class\n",
            "bbmap/current/jgi/RenameReads.java\n",
            "bbmap/current/jgi/Repeat$PosComparator.class\n",
            "bbmap/current/jgi/Repeat$PosComparator2.class\n",
            "bbmap/current/jgi/Repeat.class\n",
            "bbmap/current/jgi/Repeat.java\n",
            "bbmap/current/jgi/RepeatFinder$ProcessThread.class\n",
            "bbmap/current/jgi/RepeatFinder.class\n",
            "bbmap/current/jgi/RepeatFinder.java\n",
            "bbmap/current/jgi/ReplaceHeaders.class\n",
            "bbmap/current/jgi/ReplaceHeaders.java\n",
            "bbmap/current/jgi/RepresentativeSet$Edge.class\n",
            "bbmap/current/jgi/RepresentativeSet$Node.class\n",
            "bbmap/current/jgi/RepresentativeSet.class\n",
            "bbmap/current/jgi/RepresentativeSet.java\n",
            "bbmap/current/jgi/RQCFilter2.class\n",
            "bbmap/current/jgi/RQCFilter2.java\n",
            "bbmap/current/jgi/RQCFilterStats.class\n",
            "bbmap/current/jgi/RQCFilterStats.java\n",
            "bbmap/current/jgi/SamToEst$1.class\n",
            "bbmap/current/jgi/SamToEst$EST.class\n",
            "bbmap/current/jgi/SamToEst.class\n",
            "bbmap/current/jgi/SamToEst.java\n",
            "bbmap/current/jgi/Seal$LoadThread.class\n",
            "bbmap/current/jgi/Seal$ProcessThread.class\n",
            "bbmap/current/jgi/Seal.class\n",
            "bbmap/current/jgi/Seal.java\n",
            "bbmap/current/jgi/SealRefInfo.class\n",
            "bbmap/current/jgi/SealRefInfo.java\n",
            "bbmap/current/jgi/Shred.class\n",
            "bbmap/current/jgi/Shred.java\n",
            "bbmap/current/jgi/SmallKmerFrequency$CountComparator.class\n",
            "bbmap/current/jgi/SmallKmerFrequency$Kmer.class\n",
            "bbmap/current/jgi/SmallKmerFrequency$NumComparator.class\n",
            "bbmap/current/jgi/SmallKmerFrequency.class\n",
            "bbmap/current/jgi/SmallKmerFrequency.java\n",
            "bbmap/current/jgi/SplitNexteraLMP.class\n",
            "bbmap/current/jgi/SplitNexteraLMP.java\n",
            "bbmap/current/jgi/SplitPairsAndSingles.class\n",
            "bbmap/current/jgi/SplitPairsAndSingles.java\n",
            "bbmap/current/jgi/SplitSam4Way.class\n",
            "bbmap/current/jgi/SplitSam4Way.java\n",
            "bbmap/current/jgi/SplitSam6Way.class\n",
            "bbmap/current/jgi/SplitSam6Way.java\n",
            "bbmap/current/jgi/SplitSamFile.class\n",
            "bbmap/current/jgi/SplitSamFile.java\n",
            "bbmap/current/jgi/StringCount.class\n",
            "bbmap/current/jgi/StringCount.java\n",
            "bbmap/current/jgi/SynthMDA.class\n",
            "bbmap/current/jgi/SynthMDA.java\n",
            "bbmap/current/jgi/TestFilesystem$WriteThread.class\n",
            "bbmap/current/jgi/TestFilesystem.class\n",
            "bbmap/current/jgi/TestFilesystem.java\n",
            "bbmap/current/jgi/TestFormat$Barcode.class\n",
            "bbmap/current/jgi/TestFormat$TestThread.class\n",
            "bbmap/current/jgi/TestFormat.class\n",
            "bbmap/current/jgi/TestFormat.java\n",
            "bbmap/current/jgi/TetramerFrequencies$Line.class\n",
            "bbmap/current/jgi/TetramerFrequencies$PrintThread.class\n",
            "bbmap/current/jgi/TetramerFrequencies.class\n",
            "bbmap/current/jgi/TetramerFrequencies.java\n",
            "bbmap/current/jgi/TranslateSixFrames.class\n",
            "bbmap/current/jgi/TranslateSixFrames.java\n",
            "bbmap/current/jgi/UnicodeToAscii.class\n",
            "bbmap/current/jgi/UnicodeToAscii.java\n",
            "bbmap/current/jgi/Unzip.class\n",
            "bbmap/current/jgi/Unzip.java\n",
            "bbmap/current/json/\n",
            "bbmap/current/json/JsonLiteral.class\n",
            "bbmap/current/json/JsonLiteral.java\n",
            "bbmap/current/json/JsonObject.class\n",
            "bbmap/current/json/JsonObject.java\n",
            "bbmap/current/json/JsonParser.class\n",
            "bbmap/current/json/JsonParser.java\n",
            "bbmap/current/kmer/\n",
            "bbmap/current/kmer/AbstractKmerTable$AllocThread.class\n",
            "bbmap/current/kmer/AbstractKmerTable.class\n",
            "bbmap/current/kmer/AbstractKmerTable.java\n",
            "bbmap/current/kmer/AbstractKmerTableSet.class\n",
            "bbmap/current/kmer/AbstractKmerTableSet.java\n",
            "bbmap/current/kmer/DumpThread.class\n",
            "bbmap/current/kmer/DumpThread.java\n",
            "bbmap/current/kmer/HashArray.class\n",
            "bbmap/current/kmer/HashArray.java\n",
            "bbmap/current/kmer/HashArray1D$Walker1D.class\n",
            "bbmap/current/kmer/HashArray1D.class\n",
            "bbmap/current/kmer/HashArray1D.java\n",
            "bbmap/current/kmer/HashArray2D.class\n",
            "bbmap/current/kmer/HashArray2D.java\n",
            "bbmap/current/kmer/HashArrayHybrid.class\n",
            "bbmap/current/kmer/HashArrayHybrid.java\n",
            "bbmap/current/kmer/HashArrayHybridFast.class\n",
            "bbmap/current/kmer/HashArrayHybridFast.java\n",
            "bbmap/current/kmer/HashBuffer.class\n",
            "bbmap/current/kmer/HashBuffer.java\n",
            "bbmap/current/kmer/HashForest.class\n",
            "bbmap/current/kmer/HashForest.java\n",
            "bbmap/current/kmer/HistogramMaker$FillThread.class\n",
            "bbmap/current/kmer/HistogramMaker.class\n",
            "bbmap/current/kmer/HistogramMaker.java\n",
            "bbmap/current/kmer/KmerBuffer.class\n",
            "bbmap/current/kmer/KmerBuffer.java\n",
            "bbmap/current/kmer/KmerLink.class\n",
            "bbmap/current/kmer/KmerLink.java\n",
            "bbmap/current/kmer/KmerNode.class\n",
            "bbmap/current/kmer/KmerNode.java\n",
            "bbmap/current/kmer/KmerNode1D.class\n",
            "bbmap/current/kmer/KmerNode1D.java\n",
            "bbmap/current/kmer/KmerNode2D.class\n",
            "bbmap/current/kmer/KmerNode2D.java\n",
            "bbmap/current/kmer/KmerTable.class\n",
            "bbmap/current/kmer/KmerTable.java\n",
            "bbmap/current/kmer/KmerTableSet$LoadThread.class\n",
            "bbmap/current/kmer/KmerTableSet$WalkerKST.class\n",
            "bbmap/current/kmer/KmerTableSet.class\n",
            "bbmap/current/kmer/KmerTableSet.java\n",
            "bbmap/current/kmer/OwnershipThread.class\n",
            "bbmap/current/kmer/OwnershipThread.java\n",
            "bbmap/current/kmer/ScheduleMaker.class\n",
            "bbmap/current/kmer/ScheduleMaker.java\n",
            "bbmap/current/kmer/SimpleKmerTable.class\n",
            "bbmap/current/kmer/SimpleKmerTable.java\n",
            "bbmap/current/kmer/TableLoaderLockFree$LoadThread.class\n",
            "bbmap/current/kmer/TableLoaderLockFree.class\n",
            "bbmap/current/kmer/TableLoaderLockFree.java\n",
            "bbmap/current/kmer/TableReader.class\n",
            "bbmap/current/kmer/TableReader.java\n",
            "bbmap/current/kmer/Walker.class\n",
            "bbmap/current/kmer/Walker.java\n",
            "bbmap/current/ml/\n",
            "bbmap/current/ml/Bell.class\n",
            "bbmap/current/ml/Bell.java\n",
            "bbmap/current/ml/Cell.class\n",
            "bbmap/current/ml/Cell.java\n",
            "bbmap/current/ml/CellDouble.class\n",
            "bbmap/current/ml/CellDouble.java\n",
            "bbmap/current/ml/CellNet.class\n",
            "bbmap/current/ml/CellNet.java\n",
            "bbmap/current/ml/CellNetDouble.class\n",
            "bbmap/current/ml/CellNetDouble.java\n",
            "bbmap/current/ml/CellNetParser.class\n",
            "bbmap/current/ml/CellNetParser.java\n",
            "bbmap/current/ml/CellNetReverseComparator.class\n",
            "bbmap/current/ml/CellNetReverseComparator.java\n",
            "bbmap/current/ml/DataLoader$Pair.class\n",
            "bbmap/current/ml/DataLoader.class\n",
            "bbmap/current/ml/DataLoader.java\n",
            "bbmap/current/ml/Edge.class\n",
            "bbmap/current/ml/Edge.java\n",
            "bbmap/current/ml/ExtendedMSig.class\n",
            "bbmap/current/ml/ExtendedMSig.java\n",
            "bbmap/current/ml/ExtendedSigmoid.class\n",
            "bbmap/current/ml/ExtendedSigmoid.java\n",
            "bbmap/current/ml/Function.class\n",
            "bbmap/current/ml/Function.java\n",
            "bbmap/current/ml/Functions.class\n",
            "bbmap/current/ml/Functions.java\n",
            "bbmap/current/ml/JobData.class\n",
            "bbmap/current/ml/JobData.java\n",
            "bbmap/current/ml/JobResults.class\n",
            "bbmap/current/ml/JobResults.java\n",
            "bbmap/current/ml/Matrix.class\n",
            "bbmap/current/ml/Matrix.java\n",
            "bbmap/current/ml/MSig.class\n",
            "bbmap/current/ml/MSig.java\n",
            "bbmap/current/ml/ProcessBBMergeHeaders$Header.class\n",
            "bbmap/current/ml/ProcessBBMergeHeaders.class\n",
            "bbmap/current/ml/ProcessBBMergeHeaders.java\n",
            "bbmap/current/ml/Profiler.class\n",
            "bbmap/current/ml/Profiler.java\n",
            "bbmap/current/ml/RSLog.class\n",
            "bbmap/current/ml/RSLog.java\n",
            "bbmap/current/ml/Sample.class\n",
            "bbmap/current/ml/Sample.java\n",
            "bbmap/current/ml/SampleErrorComparator.class\n",
            "bbmap/current/ml/SampleErrorComparator.java\n",
            "bbmap/current/ml/SampleSet.class\n",
            "bbmap/current/ml/SampleSet.java\n",
            "bbmap/current/ml/SampleValueComparator.class\n",
            "bbmap/current/ml/SampleValueComparator.java\n",
            "bbmap/current/ml/ScannerThread$LaunchThread.class\n",
            "bbmap/current/ml/ScannerThread.class\n",
            "bbmap/current/ml/ScannerThread.java\n",
            "bbmap/current/ml/ScoreSequence.class\n",
            "bbmap/current/ml/ScoreSequence.java\n",
            "bbmap/current/ml/Seed.class\n",
            "bbmap/current/ml/Seed.java\n",
            "bbmap/current/ml/SequenceToVector$1.class\n",
            "bbmap/current/ml/SequenceToVector.class\n",
            "bbmap/current/ml/SequenceToVector.java\n",
            "bbmap/current/ml/Sigmoid.class\n",
            "bbmap/current/ml/Sigmoid.java\n",
            "bbmap/current/ml/Source.class\n",
            "bbmap/current/ml/Source.java\n",
            "bbmap/current/ml/Status.class\n",
            "bbmap/current/ml/Status.java\n",
            "bbmap/current/ml/Subset.class\n",
            "bbmap/current/ml/Subset.java\n",
            "bbmap/current/ml/Swish.class\n",
            "bbmap/current/ml/Swish.java\n",
            "bbmap/current/ml/Tanh.class\n",
            "bbmap/current/ml/Tanh.java\n",
            "bbmap/current/ml/Trainer.class\n",
            "bbmap/current/ml/Trainer.java\n",
            "bbmap/current/ml/TrainerThread$LaunchThread.class\n",
            "bbmap/current/ml/TrainerThread.class\n",
            "bbmap/current/ml/TrainerThread.java\n",
            "bbmap/current/ml/WorkerThread.class\n",
            "bbmap/current/ml/WorkerThread.java\n",
            "bbmap/current/pacbio/\n",
            "bbmap/current/pacbio/CalcCoverageFromSites$1.class\n",
            "bbmap/current/pacbio/CalcCoverageFromSites.class\n",
            "bbmap/current/pacbio/CalcCoverageFromSites.java\n",
            "bbmap/current/pacbio/GenerateMultiChrom.class\n",
            "bbmap/current/pacbio/GenerateMultiChrom.java\n",
            "bbmap/current/pacbio/MakePacBioScript.class\n",
            "bbmap/current/pacbio/MakePacBioScript.java\n",
            "bbmap/current/pacbio/MergeFastaContigs$1.class\n",
            "bbmap/current/pacbio/MergeFastaContigs.class\n",
            "bbmap/current/pacbio/MergeFastaContigs.java\n",
            "bbmap/current/pacbio/MergeReadsAndGenome$1.class\n",
            "bbmap/current/pacbio/MergeReadsAndGenome.class\n",
            "bbmap/current/pacbio/MergeReadsAndGenome.java\n",
            "bbmap/current/pacbio/PartitionFastaFile.class\n",
            "bbmap/current/pacbio/PartitionFastaFile.java\n",
            "bbmap/current/pacbio/PartitionReads$1.class\n",
            "bbmap/current/pacbio/PartitionReads.class\n",
            "bbmap/current/pacbio/PartitionReads.java\n",
            "bbmap/current/pacbio/ProcessStackedSitesNormalized$1.class\n",
            "bbmap/current/pacbio/ProcessStackedSitesNormalized$Buffer.class\n",
            "bbmap/current/pacbio/ProcessStackedSitesNormalized$Ssra.class\n",
            "bbmap/current/pacbio/ProcessStackedSitesNormalized.class\n",
            "bbmap/current/pacbio/ProcessStackedSitesNormalized.java\n",
            "bbmap/current/pacbio/RemoveAdapters2$1.class\n",
            "bbmap/current/pacbio/RemoveAdapters2$ProcessThread.class\n",
            "bbmap/current/pacbio/RemoveAdapters2.class\n",
            "bbmap/current/pacbio/RemoveAdapters2.java\n",
            "bbmap/current/pacbio/RemoveAdapters3$ProcessThread.class\n",
            "bbmap/current/pacbio/RemoveAdapters3.class\n",
            "bbmap/current/pacbio/RemoveAdapters3.java\n",
            "bbmap/current/pacbio/SiteR.class\n",
            "bbmap/current/pacbio/SiteR.java\n",
            "bbmap/current/pacbio/SortSites$1.class\n",
            "bbmap/current/pacbio/SortSites.class\n",
            "bbmap/current/pacbio/SortSites.java\n",
            "bbmap/current/pacbio/SplitOffPerfectContigs$1.class\n",
            "bbmap/current/pacbio/SplitOffPerfectContigs.class\n",
            "bbmap/current/pacbio/SplitOffPerfectContigs.java\n",
            "bbmap/current/pacbio/StackSites$1.class\n",
            "bbmap/current/pacbio/StackSites$Glob.class\n",
            "bbmap/current/pacbio/StackSites.class\n",
            "bbmap/current/pacbio/StackSites.java\n",
            "bbmap/current/pacbio/StackSites2$1.class\n",
            "bbmap/current/pacbio/StackSites2$Glob.class\n",
            "bbmap/current/pacbio/StackSites2.class\n",
            "bbmap/current/pacbio/StackSites2.java\n",
            "bbmap/current/prok/\n",
            "bbmap/current/prok/AnalyzeGenes$FileThread.class\n",
            "bbmap/current/prok/AnalyzeGenes.class\n",
            "bbmap/current/prok/AnalyzeGenes.java\n",
            "bbmap/current/prok/CallGenes$ProcessThread.class\n",
            "bbmap/current/prok/CallGenes.class\n",
            "bbmap/current/prok/CallGenes.java\n",
            "bbmap/current/prok/Feature$FeatureComparatorScore.class\n",
            "bbmap/current/prok/Feature.class\n",
            "bbmap/current/prok/Feature.java\n",
            "bbmap/current/prok/FetchProks$ProcessThread.class\n",
            "bbmap/current/prok/FetchProks$Stats.class\n",
            "bbmap/current/prok/FetchProks.class\n",
            "bbmap/current/prok/FetchProks.java\n",
            "bbmap/current/prok/FilterSilva.class\n",
            "bbmap/current/prok/FilterSilva.java\n",
            "bbmap/current/prok/FrameStats.class\n",
            "bbmap/current/prok/FrameStats.java\n",
            "bbmap/current/prok/GeneCaller.class\n",
            "bbmap/current/prok/GeneCaller.java\n",
            "bbmap/current/prok/GeneModel.class\n",
            "bbmap/current/prok/GeneModel.java\n",
            "bbmap/current/prok/GeneModelParser.class\n",
            "bbmap/current/prok/GeneModelParser.java\n",
            "bbmap/current/prok/MergeRibo$ListComparator.class\n",
            "bbmap/current/prok/MergeRibo$ProcessThread.class\n",
            "bbmap/current/prok/MergeRibo$Ribo.class\n",
            "bbmap/current/prok/MergeRibo.class\n",
            "bbmap/current/prok/MergeRibo.java\n",
            "bbmap/current/prok/MergeRibo_Fast$ProcessThread.class\n",
            "bbmap/current/prok/MergeRibo_Fast$Ribo.class\n",
            "bbmap/current/prok/MergeRibo_Fast.class\n",
            "bbmap/current/prok/MergeRibo_Fast.java\n",
            "bbmap/current/prok/Orf.class\n",
            "bbmap/current/prok/Orf.java\n",
            "bbmap/current/prok/PFeature$FeatureComparatorScore.class\n",
            "bbmap/current/prok/PFeature.class\n",
            "bbmap/current/prok/PFeature.java\n",
            "bbmap/current/prok/PGMTools$1.class\n",
            "bbmap/current/prok/PGMTools.class\n",
            "bbmap/current/prok/PGMTools.java\n",
            "bbmap/current/prok/ProkObject.class\n",
            "bbmap/current/prok/ProkObject.java\n",
            "bbmap/current/prok/RiboMaker$ProcessThread.class\n",
            "bbmap/current/prok/RiboMaker.class\n",
            "bbmap/current/prok/RiboMaker.java\n",
            "bbmap/current/prok/ScafData.class\n",
            "bbmap/current/prok/ScafData.java\n",
            "bbmap/current/prok/ScoreTracker.class\n",
            "bbmap/current/prok/ScoreTracker.java\n",
            "bbmap/current/prok/SplitRibo$ProcessThread.class\n",
            "bbmap/current/prok/SplitRibo.class\n",
            "bbmap/current/prok/SplitRibo.java\n",
            "bbmap/current/prok/StatsContainer.class\n",
            "bbmap/current/prok/StatsContainer.java\n",
            "bbmap/current/repeat/\n",
            "bbmap/current/repeat/Palindrome.class\n",
            "bbmap/current/repeat/Palindrome.java\n",
            "bbmap/current/repeat/PalindromeFinder.class\n",
            "bbmap/current/repeat/PalindromeFinder.java\n",
            "bbmap/current/repeat/Range.class\n",
            "bbmap/current/repeat/Range.java\n",
            "bbmap/current/repeat/Repeat$PosComparator.class\n",
            "bbmap/current/repeat/Repeat$PosComparator2.class\n",
            "bbmap/current/repeat/Repeat.class\n",
            "bbmap/current/repeat/Repeat.java\n",
            "bbmap/current/repeat/RepeatFinder$ProcessThread.class\n",
            "bbmap/current/repeat/RepeatFinder.class\n",
            "bbmap/current/repeat/RepeatFinder.java\n",
            "bbmap/current/repeat/RepeatSet.class\n",
            "bbmap/current/repeat/RepeatSet.java\n",
            "bbmap/current/server/\n",
            "bbmap/current/server/PercentEncoding.class\n",
            "bbmap/current/server/PercentEncoding.java\n",
            "bbmap/current/server/ServerTools.class\n",
            "bbmap/current/server/ServerTools.java\n",
            "bbmap/current/server/SimpleHttpServer$GetHandler.class\n",
            "bbmap/current/server/SimpleHttpServer.class\n",
            "bbmap/current/server/SimpleHttpServer.java\n",
            "bbmap/current/shared/\n",
            "bbmap/current/shared/Colors.class\n",
            "bbmap/current/shared/Colors.java\n",
            "bbmap/current/shared/KillSwitch.class\n",
            "bbmap/current/shared/KillSwitch.java\n",
            "bbmap/current/shared/LineParser.class\n",
            "bbmap/current/shared/LineParser.java\n",
            "bbmap/current/shared/LineParser1.class\n",
            "bbmap/current/shared/LineParser1.java\n",
            "bbmap/current/shared/LineParser2.class\n",
            "bbmap/current/shared/LineParser2.java\n",
            "bbmap/current/shared/LineParser2S.class\n",
            "bbmap/current/shared/LineParser2S.java\n",
            "bbmap/current/shared/LineParserSimple.class\n",
            "bbmap/current/shared/LineParserSimple.java\n",
            "bbmap/current/shared/MetadataWriter.class\n",
            "bbmap/current/shared/MetadataWriter.java\n",
            "bbmap/current/shared/Parse.class\n",
            "bbmap/current/shared/Parse.java\n",
            "bbmap/current/shared/Parser.class\n",
            "bbmap/current/shared/Parser.java\n",
            "bbmap/current/shared/PreParser.class\n",
            "bbmap/current/shared/PreParser.java\n",
            "bbmap/current/shared/Primes.class\n",
            "bbmap/current/shared/Primes.java\n",
            "bbmap/current/shared/ReadStats.class\n",
            "bbmap/current/shared/ReadStats.java\n",
            "bbmap/current/shared/Shared.class\n",
            "bbmap/current/shared/Shared.java\n",
            "bbmap/current/shared/SIMD.class\n",
            "bbmap/current/shared/SIMD.java\n",
            "bbmap/current/shared/Timer.class\n",
            "bbmap/current/shared/Timer.java\n",
            "bbmap/current/shared/Tools$1.class\n",
            "bbmap/current/shared/Tools.class\n",
            "bbmap/current/shared/Tools.java\n",
            "bbmap/current/shared/TrimRead.class\n",
            "bbmap/current/shared/TrimRead.java\n",
            "bbmap/current/shared/Vector.class\n",
            "bbmap/current/shared/Vector.java\n",
            "bbmap/current/sketch/\n",
            "bbmap/current/sketch/AddSSU$SketchHeader.class\n",
            "bbmap/current/sketch/AddSSU.class\n",
            "bbmap/current/sketch/AddSSU.java\n",
            "bbmap/current/sketch/AlignmentJob.class\n",
            "bbmap/current/sketch/AlignmentJob.java\n",
            "bbmap/current/sketch/AlignmentThreadPool$AlignmentThread.class\n",
            "bbmap/current/sketch/AlignmentThreadPool.class\n",
            "bbmap/current/sketch/AlignmentThreadPool.java\n",
            "bbmap/current/sketch/AnalyzeSketchResults$SSUThread.class\n",
            "bbmap/current/sketch/AnalyzeSketchResults.class\n",
            "bbmap/current/sketch/AnalyzeSketchResults.java\n",
            "bbmap/current/sketch/Blacklist.class\n",
            "bbmap/current/sketch/Blacklist.java\n",
            "bbmap/current/sketch/BlacklistMaker$ProcessThread.class\n",
            "bbmap/current/sketch/BlacklistMaker.class\n",
            "bbmap/current/sketch/BlacklistMaker.java\n",
            "bbmap/current/sketch/BlacklistMaker2$KeyPair.class\n",
            "bbmap/current/sketch/BlacklistMaker2$ProcessThread.class\n",
            "bbmap/current/sketch/BlacklistMaker2.class\n",
            "bbmap/current/sketch/BlacklistMaker2.java\n",
            "bbmap/current/sketch/CompareBuffer.class\n",
            "bbmap/current/sketch/CompareBuffer.java\n",
            "bbmap/current/sketch/CompareSketch$CompareThread.class\n",
            "bbmap/current/sketch/CompareSketch.class\n",
            "bbmap/current/sketch/CompareSketch.java\n",
            "bbmap/current/sketch/CompareSSU$ProcessThread.class\n",
            "bbmap/current/sketch/CompareSSU.class\n",
            "bbmap/current/sketch/CompareSSU.java\n",
            "bbmap/current/sketch/Comparison$Depth2Comparator.class\n",
            "bbmap/current/sketch/Comparison$DepthComparator.class\n",
            "bbmap/current/sketch/Comparison$HitsComparator.class\n",
            "bbmap/current/sketch/Comparison$KIDComparator.class\n",
            "bbmap/current/sketch/Comparison$ScoreComparator.class\n",
            "bbmap/current/sketch/Comparison$SSUComparator.class\n",
            "bbmap/current/sketch/Comparison$VolumeComparator.class\n",
            "bbmap/current/sketch/Comparison$WKIDComparator.class\n",
            "bbmap/current/sketch/Comparison.class\n",
            "bbmap/current/sketch/Comparison.java\n",
            "bbmap/current/sketch/DisplayParams.class\n",
            "bbmap/current/sketch/DisplayParams.java\n",
            "bbmap/current/sketch/GlocalAligner.class\n",
            "bbmap/current/sketch/GlocalAligner.java\n",
            "bbmap/current/sketch/InvertKey.class\n",
            "bbmap/current/sketch/InvertKey.java\n",
            "bbmap/current/sketch/KmerLimit$ProcessThread.class\n",
            "bbmap/current/sketch/KmerLimit.class\n",
            "bbmap/current/sketch/KmerLimit.java\n",
            "bbmap/current/sketch/KmerLimit2$ProcessThread.class\n",
            "bbmap/current/sketch/KmerLimit2.class\n",
            "bbmap/current/sketch/KmerLimit2.java\n",
            "bbmap/current/sketch/MergeSketch.class\n",
            "bbmap/current/sketch/MergeSketch.java\n",
            "bbmap/current/sketch/Record.class\n",
            "bbmap/current/sketch/Record.java\n",
            "bbmap/current/sketch/RecordSet.class\n",
            "bbmap/current/sketch/RecordSet.java\n",
            "bbmap/current/sketch/ResultLineParser.class\n",
            "bbmap/current/sketch/ResultLineParser.java\n",
            "bbmap/current/sketch/SendSketch.class\n",
            "bbmap/current/sketch/SendSketch.java\n",
            "bbmap/current/sketch/Sketch.class\n",
            "bbmap/current/sketch/Sketch.java\n",
            "bbmap/current/sketch/SketchHeap.class\n",
            "bbmap/current/sketch/SketchHeap.java\n",
            "bbmap/current/sketch/SketchIdComparator.class\n",
            "bbmap/current/sketch/SketchIdComparator.java\n",
            "bbmap/current/sketch/SketchIndex$IndexThread.class\n",
            "bbmap/current/sketch/SketchIndex.class\n",
            "bbmap/current/sketch/SketchIndex.java\n",
            "bbmap/current/sketch/SketchMaker$ProcessThread.class\n",
            "bbmap/current/sketch/SketchMaker$WriteThread.class\n",
            "bbmap/current/sketch/SketchMaker.class\n",
            "bbmap/current/sketch/SketchMaker.java\n",
            "bbmap/current/sketch/SketchMakerMini.class\n",
            "bbmap/current/sketch/SketchMakerMini.java\n",
            "bbmap/current/sketch/SketchObject.class\n",
            "bbmap/current/sketch/SketchObject.java\n",
            "bbmap/current/sketch/SketchResults.class\n",
            "bbmap/current/sketch/SketchResults.java\n",
            "bbmap/current/sketch/SketchSearcher$CompareThread.class\n",
            "bbmap/current/sketch/SketchSearcher.class\n",
            "bbmap/current/sketch/SketchSearcher.java\n",
            "bbmap/current/sketch/SketchTool$LoadThread.class\n",
            "bbmap/current/sketch/SketchTool$LoadThread2.class\n",
            "bbmap/current/sketch/SketchTool$SketchThread.class\n",
            "bbmap/current/sketch/SketchTool.class\n",
            "bbmap/current/sketch/SketchTool.java\n",
            "bbmap/current/sketch/SSUMap.class\n",
            "bbmap/current/sketch/SSUMap.java\n",
            "bbmap/current/sketch/SubSketch.class\n",
            "bbmap/current/sketch/SubSketch.java\n",
            "bbmap/current/sketch/SummarizeSketchStats$SketchResultsLine.class\n",
            "bbmap/current/sketch/SummarizeSketchStats$SketchResultsSummary.class\n",
            "bbmap/current/sketch/SummarizeSketchStats.class\n",
            "bbmap/current/sketch/SummarizeSketchStats.java\n",
            "bbmap/current/sketch/Whitelist.class\n",
            "bbmap/current/sketch/Whitelist.java\n",
            "bbmap/current/sort/\n",
            "bbmap/current/sort/ContigLengthComparator.class\n",
            "bbmap/current/sort/ContigLengthComparator.java\n",
            "bbmap/current/sort/MergeSorted.class\n",
            "bbmap/current/sort/MergeSorted.java\n",
            "bbmap/current/sort/ReadComparator.class\n",
            "bbmap/current/sort/ReadComparator.java\n",
            "bbmap/current/sort/ReadComparatorClump.class\n",
            "bbmap/current/sort/ReadComparatorClump.java\n",
            "bbmap/current/sort/ReadComparatorCrispr.class\n",
            "bbmap/current/sort/ReadComparatorCrispr.java\n",
            "bbmap/current/sort/ReadComparatorFlowcell.class\n",
            "bbmap/current/sort/ReadComparatorFlowcell.java\n",
            "bbmap/current/sort/ReadComparatorID.class\n",
            "bbmap/current/sort/ReadComparatorID.java\n",
            "bbmap/current/sort/ReadComparatorList.class\n",
            "bbmap/current/sort/ReadComparatorList.java\n",
            "bbmap/current/sort/ReadComparatorMapping.class\n",
            "bbmap/current/sort/ReadComparatorMapping.java\n",
            "bbmap/current/sort/ReadComparatorName.class\n",
            "bbmap/current/sort/ReadComparatorName.java\n",
            "bbmap/current/sort/ReadComparatorPosition.class\n",
            "bbmap/current/sort/ReadComparatorPosition.java\n",
            "bbmap/current/sort/ReadComparatorRandom.class\n",
            "bbmap/current/sort/ReadComparatorRandom.java\n",
            "bbmap/current/sort/ReadComparatorTaxa.class\n",
            "bbmap/current/sort/ReadComparatorTaxa.java\n",
            "bbmap/current/sort/ReadComparatorTopological.class\n",
            "bbmap/current/sort/ReadComparatorTopological.java\n",
            "bbmap/current/sort/ReadComparatorTopological5Bit.class\n",
            "bbmap/current/sort/ReadComparatorTopological5Bit.java\n",
            "bbmap/current/sort/ReadErrorComparator.class\n",
            "bbmap/current/sort/ReadErrorComparator.java\n",
            "bbmap/current/sort/ReadLengthComparator.class\n",
            "bbmap/current/sort/ReadLengthComparator.java\n",
            "bbmap/current/sort/ReadQualityComparator.class\n",
            "bbmap/current/sort/ReadQualityComparator.java\n",
            "bbmap/current/sort/Shuffle$ShuffleThread.class\n",
            "bbmap/current/sort/Shuffle.class\n",
            "bbmap/current/sort/Shuffle.java\n",
            "bbmap/current/sort/Shuffle2$WriteThread.class\n",
            "bbmap/current/sort/Shuffle2.class\n",
            "bbmap/current/sort/Shuffle2.java\n",
            "bbmap/current/sort/SortByName$WriteThread.class\n",
            "bbmap/current/sort/SortByName.class\n",
            "bbmap/current/sort/SortByName.java\n",
            "bbmap/current/sort/SortReadsByID$1.class\n",
            "bbmap/current/sort/SortReadsByID$Block.class\n",
            "bbmap/current/sort/SortReadsByID.class\n",
            "bbmap/current/sort/SortReadsByID.java\n",
            "bbmap/current/stream/\n",
            "bbmap/current/stream/ArrayListSet$Pack.class\n",
            "bbmap/current/stream/ArrayListSet.class\n",
            "bbmap/current/stream/ArrayListSet.java\n",
            "bbmap/current/stream/BufferedMultiCros.class\n",
            "bbmap/current/stream/BufferedMultiCros.java\n",
            "bbmap/current/stream/ConcurrentCollectionReadInputStream.class\n",
            "bbmap/current/stream/ConcurrentCollectionReadInputStream.java\n",
            "bbmap/current/stream/ConcurrentDepot.class\n",
            "bbmap/current/stream/ConcurrentDepot.java\n",
            "bbmap/current/stream/ConcurrentGenericReadInputStream$1.class\n",
            "bbmap/current/stream/ConcurrentGenericReadInputStream$ReadThread.class\n",
            "bbmap/current/stream/ConcurrentGenericReadInputStream.class\n",
            "bbmap/current/stream/ConcurrentGenericReadInputStream.java\n",
            "bbmap/current/stream/ConcurrentGenericReadOutputStream.class\n",
            "bbmap/current/stream/ConcurrentGenericReadOutputStream.java\n",
            "bbmap/current/stream/ConcurrentLegacyReadInputStream.class\n",
            "bbmap/current/stream/ConcurrentLegacyReadInputStream.java\n",
            "bbmap/current/stream/ConcurrentReadInputStream.class\n",
            "bbmap/current/stream/ConcurrentReadInputStream.java\n",
            "bbmap/current/stream/ConcurrentReadInputStreamD.class\n",
            "bbmap/current/stream/ConcurrentReadInputStreamD.java\n",
            "bbmap/current/stream/ConcurrentReadListDepot.class\n",
            "bbmap/current/stream/ConcurrentReadListDepot.java\n",
            "bbmap/current/stream/ConcurrentReadOutputStream.class\n",
            "bbmap/current/stream/ConcurrentReadOutputStream.java\n",
            "bbmap/current/stream/ConcurrentReadOutputStreamD$ListenThread.class\n",
            "bbmap/current/stream/ConcurrentReadOutputStreamD.class\n",
            "bbmap/current/stream/ConcurrentReadOutputStreamD.java\n",
            "bbmap/current/stream/ConcurrentReadStreamInterface.class\n",
            "bbmap/current/stream/ConcurrentReadStreamInterface.java\n",
            "bbmap/current/stream/CrisContainer.class\n",
            "bbmap/current/stream/CrisContainer.java\n",
            "bbmap/current/stream/CrisWrapper.class\n",
            "bbmap/current/stream/CrisWrapper.java\n",
            "bbmap/current/stream/CustomHeader.class\n",
            "bbmap/current/stream/CustomHeader.java\n",
            "bbmap/current/stream/DualCris.class\n",
            "bbmap/current/stream/DualCris.java\n",
            "bbmap/current/stream/EmblReadInputStream.class\n",
            "bbmap/current/stream/EmblReadInputStream.java\n",
            "bbmap/current/stream/FastaQualReadInputStream.class\n",
            "bbmap/current/stream/FastaQualReadInputStream.java\n",
            "bbmap/current/stream/FastaReadInputStream.class\n",
            "bbmap/current/stream/FastaReadInputStream.java\n",
            "bbmap/current/stream/FastaShredInputStream.class\n",
            "bbmap/current/stream/FastaShredInputStream.java\n",
            "bbmap/current/stream/FASTQ.class\n",
            "bbmap/current/stream/FASTQ.java\n",
            "bbmap/current/stream/FastqReadInputStream.class\n",
            "bbmap/current/stream/FastqReadInputStream.java\n",
            "bbmap/current/stream/GbkReadInputStream.class\n",
            "bbmap/current/stream/GbkReadInputStream.java\n",
            "bbmap/current/stream/Header.class\n",
            "bbmap/current/stream/Header.java\n",
            "bbmap/current/stream/HeaderInputStream.class\n",
            "bbmap/current/stream/HeaderInputStream.java\n",
            "bbmap/current/stream/MDWalker.class\n",
            "bbmap/current/stream/MDWalker.java\n",
            "bbmap/current/stream/MultiCros.class\n",
            "bbmap/current/stream/MultiCros.java\n",
            "bbmap/current/stream/MultiCros2$Buffer.class\n",
            "bbmap/current/stream/MultiCros2.class\n",
            "bbmap/current/stream/MultiCros2.java\n",
            "bbmap/current/stream/MultiCros3$Buffer.class\n",
            "bbmap/current/stream/MultiCros3.class\n",
            "bbmap/current/stream/MultiCros3.java\n",
            "bbmap/current/stream/NullOutputStream.class\n",
            "bbmap/current/stream/NullOutputStream.java\n",
            "bbmap/current/stream/OnelineReadInputStream.class\n",
            "bbmap/current/stream/OnelineReadInputStream.java\n",
            "bbmap/current/stream/RandomReadInputStream3.class\n",
            "bbmap/current/stream/RandomReadInputStream3.java\n",
            "bbmap/current/stream/Read.class\n",
            "bbmap/current/stream/Read.java\n",
            "bbmap/current/stream/ReadInputStream.class\n",
            "bbmap/current/stream/ReadInputStream.java\n",
            "bbmap/current/stream/ReadStreamByteWriter.class\n",
            "bbmap/current/stream/ReadStreamByteWriter.java\n",
            "bbmap/current/stream/ReadStreamWriter$Job.class\n",
            "bbmap/current/stream/ReadStreamWriter.class\n",
            "bbmap/current/stream/ReadStreamWriter.java\n",
            "bbmap/current/stream/RTextInputStream.class\n",
            "bbmap/current/stream/RTextInputStream.java\n",
            "bbmap/current/stream/SamHeader.class\n",
            "bbmap/current/stream/SamHeader.java\n",
            "bbmap/current/stream/SamLine.class\n",
            "bbmap/current/stream/SamLine.java\n",
            "bbmap/current/stream/SamLineStreamer$ProcessThread.class\n",
            "bbmap/current/stream/SamLineStreamer.class\n",
            "bbmap/current/stream/SamLineStreamer.java\n",
            "bbmap/current/stream/SamReadInputStream.class\n",
            "bbmap/current/stream/SamReadInputStream.java\n",
            "bbmap/current/stream/SamReadStreamer$ProcessThread.class\n",
            "bbmap/current/stream/SamReadStreamer.class\n",
            "bbmap/current/stream/SamReadStreamer.java\n",
            "bbmap/current/stream/SamStreamer.class\n",
            "bbmap/current/stream/SamStreamer.java\n",
            "bbmap/current/stream/SamStreamerMF.class\n",
            "bbmap/current/stream/SamStreamerMF.java\n",
            "bbmap/current/stream/SamStreamerWrapper.class\n",
            "bbmap/current/stream/SamStreamerWrapper.java\n",
            "bbmap/current/stream/ScaffoldCoordinates.class\n",
            "bbmap/current/stream/ScaffoldCoordinates.java\n",
            "bbmap/current/stream/ScarfReadInputStream.class\n",
            "bbmap/current/stream/ScarfReadInputStream.java\n",
            "bbmap/current/stream/SequentialReadInputStream.class\n",
            "bbmap/current/stream/SequentialReadInputStream.java\n",
            "bbmap/current/stream/SiteScore$PositionComparator.class\n",
            "bbmap/current/stream/SiteScore.class\n",
            "bbmap/current/stream/SiteScore.java\n",
            "bbmap/current/stream/SiteScoreR$IDComparator.class\n",
            "bbmap/current/stream/SiteScoreR$NormalizedComparator.class\n",
            "bbmap/current/stream/SiteScoreR$PositionComparator.class\n",
            "bbmap/current/stream/SiteScoreR.class\n",
            "bbmap/current/stream/SiteScoreR.java\n",
            "bbmap/current/structures/\n",
            "bbmap/current/structures/AbstractBitSet.class\n",
            "bbmap/current/structures/AbstractBitSet.java\n",
            "bbmap/current/structures/AbstractIntHashMap.class\n",
            "bbmap/current/structures/AbstractIntHashMap.java\n",
            "bbmap/current/structures/AdapterTracker.class\n",
            "bbmap/current/structures/AdapterTracker.java\n",
            "bbmap/current/structures/AtomicBitSet.class\n",
            "bbmap/current/structures/AtomicBitSet.java\n",
            "bbmap/current/structures/Barcode.class\n",
            "bbmap/current/structures/Barcode.java\n",
            "bbmap/current/structures/BarcodeStats.class\n",
            "bbmap/current/structures/BarcodeStats.java\n",
            "bbmap/current/structures/ByteBuilder.class\n",
            "bbmap/current/structures/ByteBuilder.java\n",
            "bbmap/current/structures/CoverageArray.class\n",
            "bbmap/current/structures/CoverageArray.java\n",
            "bbmap/current/structures/CoverageArray2.class\n",
            "bbmap/current/structures/CoverageArray2.java\n",
            "bbmap/current/structures/CoverageArray3.class\n",
            "bbmap/current/structures/CoverageArray3.java\n",
            "bbmap/current/structures/CoverageArray3A.class\n",
            "bbmap/current/structures/CoverageArray3A.java\n",
            "bbmap/current/structures/CRange.class\n",
            "bbmap/current/structures/CRange.java\n",
            "bbmap/current/structures/DoubleList.class\n",
            "bbmap/current/structures/DoubleList.java\n",
            "bbmap/current/structures/EntropyTracker.class\n",
            "bbmap/current/structures/EntropyTracker.java\n",
            "bbmap/current/structures/Feature.class\n",
            "bbmap/current/structures/Feature.java\n",
            "bbmap/current/structures/FloatList.class\n",
            "bbmap/current/structures/FloatList.java\n",
            "bbmap/current/structures/Heap.class\n",
            "bbmap/current/structures/Heap.java\n",
            "bbmap/current/structures/IntHashMap.class\n",
            "bbmap/current/structures/IntHashMap.java\n",
            "bbmap/current/structures/IntHashMapBinary.class\n",
            "bbmap/current/structures/IntHashMapBinary.java\n",
            "bbmap/current/structures/IntHashSet.class\n",
            "bbmap/current/structures/IntHashSet.java\n",
            "bbmap/current/structures/IntHashSetList.class\n",
            "bbmap/current/structures/IntHashSetList.java\n",
            "bbmap/current/structures/IntList.class\n",
            "bbmap/current/structures/IntList.java\n",
            "bbmap/current/structures/IntList2.class\n",
            "bbmap/current/structures/IntList2.java\n",
            "bbmap/current/structures/IntList3.class\n",
            "bbmap/current/structures/IntList3.java\n",
            "bbmap/current/structures/IntListCompressor.class\n",
            "bbmap/current/structures/IntListCompressor.java\n",
            "bbmap/current/structures/IntLongHashMap.class\n",
            "bbmap/current/structures/IntLongHashMap.java\n",
            "bbmap/current/structures/IntMap.class\n",
            "bbmap/current/structures/IntMap.java\n",
            "bbmap/current/structures/ListNum.class\n",
            "bbmap/current/structures/ListNum.java\n",
            "bbmap/current/structures/LongArrayListHashMap.class\n",
            "bbmap/current/structures/LongArrayListHashMap.java\n",
            "bbmap/current/structures/LongHashMap.class\n",
            "bbmap/current/structures/LongHashMap.java\n",
            "bbmap/current/structures/LongHashSet.class\n",
            "bbmap/current/structures/LongHashSet.java\n",
            "bbmap/current/structures/LongHeap.class\n",
            "bbmap/current/structures/LongHeap.java\n",
            "bbmap/current/structures/LongHeapMap.class\n",
            "bbmap/current/structures/LongHeapMap.java\n",
            "bbmap/current/structures/LongHeapSet.class\n",
            "bbmap/current/structures/LongHeapSet.java\n",
            "bbmap/current/structures/LongHeapSetInterface.class\n",
            "bbmap/current/structures/LongHeapSetInterface.java\n",
            "bbmap/current/structures/LongList.class\n",
            "bbmap/current/structures/LongList.java\n",
            "bbmap/current/structures/LongList2.class\n",
            "bbmap/current/structures/LongList2.java\n",
            "bbmap/current/structures/LongList3.class\n",
            "bbmap/current/structures/LongList3.java\n",
            "bbmap/current/structures/LongListSet$LongListSetIterator.class\n",
            "bbmap/current/structures/LongListSet.class\n",
            "bbmap/current/structures/LongListSet.java\n",
            "bbmap/current/structures/LongLongHashMap.class\n",
            "bbmap/current/structures/LongLongHashMap.java\n",
            "bbmap/current/structures/LongLongListHashMap.class\n",
            "bbmap/current/structures/LongLongListHashMap.java\n",
            "bbmap/current/structures/LongM.class\n",
            "bbmap/current/structures/LongM.java\n",
            "bbmap/current/structures/LongPair.class\n",
            "bbmap/current/structures/LongPair.java\n",
            "bbmap/current/structures/MultiBitSet.class\n",
            "bbmap/current/structures/MultiBitSet.java\n",
            "bbmap/current/structures/PolymerTracker.class\n",
            "bbmap/current/structures/PolymerTracker.java\n",
            "bbmap/current/structures/Quantizer.class\n",
            "bbmap/current/structures/Quantizer.java\n",
            "bbmap/current/structures/Range.class\n",
            "bbmap/current/structures/Range.java\n",
            "bbmap/current/structures/RangePair.class\n",
            "bbmap/current/structures/RangePair.java\n",
            "bbmap/current/structures/RawBitSet.class\n",
            "bbmap/current/structures/RawBitSet.java\n",
            "bbmap/current/structures/SeqCount.class\n",
            "bbmap/current/structures/SeqCount.java\n",
            "bbmap/current/structures/SeqCountM.class\n",
            "bbmap/current/structures/SeqCountM.java\n",
            "bbmap/current/structures/SeqMap$1.class\n",
            "bbmap/current/structures/SeqMap$2.class\n",
            "bbmap/current/structures/SeqMap$3.class\n",
            "bbmap/current/structures/SeqMap.class\n",
            "bbmap/current/structures/SeqMap.java\n",
            "bbmap/current/structures/SeqPos.class\n",
            "bbmap/current/structures/SeqPos.java\n",
            "bbmap/current/structures/SeqPosM.class\n",
            "bbmap/current/structures/SeqPosM.java\n",
            "bbmap/current/structures/StringNum.class\n",
            "bbmap/current/structures/StringNum.java\n",
            "bbmap/current/structures/StringPair.class\n",
            "bbmap/current/structures/StringPair.java\n",
            "bbmap/current/structures/SuperLongList.class\n",
            "bbmap/current/structures/SuperLongList.java\n",
            "bbmap/current/tax/\n",
            "bbmap/current/tax/AccessionToTaxid$HashThread.class\n",
            "bbmap/current/tax/AccessionToTaxid.class\n",
            "bbmap/current/tax/AccessionToTaxid.java\n",
            "bbmap/current/tax/AccessionToTaxid_old$HashThread.class\n",
            "bbmap/current/tax/AccessionToTaxid_old.class\n",
            "bbmap/current/tax/AccessionToTaxid_old.java\n",
            "bbmap/current/tax/AnalyzeAccession$ProcessThread.class\n",
            "bbmap/current/tax/AnalyzeAccession.class\n",
            "bbmap/current/tax/AnalyzeAccession.java\n",
            "bbmap/current/tax/AnalyzeAccession_ST.class\n",
            "bbmap/current/tax/AnalyzeAccession_ST.java\n",
            "bbmap/current/tax/ExplodeTree.class\n",
            "bbmap/current/tax/ExplodeTree.java\n",
            "bbmap/current/tax/FilterByTaxa.class\n",
            "bbmap/current/tax/FilterByTaxa.java\n",
            "bbmap/current/tax/FindAncestor.class\n",
            "bbmap/current/tax/FindAncestor.java\n",
            "bbmap/current/tax/GiToTaxid.class\n",
            "bbmap/current/tax/GiToTaxid.java\n",
            "bbmap/current/tax/GiToTaxidInt.class\n",
            "bbmap/current/tax/GiToTaxidInt.java\n",
            "bbmap/current/tax/IDNode.class\n",
            "bbmap/current/tax/IDNode.java\n",
            "bbmap/current/tax/IDTree.class\n",
            "bbmap/current/tax/IDTree.java\n",
            "bbmap/current/tax/ImgRecord.class\n",
            "bbmap/current/tax/ImgRecord.java\n",
            "bbmap/current/tax/ImgRecord2.class\n",
            "bbmap/current/tax/ImgRecord2.java\n",
            "bbmap/current/tax/PrintTaxonomy.class\n",
            "bbmap/current/tax/PrintTaxonomy.java\n",
            "bbmap/current/tax/Query.class\n",
            "bbmap/current/tax/Query.java\n",
            "bbmap/current/tax/RenameGiToTaxid.class\n",
            "bbmap/current/tax/RenameGiToTaxid.java\n",
            "bbmap/current/tax/RenameIMG.class\n",
            "bbmap/current/tax/RenameIMG.java\n",
            "bbmap/current/tax/ShrinkAccession.class\n",
            "bbmap/current/tax/ShrinkAccession.java\n",
            "bbmap/current/tax/SplitByTaxa.class\n",
            "bbmap/current/tax/SplitByTaxa.java\n",
            "bbmap/current/tax/TaxApp$1.class\n",
            "bbmap/current/tax/TaxApp.class\n",
            "bbmap/current/tax/TaxApp.java\n",
            "bbmap/current/tax/TaxClient.class\n",
            "bbmap/current/tax/TaxClient.java\n",
            "bbmap/current/tax/TaxFilter$1.class\n",
            "bbmap/current/tax/TaxFilter.class\n",
            "bbmap/current/tax/TaxFilter.java\n",
            "bbmap/current/tax/TaxNode$CountComparator.class\n",
            "bbmap/current/tax/TaxNode.class\n",
            "bbmap/current/tax/TaxNode.java\n",
            "bbmap/current/tax/TaxServer$CompareThread.class\n",
            "bbmap/current/tax/TaxServer$HelpHandler.class\n",
            "bbmap/current/tax/TaxServer$IconHandler.class\n",
            "bbmap/current/tax/TaxServer$KillHandler.class\n",
            "bbmap/current/tax/TaxServer$SketchHandler$SketchInstance.class\n",
            "bbmap/current/tax/TaxServer$SketchHandler.class\n",
            "bbmap/current/tax/TaxServer$StatsHandler.class\n",
            "bbmap/current/tax/TaxServer$TaxHandler.class\n",
            "bbmap/current/tax/TaxServer.class\n",
            "bbmap/current/tax/TaxServer.java\n",
            "bbmap/current/tax/TaxSize.class\n",
            "bbmap/current/tax/TaxSize.java\n",
            "bbmap/current/tax/TaxTree.class\n",
            "bbmap/current/tax/TaxTree.java\n",
            "bbmap/current/template/\n",
            "bbmap/current/template/Accumulator.class\n",
            "bbmap/current/template/Accumulator.java\n",
            "bbmap/current/template/A_Sample.class\n",
            "bbmap/current/template/A_Sample.java\n",
            "bbmap/current/template/A_Sample2.class\n",
            "bbmap/current/template/A_Sample2.java\n",
            "bbmap/current/template/A_SampleBasic.class\n",
            "bbmap/current/template/A_SampleBasic.java\n",
            "bbmap/current/template/A_SampleBasic2.class\n",
            "bbmap/current/template/A_SampleBasic2.java\n",
            "bbmap/current/template/A_SampleByteFile.class\n",
            "bbmap/current/template/A_SampleByteFile.java\n",
            "bbmap/current/template/A_SampleByteFileMT$ProcessThread.class\n",
            "bbmap/current/template/A_SampleByteFileMT.class\n",
            "bbmap/current/template/A_SampleByteFileMT.java\n",
            "bbmap/current/template/A_SampleD.class\n",
            "bbmap/current/template/A_SampleD.java\n",
            "bbmap/current/template/A_SampleMT$ProcessThread.class\n",
            "bbmap/current/template/A_SampleMT.class\n",
            "bbmap/current/template/A_SampleMT.java\n",
            "bbmap/current/template/A_SampleSamStreamer$ProcessThread.class\n",
            "bbmap/current/template/A_SampleSamStreamer.class\n",
            "bbmap/current/template/A_SampleSamStreamer.java\n",
            "bbmap/current/template/A_SampleSummary.class\n",
            "bbmap/current/template/A_SampleSummary.java\n",
            "bbmap/current/template/A_Sample_Generator$ProcessThread.class\n",
            "bbmap/current/template/A_Sample_Generator.class\n",
            "bbmap/current/template/A_Sample_Generator.java\n",
            "bbmap/current/template/A_Sample_Unpaired.class\n",
            "bbmap/current/template/A_Sample_Unpaired.java\n",
            "bbmap/current/template/BBTool_ST.class\n",
            "bbmap/current/template/BBTool_ST.java\n",
            "bbmap/current/template/DoWorker.class\n",
            "bbmap/current/template/DoWorker.java\n",
            "bbmap/current/template/ThreadPoolJob.class\n",
            "bbmap/current/template/ThreadPoolJob.java\n",
            "bbmap/current/template/ThreadWaiter.class\n",
            "bbmap/current/template/ThreadWaiter.java\n",
            "bbmap/current/tracker/\n",
            "bbmap/current/tracker/AdapterTracker.class\n",
            "bbmap/current/tracker/AdapterTracker.java\n",
            "bbmap/current/tracker/Barcode.class\n",
            "bbmap/current/tracker/Barcode.java\n",
            "bbmap/current/tracker/BarcodeStats.class\n",
            "bbmap/current/tracker/BarcodeStats.java\n",
            "bbmap/current/tracker/CrisprTracker.class\n",
            "bbmap/current/tracker/CrisprTracker.java\n",
            "bbmap/current/tracker/EntropyTracker.class\n",
            "bbmap/current/tracker/EntropyTracker.java\n",
            "bbmap/current/tracker/PalindromeTracker.class\n",
            "bbmap/current/tracker/PalindromeTracker.java\n",
            "bbmap/current/tracker/PolymerTracker.class\n",
            "bbmap/current/tracker/PolymerTracker.java\n",
            "bbmap/current/tracker/QualityTracker.class\n",
            "bbmap/current/tracker/QualityTracker.java\n",
            "bbmap/current/tracker/ReadStats.class\n",
            "bbmap/current/tracker/ReadStats.java\n",
            "bbmap/current/tracker/SealStats$SealStatsLine.class\n",
            "bbmap/current/tracker/SealStats.class\n",
            "bbmap/current/tracker/SealStats.java\n",
            "bbmap/current/ukmer/\n",
            "bbmap/current/ukmer/AbstractKmerTableU$AllocThread.class\n",
            "bbmap/current/ukmer/AbstractKmerTableU.class\n",
            "bbmap/current/ukmer/AbstractKmerTableU.java\n",
            "bbmap/current/ukmer/DumpThreadU.class\n",
            "bbmap/current/ukmer/DumpThreadU.java\n",
            "bbmap/current/ukmer/HashArrayU.class\n",
            "bbmap/current/ukmer/HashArrayU.java\n",
            "bbmap/current/ukmer/HashArrayU1D$WalkerU1D.class\n",
            "bbmap/current/ukmer/HashArrayU1D.class\n",
            "bbmap/current/ukmer/HashArrayU1D.java\n",
            "bbmap/current/ukmer/HashArrayU2D.class\n",
            "bbmap/current/ukmer/HashArrayU2D.java\n",
            "bbmap/current/ukmer/HashArrayUHybrid.class\n",
            "bbmap/current/ukmer/HashArrayUHybrid.java\n",
            "bbmap/current/ukmer/HashBufferU.class\n",
            "bbmap/current/ukmer/HashBufferU.java\n",
            "bbmap/current/ukmer/HashForestU.class\n",
            "bbmap/current/ukmer/HashForestU.java\n",
            "bbmap/current/ukmer/HistogramMakerU$FillThread.class\n",
            "bbmap/current/ukmer/HistogramMakerU.class\n",
            "bbmap/current/ukmer/HistogramMakerU.java\n",
            "bbmap/current/ukmer/Kmer.class\n",
            "bbmap/current/ukmer/Kmer.java\n",
            "bbmap/current/ukmer/KmerBufferU.class\n",
            "bbmap/current/ukmer/KmerBufferU.java\n",
            "bbmap/current/ukmer/KmerNodeU.class\n",
            "bbmap/current/ukmer/KmerNodeU.java\n",
            "bbmap/current/ukmer/KmerNodeU1D.class\n",
            "bbmap/current/ukmer/KmerNodeU1D.java\n",
            "bbmap/current/ukmer/KmerNodeU2D.class\n",
            "bbmap/current/ukmer/KmerNodeU2D.java\n",
            "bbmap/current/ukmer/KmerTableSetU$LoadThread.class\n",
            "bbmap/current/ukmer/KmerTableSetU$WalkerKSTU.class\n",
            "bbmap/current/ukmer/KmerTableSetU.class\n",
            "bbmap/current/ukmer/KmerTableSetU.java\n",
            "bbmap/current/ukmer/OwnershipThread.class\n",
            "bbmap/current/ukmer/OwnershipThread.java\n",
            "bbmap/current/ukmer/WalkerU.class\n",
            "bbmap/current/ukmer/WalkerU.java\n",
            "bbmap/current/var/\n",
            "bbmap/current/var/ApplyVarsToReference$1.class\n",
            "bbmap/current/var/ApplyVarsToReference.class\n",
            "bbmap/current/var/ApplyVarsToReference.java\n",
            "bbmap/current/var/GenerateConsensusVariations$1.class\n",
            "bbmap/current/var/GenerateConsensusVariations.class\n",
            "bbmap/current/var/GenerateConsensusVariations.java\n",
            "bbmap/current/var/GenerateVarlets$ProcessThread.class\n",
            "bbmap/current/var/GenerateVarlets.class\n",
            "bbmap/current/var/GenerateVarlets.java\n",
            "bbmap/current/var/GenerateVarlets2$1.class\n",
            "bbmap/current/var/GenerateVarlets2$ProcessThread.class\n",
            "bbmap/current/var/GenerateVarlets2.class\n",
            "bbmap/current/var/GenerateVarlets2.java\n",
            "bbmap/current/var/GenerateVarlets3$1.class\n",
            "bbmap/current/var/GenerateVarlets3$ProcessThread.class\n",
            "bbmap/current/var/GenerateVarlets3.class\n",
            "bbmap/current/var/GenerateVarlets3.java\n",
            "bbmap/current/var/StackVariations$1.class\n",
            "bbmap/current/var/StackVariations$SVThread.class\n",
            "bbmap/current/var/StackVariations.class\n",
            "bbmap/current/var/StackVariations.java\n",
            "bbmap/current/var/StackVariations2$1.class\n",
            "bbmap/current/var/StackVariations2$SVThread.class\n",
            "bbmap/current/var/StackVariations2.class\n",
            "bbmap/current/var/StackVariations2.java\n",
            "bbmap/current/var/Variation.class\n",
            "bbmap/current/var/Variation.java\n",
            "bbmap/current/var/Varlet.class\n",
            "bbmap/current/var/Varlet.java\n",
            "bbmap/current/var/VarLine.class\n",
            "bbmap/current/var/VarLine.java\n",
            "bbmap/current/var2/\n",
            "bbmap/current/var2/ApplyVariants.class\n",
            "bbmap/current/var2/ApplyVariants.java\n",
            "bbmap/current/var2/CallVariants$ProcessThread.class\n",
            "bbmap/current/var2/CallVariants.class\n",
            "bbmap/current/var2/CallVariants.java\n",
            "bbmap/current/var2/CallVariants2$Sample$ProcessThread.class\n",
            "bbmap/current/var2/CallVariants2$Sample.class\n",
            "bbmap/current/var2/CallVariants2.class\n",
            "bbmap/current/var2/CallVariants2.java\n",
            "bbmap/current/var2/CompareVCF.class\n",
            "bbmap/current/var2/CompareVCF.java\n",
            "bbmap/current/var2/FilterSam$ProcessThread.class\n",
            "bbmap/current/var2/FilterSam.class\n",
            "bbmap/current/var2/FilterSam.java\n",
            "bbmap/current/var2/FilterVCF$ProcessThread.class\n",
            "bbmap/current/var2/FilterVCF.class\n",
            "bbmap/current/var2/FilterVCF.java\n",
            "bbmap/current/var2/MergeSamples$MergeThread.class\n",
            "bbmap/current/var2/MergeSamples.class\n",
            "bbmap/current/var2/MergeSamples.java\n",
            "bbmap/current/var2/Realign$ProcessThread.class\n",
            "bbmap/current/var2/Realign.class\n",
            "bbmap/current/var2/Realign.java\n",
            "bbmap/current/var2/Realigner.class\n",
            "bbmap/current/var2/Realigner.java\n",
            "bbmap/current/var2/SamFilter.class\n",
            "bbmap/current/var2/SamFilter.java\n",
            "bbmap/current/var2/Scaffold.class\n",
            "bbmap/current/var2/Scaffold.java\n",
            "bbmap/current/var2/ScafMap.class\n",
            "bbmap/current/var2/ScafMap.java\n",
            "bbmap/current/var2/SoftClipper.class\n",
            "bbmap/current/var2/SoftClipper.java\n",
            "bbmap/current/var2/Var.class\n",
            "bbmap/current/var2/Var.java\n",
            "bbmap/current/var2/VarFilter.class\n",
            "bbmap/current/var2/VarFilter.java\n",
            "bbmap/current/var2/VarKey.class\n",
            "bbmap/current/var2/VarKey.java\n",
            "bbmap/current/var2/VarMap$ProcessThread.class\n",
            "bbmap/current/var2/VarMap$VarMapIterator.class\n",
            "bbmap/current/var2/VarMap.class\n",
            "bbmap/current/var2/VarMap.java\n",
            "bbmap/current/var2/VCFFile$1.class\n",
            "bbmap/current/var2/VCFFile.class\n",
            "bbmap/current/var2/VCFFile.java\n",
            "bbmap/current/var2/VCFLine.class\n",
            "bbmap/current/var2/VCFLine.java\n",
            "bbmap/current/var2/VcfLoader$ProcessThread.class\n",
            "bbmap/current/var2/VcfLoader.class\n",
            "bbmap/current/var2/VcfLoader.java\n",
            "bbmap/current/var2/VcfWriter$ProcessThread.class\n",
            "bbmap/current/var2/VcfWriter.class\n",
            "bbmap/current/var2/VcfWriter.java\n",
            "bbmap/cutgff.sh\n",
            "bbmap/cutprimers.sh\n",
            "bbmap/decontaminate.sh\n",
            "bbmap/dedupe.sh\n",
            "bbmap/dedupe2.sh\n",
            "bbmap/dedupebymapping.sh\n",
            "bbmap/demuxbyname.sh\n",
            "bbmap/diskbench.sh\n",
            "bbmap/docs/\n",
            "bbmap/docs/bbcms.txt\n",
            "bbmap/docs/changelog.txt\n",
            "bbmap/docs/citation.txt\n",
            "bbmap/docs/compiling.txt\n",
            "bbmap/docs/git.txt\n",
            "bbmap/docs/guides/\n",
            "bbmap/docs/guides/AddAdaptersGuide.txt\n",
            "bbmap/docs/guides/A_SampleGuide.txt\n",
            "bbmap/docs/guides/BBDukGuide.txt\n",
            "bbmap/docs/guides/BBMapGuide.txt\n",
            "bbmap/docs/guides/BBMap_old_readme.txt\n",
            "bbmap/docs/guides/BBMaskGuide.txt\n",
            "bbmap/docs/guides/BBMergeGuide.txt\n",
            "bbmap/docs/guides/BBNormGuide.txt\n",
            "bbmap/docs/guides/BBSketchGuide.txt\n",
            "bbmap/docs/guides/CalcUniquenessGuide.txt\n",
            "bbmap/docs/guides/CallVariantsGuide.txt\n",
            "bbmap/docs/guides/ClumpifyGuide.txt\n",
            "bbmap/docs/guides/DedupeGuide.txt\n",
            "bbmap/docs/guides/FilterByTileGuide.txt\n",
            "bbmap/docs/guides/PreprocessingGuide.txt\n",
            "bbmap/docs/guides/ReformatGuide.txt\n",
            "bbmap/docs/guides/RepairGuide.txt\n",
            "bbmap/docs/guides/SealGuide.txt\n",
            "bbmap/docs/guides/SplitNexteraGuide.txt\n",
            "bbmap/docs/guides/StatsGuide.txt\n",
            "bbmap/docs/guides/TadpoleGuide.txt\n",
            "bbmap/docs/guides/TaxonomyGuide.txt\n",
            "bbmap/docs/Legal.txt\n",
            "bbmap/docs/Legal_Illumina.txt\n",
            "bbmap/docs/readme.txt\n",
            "bbmap/docs/readme_config.txt\n",
            "bbmap/docs/readme_filetypes.txt\n",
            "bbmap/docs/RestartingServers.txt\n",
            "bbmap/docs/TableOfContents.txt\n",
            "bbmap/docs/ToolDescriptions.txt\n",
            "bbmap/docs/UsageGuide.txt\n",
            "bbmap/estherfilter.sh\n",
            "bbmap/explodetree.sh\n",
            "bbmap/fetchproks.sh\n",
            "bbmap/filterassemblysummary.sh\n",
            "bbmap/filterbarcodes.sh\n",
            "bbmap/filterbycoverage.sh\n",
            "bbmap/filterbyname.sh\n",
            "bbmap/filterbysequence.sh\n",
            "bbmap/filterbytaxa.sh\n",
            "bbmap/filterbytile.sh\n",
            "bbmap/filterlines.sh\n",
            "bbmap/filterqc.sh\n",
            "bbmap/filtersam.sh\n",
            "bbmap/filtersilva.sh\n",
            "bbmap/filtersubs.sh\n",
            "bbmap/filtervcf.sh\n",
            "bbmap/findrepeats.sh\n",
            "bbmap/fixgaps.sh\n",
            "bbmap/fungalrelease.sh\n",
            "bbmap/fuse.sh\n",
            "bbmap/gbff2gff.sh\n",
            "bbmap/getreads.sh\n",
            "bbmap/gi2ancestors.sh\n",
            "bbmap/gi2taxid.sh\n",
            "bbmap/gitable.sh\n",
            "bbmap/grademerge.sh\n",
            "bbmap/gradesam.sh\n",
            "bbmap/icecreamfinder.sh\n",
            "bbmap/icecreamgrader.sh\n",
            "bbmap/icecreammaker.sh\n",
            "bbmap/idmatrix.sh\n",
            "bbmap/idtree.sh\n",
            "bbmap/invertkey.sh\n",
            "bbmap/jni/\n",
            "bbmap/jni/align2_BandedAlignerJNI.h\n",
            "bbmap/jni/align2_MultiStateAligner11tsJNI.h\n",
            "bbmap/jni/BandedAlignerJNI.c\n",
            "bbmap/jni/BandedAlignerJNI.o\n",
            "bbmap/jni/BBMergeOverlapper.c\n",
            "bbmap/jni/BBMergeOverlapper.o\n",
            "bbmap/jni/CMakeLists.txt\n",
            "bbmap/jni/IceCreamAlignerJNI.c\n",
            "bbmap/jni/IceCreamAlignerJNI.o\n",
            "bbmap/jni/icecream_IceCreamAlignerJNI.h\n",
            "bbmap/jni/jgi_BBMergeOverlapper.h\n",
            "bbmap/jni/libbbtoolsjni.dylib\n",
            "bbmap/jni/libbbtoolsjni.so\n",
            "bbmap/jni/makefile.linux\n",
            "bbmap/jni/makefile.osx\n",
            "bbmap/jni/MultiStateAligner11tsJNI.c\n",
            "bbmap/jni/MultiStateAligner11tsJNI.o\n",
            "bbmap/jni/README.txt\n",
            "bbmap/jni/run.sh\n",
            "bbmap/kapastats.sh\n",
            "bbmap/kcompress.sh\n",
            "bbmap/keepbestcopy.sh\n",
            "bbmap/khist.sh\n",
            "bbmap/kmercountexact.sh\n",
            "bbmap/kmercountmulti.sh\n",
            "bbmap/kmercoverage.sh\n",
            "bbmap/kmerfilterset.sh\n",
            "bbmap/kmerlimit.sh\n",
            "bbmap/kmerlimit2.sh\n",
            "bbmap/kmerposition.sh\n",
            "bbmap/kmutate.sh\n",
            "bbmap/license.txt\n",
            "bbmap/lilypad.sh\n",
            "bbmap/loadreads.sh\n",
            "bbmap/loglog.sh\n",
            "bbmap/makechimeras.sh\n",
            "bbmap/makecontaminatedgenomes.sh\n",
            "bbmap/makepolymers.sh\n",
            "bbmap/mapPacBio.sh\n",
            "bbmap/matrixtocolumns.sh\n",
            "bbmap/mergebarcodes.sh\n",
            "bbmap/mergeOTUs.sh\n",
            "bbmap/mergepgm.sh\n",
            "bbmap/mergeribo.sh\n",
            "bbmap/mergesam.sh\n",
            "bbmap/mergesketch.sh\n",
            "bbmap/mergesorted.sh\n",
            "bbmap/msa.sh\n",
            "bbmap/mutate.sh\n",
            "bbmap/muxbyname.sh\n",
            "bbmap/partition.sh\n",
            "bbmap/phylip2fasta.sh\n",
            "bbmap/pileup.sh\n",
            "bbmap/pipelines/\n",
            "bbmap/pipelines/assembleMito_Illumina.sh\n",
            "bbmap/pipelines/assembleMito_PacBio.sh\n",
            "bbmap/pipelines/assemblyPipeline.sh\n",
            "bbmap/pipelines/callInsertions.sh\n",
            "bbmap/pipelines/covid/\n",
            "bbmap/pipelines/covid/makeSummary.sh\n",
            "bbmap/pipelines/covid/processCorona.sh\n",
            "bbmap/pipelines/covid/processCoronaWrapper.sh\n",
            "bbmap/pipelines/covid/readme.txt\n",
            "bbmap/pipelines/covid/recal.sh\n",
            "bbmap/pipelines/crisprPipeline.sh\n",
            "bbmap/pipelines/cutRna.sh\n",
            "bbmap/pipelines/fetch/\n",
            "bbmap/pipelines/fetch/fetchMito.sh\n",
            "bbmap/pipelines/fetch/fetchNt.sh\n",
            "bbmap/pipelines/fetch/fetchNtOuter.sh\n",
            "bbmap/pipelines/fetch/fetchPlasmid.sh\n",
            "bbmap/pipelines/fetch/fetchPlastid.sh\n",
            "bbmap/pipelines/fetch/fetchProkByGenus.sh\n",
            "bbmap/pipelines/fetch/fetchRefSeq.sh\n",
            "bbmap/pipelines/fetch/fetchRefSeqClades.sh\n",
            "bbmap/pipelines/fetch/fetchRefSeqCladesOuter.sh\n",
            "bbmap/pipelines/fetch/fetchSilva.sh\n",
            "bbmap/pipelines/fetch/fetchTaxonomy.sh\n",
            "bbmap/pipelines/fetch/runRefSeqProtein.sh\n",
            "bbmap/pipelines/fetch/sketchRefSeq.sh\n",
            "bbmap/pipelines/makeRiboKmers.sh\n",
            "bbmap/pipelines/processIMG.sh\n",
            "bbmap/pipelines/server/\n",
            "bbmap/pipelines/server/startNtServerVM.sh\n",
            "bbmap/pipelines/server/startProteinServerVM.sh\n",
            "bbmap/pipelines/server/startRefseqServerVM.sh\n",
            "bbmap/pipelines/server/startSilvaServerVM.sh\n",
            "bbmap/pipelines/server/startTaxServerVM.sh\n",
            "bbmap/pipelines/silva/\n",
            "bbmap/pipelines/silva/fetchSilva.sh\n",
            "bbmap/pipelines/silva/make15mers.sh\n",
            "bbmap/pipelines/silva/makeCoveringSetLsu.sh\n",
            "bbmap/pipelines/silva/makeCoveringSetSsu.sh\n",
            "bbmap/pipelines/silva/makeRep.sh\n",
            "bbmap/pipelines/testPlatformQuality.sh\n",
            "bbmap/pipelines/testsketch.sh\n",
            "bbmap/pipelines/variantPipeline.sh\n",
            "bbmap/plotflowcell.sh\n",
            "bbmap/plotgc.sh\n",
            "bbmap/postfilter.sh\n",
            "bbmap/printtime.sh\n",
            "bbmap/processfrag.sh\n",
            "bbmap/processhi-c.sh\n",
            "bbmap/processspeed.sh\n",
            "bbmap/pytools/\n",
            "bbmap/pytools/.gitignore\n",
            "bbmap/pytools/css/\n",
            "bbmap/pytools/css/report.css\n",
            "bbmap/pytools/filter.py\n",
            "bbmap/pytools/filter_desc/\n",
            "bbmap/pytools/filter_desc/3primerna.txt\n",
            "bbmap/pytools/filter_desc/bisulphite.txt\n",
            "bbmap/pytools/filter_desc/cell-enrichment.txt\n",
            "bbmap/pytools/filter_desc/chip-seq.txt\n",
            "bbmap/pytools/filter_desc/clip-pe.txt\n",
            "bbmap/pytools/filter_desc/clrs.txt\n",
            "bbmap/pytools/filter_desc/dna.txt\n",
            "bbmap/pytools/filter_desc/filter.json\n",
            "bbmap/pytools/filter_desc/filter_mt.json\n",
            "bbmap/pytools/filter_desc/filter_nextera.json\n",
            "bbmap/pytools/filter_desc/filter_report.txt\n",
            "bbmap/pytools/filter_desc/fungal.txt\n",
            "bbmap/pytools/filter_desc/iso.txt\n",
            "bbmap/pytools/filter_desc/itag.txt\n",
            "bbmap/pytools/filter_desc/lfpe.txt\n",
            "bbmap/pytools/filter_desc/metagenome.txt\n",
            "bbmap/pytools/filter_desc/microtrans.txt\n",
            "bbmap/pytools/filter_desc/mtaa.txt\n",
            "bbmap/pytools/filter_desc/nextera-lmp.txt\n",
            "bbmap/pytools/filter_desc/plant-2x150.txt\n",
            "bbmap/pytools/filter_desc/plant-2x250.txt\n",
            "bbmap/pytools/filter_desc/rna.txt\n",
            "bbmap/pytools/filter_desc/rnawohuman.txt\n",
            "bbmap/pytools/filter_desc/sag.txt\n",
            "bbmap/pytools/filter_desc/smrna.txt\n",
            "bbmap/pytools/filter_desc/sop_list.txt\n",
            "bbmap/pytools/filter_desc/viral-metagenome.txt\n",
            "bbmap/pytools/filter_param/\n",
            "bbmap/pytools/filter_param/3primerna.config\n",
            "bbmap/pytools/filter_param/bisulphite.config\n",
            "bbmap/pytools/filter_param/cell-enrichment.config\n",
            "bbmap/pytools/filter_param/chip-seq.config\n",
            "bbmap/pytools/filter_param/clip-pe.config\n",
            "bbmap/pytools/filter_param/clrs.config\n",
            "bbmap/pytools/filter_param/dna.config\n",
            "bbmap/pytools/filter_param/fungal.config\n",
            "bbmap/pytools/filter_param/iso.config\n",
            "bbmap/pytools/filter_param/itag.config\n",
            "bbmap/pytools/filter_param/lfpe.config\n",
            "bbmap/pytools/filter_param/metagenome.config\n",
            "bbmap/pytools/filter_param/microtrans.config\n",
            "bbmap/pytools/filter_param/mtaa.config\n",
            "bbmap/pytools/filter_param/nextera-lmp.config\n",
            "bbmap/pytools/filter_param/nextseq.config\n",
            "bbmap/pytools/filter_param/plant-2x150.config\n",
            "bbmap/pytools/filter_param/plant-2x250.config\n",
            "bbmap/pytools/filter_param/rna.config\n",
            "bbmap/pytools/filter_param/rnawohuman.config\n",
            "bbmap/pytools/filter_param/sag.config\n",
            "bbmap/pytools/filter_param/smrna.config\n",
            "bbmap/pytools/filter_param/viral-metagenome.config\n",
            "bbmap/pytools/images/\n",
            "bbmap/pytools/images/favicon.ico\n",
            "bbmap/pytools/images/JGI_logo.jpg\n",
            "bbmap/pytools/images/jgi_log_transparent.png\n",
            "bbmap/pytools/lib/\n",
            "bbmap/pytools/lib/common.py\n",
            "bbmap/pytools/lib/html_utility.py\n",
            "bbmap/pytools/lib/kmercount_pos.py\n",
            "bbmap/pytools/lib/os_utility.py\n",
            "bbmap/pytools/lib/readqc_constants.py\n",
            "bbmap/pytools/lib/readqc_report.py\n",
            "bbmap/pytools/lib/readqc_utils.py\n",
            "bbmap/pytools/lib/readSeq.py\n",
            "bbmap/pytools/lib/rqc_constants.py\n",
            "bbmap/pytools/lib/rqc_fastq.py\n",
            "bbmap/pytools/lib/rqc_utility.py\n",
            "bbmap/pytools/ReadMe\n",
            "bbmap/pytools/readqc.py\n",
            "bbmap/pytools/template/\n",
            "bbmap/pytools/template/filter_body_template.html\n",
            "bbmap/pytools/template/readqc_artifacts.html\n",
            "bbmap/pytools/template/readqc_body_template.html\n",
            "bbmap/pytools/template/template.html\n",
            "bbmap/pytools/tools/\n",
            "bbmap/pytools/tools/patterN_fastq.pl\n",
            "bbmap/randomgenome.sh\n",
            "bbmap/randomreads.sh\n",
            "bbmap/readlength.sh\n",
            "bbmap/README.md\n",
            "bbmap/readqc.sh\n",
            "bbmap/reducesilva.sh\n",
            "bbmap/reformat.sh\n",
            "bbmap/reformatpb.sh\n",
            "bbmap/removebadbarcodes.sh\n",
            "bbmap/removecatdogmousehuman.sh\n",
            "bbmap/removehuman.sh\n",
            "bbmap/removehuman2.sh\n",
            "bbmap/removemicrobes.sh\n",
            "bbmap/removesmartbell.sh\n",
            "bbmap/rename.sh\n",
            "bbmap/renameimg.sh\n",
            "bbmap/repair.sh\n",
            "bbmap/replaceheaders.sh\n",
            "bbmap/representative.sh\n",
            "bbmap/resources/\n",
            "bbmap/resources/16S_15mers.fa\n",
            "bbmap/resources/16S_consensus_sequence.fa\n",
            "bbmap/resources/18S_15mers.fa\n",
            "bbmap/resources/18S_consensus_sequence.fa\n",
            "bbmap/resources/23S_15mers.fa\n",
            "bbmap/resources/23S_consensus_sequence.fa\n",
            "bbmap/resources/5S_10mers.fa\n",
            "bbmap/resources/5S_11mers.fa\n",
            "bbmap/resources/5S_12mers.fa\n",
            "bbmap/resources/5S_15mers.fa\n",
            "bbmap/resources/5S_9mers.fa\n",
            "bbmap/resources/5S_consensus_sequence.fa\n",
            "bbmap/resources/adapters.fa\n",
            "bbmap/resources/adapters_no_transposase.fa.gz\n",
            "bbmap/resources/artic3.fasta\n",
            "bbmap/resources/bbmerge.bbnet\n",
            "bbmap/resources/blacklist_nt_merged.sketch\n",
            "bbmap/resources/blacklist_prokprot_merged.sketch\n",
            "bbmap/resources/blacklist_refseq_merged.sketch\n",
            "bbmap/resources/blacklist_silva_merged.sketch\n",
            "bbmap/resources/contents.txt\n",
            "bbmap/resources/Covid19_ref.fa\n",
            "bbmap/resources/crelox.fa.gz\n",
            "bbmap/resources/crispr.bbnet.gz\n",
            "bbmap/resources/favicon.ico\n",
            "bbmap/resources/ITS_animal_consensus_sequence.fq\n",
            "bbmap/resources/ITS_consensus_sequence.fq\n",
            "bbmap/resources/ITS_fungi_consensus_sequence.fq\n",
            "bbmap/resources/ITS_nonfungi_consensus_sequence.fq\n",
            "bbmap/resources/ITS_other_consensus_sequence.fq\n",
            "bbmap/resources/ITS_plant_consensus_sequence.fq\n",
            "bbmap/resources/ITS_universal_consensus_sequence.fq\n",
            "bbmap/resources/kapatags.L40.fa\n",
            "bbmap/resources/lambda.fa.gz\n",
            "bbmap/resources/lfpe.linker.fa.gz\n",
            "bbmap/resources/lsu_15mers.fa\n",
            "bbmap/resources/m16S_consensus_sequence.fa\n",
            "bbmap/resources/m18S_consensus_sequence.fa\n",
            "bbmap/resources/minorAlleleCount.txt\n",
            "bbmap/resources/model.pgm\n",
            "bbmap/resources/mtst.fa\n",
            "bbmap/resources/nextera.fa.gz\n",
            "bbmap/resources/nextera_LMP_adapter.fa.gz\n",
            "bbmap/resources/nextera_LMP_linker.fa.gz\n",
            "bbmap/resources/p16S_consensus_sequence.fa\n",
            "bbmap/resources/PacBioAdapter.fa\n",
            "bbmap/resources/phix174_ill.ref.fa.gz\n",
            "bbmap/resources/phix_adapters.fa.gz\n",
            "bbmap/resources/pJET1.2.fa\n",
            "bbmap/resources/polyA.fa.gz\n",
            "bbmap/resources/primes.txt.gz\n",
            "bbmap/resources/remote_files.txt\n",
            "bbmap/resources/remote_files_old.txt\n",
            "bbmap/resources/sample1.fq.gz\n",
            "bbmap/resources/sample2.fq.gz\n",
            "bbmap/resources/sequencing_artifacts.fa.gz\n",
            "bbmap/resources/short.fa\n",
            "bbmap/resources/SIP_spikeins.fa.gz\n",
            "bbmap/resources/ssu_15mers.fa\n",
            "bbmap/resources/tax_server.html\n",
            "bbmap/resources/tRNA_10mers.fa\n",
            "bbmap/resources/tRNA_11mers.fa\n",
            "bbmap/resources/tRNA_15mers.fa\n",
            "bbmap/resources/tRNA_16mers.fa\n",
            "bbmap/resources/tRNA_consensus_sequence.fa\n",
            "bbmap/resources/truseq.fa.gz\n",
            "bbmap/resources/truseq_rna.fa.gz\n",
            "bbmap/rqcfilter.sh\n",
            "bbmap/rqcfilter2.sh\n",
            "bbmap/runhmm.sh\n",
            "bbmap/samtoroc.sh\n",
            "bbmap/scoresequence.sh\n",
            "bbmap/seal.sh\n",
            "bbmap/sendsketch.sh\n",
            "bbmap/seqtovec.sh\n",
            "bbmap/shred.sh\n",
            "bbmap/shrinkaccession.sh\n",
            "bbmap/shuffle.sh\n",
            "bbmap/shuffle2.sh\n",
            "bbmap/sketch.sh\n",
            "bbmap/sketchblacklist.sh\n",
            "bbmap/sketchblacklist2.sh\n",
            "bbmap/sortbyname.sh\n",
            "bbmap/splitbytaxa.sh\n",
            "bbmap/splitnextera.sh\n",
            "bbmap/splitribo.sh\n",
            "bbmap/splitsam.sh\n",
            "bbmap/splitsam4way.sh\n",
            "bbmap/splitsam6way.sh\n",
            "bbmap/stats.sh\n",
            "bbmap/statswrapper.sh\n",
            "bbmap/streamsam.sh\n",
            "bbmap/subsketch.sh\n",
            "bbmap/summarizecontam.sh\n",
            "bbmap/summarizecoverage.sh\n",
            "bbmap/summarizecrossblock.sh\n",
            "bbmap/summarizemerge.sh\n",
            "bbmap/summarizequast.sh\n",
            "bbmap/summarizescafstats.sh\n",
            "bbmap/summarizeseal.sh\n",
            "bbmap/summarizesketch.sh\n",
            "bbmap/synthmda.sh\n",
            "bbmap/tadpipe.sh\n",
            "bbmap/tadpole.sh\n",
            "bbmap/tadwrapper.sh\n",
            "bbmap/taxonomy.sh\n",
            "bbmap/taxserver.sh\n",
            "bbmap/taxsize.sh\n",
            "bbmap/taxtree.sh\n",
            "bbmap/testfilesystem.sh\n",
            "bbmap/testformat.sh\n",
            "bbmap/testformat2.sh\n",
            "bbmap/tetramerfreq.sh\n",
            "bbmap/textfile.sh\n",
            "bbmap/train.sh\n",
            "bbmap/translate6frames.sh\n",
            "bbmap/unicode2ascii.sh\n",
            "bbmap/unzip.sh\n",
            "bbmap/vcf2gff.sh\n",
            "bbmap/webcheck.sh\n",
            "bbmap/Xcalcmem.sh\n",
            "--2023-11-07 18:33:41--  https://github.com/arq5x/bedtools2/releases/download/v2.31.0/bedtools-2.31.0.tar.gz\n",
            "Resolving github.com (github.com)... 140.82.112.3\n",
            "Connecting to github.com (github.com)|140.82.112.3|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/15059334/385e268e-7f82-4962-adfc-52ab1a77573f?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20231107%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20231107T183341Z&X-Amz-Expires=300&X-Amz-Signature=f4694846158c2f279dfc5ecc920d5facee144a3c24eabcf0e3618d9880a3d0ce&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=15059334&response-content-disposition=attachment%3B%20filename%3Dbedtools-2.31.0.tar.gz&response-content-type=application%2Foctet-stream [following]\n",
            "--2023-11-07 18:33:41--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/15059334/385e268e-7f82-4962-adfc-52ab1a77573f?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20231107%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20231107T183341Z&X-Amz-Expires=300&X-Amz-Signature=f4694846158c2f279dfc5ecc920d5facee144a3c24eabcf0e3618d9880a3d0ce&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=15059334&response-content-disposition=attachment%3B%20filename%3Dbedtools-2.31.0.tar.gz&response-content-type=application%2Foctet-stream\n",
            "Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...\n",
            "Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.108.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 21451189 (20M) [application/octet-stream]\n",
            "Saving to: ‘bedtools-2.31.0.tar.gz’\n",
            "\n",
            "bedtools-2.31.0.tar 100%[===================>]  20.46M  --.-KB/s    in 0.1s    \n",
            "\n",
            "2023-11-07 18:33:41 (140 MB/s) - ‘bedtools-2.31.0.tar.gz’ saved [21451189/21451189]\n",
            "\n",
            "bedtools2/\n",
            "bedtools2/obj/\n",
            "bedtools2/LICENSE\n",
            "bedtools2/test/\n",
            "bedtools2/bin/\n",
            "bedtools2/Makefile\n",
            "bedtools2/README.md\n",
            "bedtools2/genomes/\n",
            "bedtools2/.gitignore\n",
            "bedtools2/scripts/\n",
            "bedtools2/data/\n",
            "bedtools2/src/\n",
            "bedtools2/src/spacingFile/\n",
            "bedtools2/src/sortBed/\n",
            "bedtools2/src/split/\n",
            "bedtools2/src/bedToBam/\n",
            "bedtools2/src/complementFile/\n",
            "bedtools2/src/expand/\n",
            "bedtools2/src/fisher/\n",
            "bedtools2/src/.DS_Store\n",
            "bedtools2/src/linksBed/\n",
            "bedtools2/src/pairToPair/\n",
            "bedtools2/src/bedtools.cpp\n",
            "bedtools2/src/mapFile/\n",
            "bedtools2/src/slopBed/\n",
            "bedtools2/src/annotateBed/\n",
            "bedtools2/src/randomBed/\n",
            "bedtools2/src/getOverlap/\n",
            "bedtools2/src/reldist/\n",
            "bedtools2/src/nucBed/\n",
            "bedtools2/src/bedToIgv/\n",
            "bedtools2/src/regressTest/\n",
            "bedtools2/src/maskFastaFromBed/\n",
            "bedtools2/src/genomeCoverageBed/\n",
            "bedtools2/src/utils/\n",
            "bedtools2/src/windowBed/\n",
            "bedtools2/src/jaccard/\n",
            "bedtools2/src/groupBy/\n",
            "bedtools2/src/bamToFastq/\n",
            "bedtools2/src/htsutil.cpp\n",
            "bedtools2/src/fastaFromBed/\n",
            "bedtools2/src/nekSandbox1/\n",
            "bedtools2/src/unionBedGraphs/\n",
            "bedtools2/src/bed12ToBed6/\n",
            "bedtools2/src/subtractFile/\n",
            "bedtools2/src/bedpeToBam/\n",
            "bedtools2/src/windowMaker/\n",
            "bedtools2/src/coverageFile/\n",
            "bedtools2/src/summaryFile/\n",
            "bedtools2/src/multiIntersectBed/\n",
            "bedtools2/src/intersectFile/\n",
            "bedtools2/src/closestFile/\n",
            "bedtools2/src/shuffleBed/\n",
            "bedtools2/src/flankBed/\n",
            "bedtools2/src/tagBam/\n",
            "bedtools2/src/multiBamCov/\n",
            "bedtools2/src/sampleFile/\n",
            "bedtools2/src/clusterBed/\n",
            "bedtools2/src/shiftBed/\n",
            "bedtools2/src/mergeFile/\n",
            "bedtools2/src/bamToBed/\n",
            "bedtools2/src/pairToBed/\n",
            "bedtools2/src/pairToBed/pairToBed.h\n",
            "bedtools2/src/pairToBed/pairToBedMain.cpp\n",
            "bedtools2/src/pairToBed/Makefile.frag\n",
            "bedtools2/src/pairToBed/pairToBed.cpp\n",
            "bedtools2/src/bamToBed/bamToBed.cpp\n",
            "bedtools2/src/bamToBed/Makefile.frag\n",
            "bedtools2/src/mergeFile/mergeFile.cpp\n",
            "bedtools2/src/mergeFile/Makefile.frag\n",
            "bedtools2/src/mergeFile/mergeHelp.cpp\n",
            "bedtools2/src/mergeFile/mergeFile.h\n",
            "bedtools2/src/shiftBed/shiftBed.cpp\n",
            "bedtools2/src/shiftBed/shiftBedMain.cpp\n",
            "bedtools2/src/shiftBed/Makefile.frag\n",
            "bedtools2/src/shiftBed/shiftBed.h\n",
            "bedtools2/src/clusterBed/clusterBed.h\n",
            "bedtools2/src/clusterBed/clusterBed.cpp\n",
            "bedtools2/src/clusterBed/clusterMain.cpp\n",
            "bedtools2/src/clusterBed/Makefile.frag\n",
            "bedtools2/src/sampleFile/sampleFile.cpp\n",
            "bedtools2/src/sampleFile/sampleFile.h\n",
            "bedtools2/src/sampleFile/Makefile.frag\n",
            "bedtools2/src/sampleFile/sampleHelp.cpp\n",
            "bedtools2/src/multiBamCov/multiBamCov.cpp\n",
            "bedtools2/src/multiBamCov/Makefile.frag\n",
            "bedtools2/src/multiBamCov/multiBamCov.h\n",
            "bedtools2/src/multiBamCov/multiBamCovMain.cpp\n",
            "bedtools2/src/tagBam/tagBam.cpp\n",
            "bedtools2/src/tagBam/tagBamMain.cpp\n",
            "bedtools2/src/tagBam/tagBam.h\n",
            "bedtools2/src/tagBam/Makefile.frag\n",
            "bedtools2/src/flankBed/flankBed.cpp\n",
            "bedtools2/src/flankBed/flankBedMain.cpp\n",
            "bedtools2/src/flankBed/test.fa.fai\n",
            "bedtools2/src/flankBed/test.gff\n",
            "bedtools2/src/flankBed/Makefile.frag\n",
            "bedtools2/src/flankBed/test.fa\n",
            "bedtools2/src/flankBed/flankBed.h\n",
            "bedtools2/src/shuffleBed/shuffleBed.h\n",
            "bedtools2/src/shuffleBed/shuffleBed.cpp\n",
            "bedtools2/src/shuffleBed/Makefile.frag\n",
            "bedtools2/src/shuffleBed/shuffleBedMain.cpp\n",
            "bedtools2/src/closestFile/closestHelp.cpp\n",
            "bedtools2/src/closestFile/closestFile.h\n",
            "bedtools2/src/closestFile/closestFile.cpp\n",
            "bedtools2/src/closestFile/Makefile.frag\n",
            "bedtools2/src/intersectFile/intersectHelp.cpp\n",
            "bedtools2/src/intersectFile/Makefile.frag\n",
            "bedtools2/src/intersectFile/intersectFile.cpp\n",
            "bedtools2/src/intersectFile/intersectFile.h\n",
            "bedtools2/src/multiIntersectBed/multiIntersectBed.cpp\n",
            "bedtools2/src/multiIntersectBed/multiIntersectBed.h\n",
            "bedtools2/src/multiIntersectBed/Makefile.frag\n",
            "bedtools2/src/multiIntersectBed/multiIntersectBedMain.cpp\n",
            "bedtools2/src/summaryFile/summaryHelp.cpp\n",
            "bedtools2/src/summaryFile/Makefile.frag\n",
            "bedtools2/src/summaryFile/summaryFile.h\n",
            "bedtools2/src/summaryFile/summaryFile.cpp\n",
            "bedtools2/src/coverageFile/coverageFile.cpp\n",
            "bedtools2/src/coverageFile/coverageFile.h\n",
            "bedtools2/src/coverageFile/Makefile.frag\n",
            "bedtools2/src/coverageFile/coverageHelp.cpp\n",
            "bedtools2/src/windowMaker/windowMakerMain.cpp\n",
            "bedtools2/src/windowMaker/windowMaker.h\n",
            "bedtools2/src/windowMaker/windowMaker.cpp\n",
            "bedtools2/src/windowMaker/Makefile.frag\n",
            "bedtools2/src/bedpeToBam/notes.txt\n",
            "bedtools2/src/bedpeToBam/bedpeToBam.cpp\n",
            "bedtools2/src/bedpeToBam/Makefile.frag\n",
            "bedtools2/src/subtractFile/subtractHelp.cpp\n",
            "bedtools2/src/subtractFile/subtractFile.h\n",
            "bedtools2/src/subtractFile/Makefile.frag\n",
            "bedtools2/src/subtractFile/subtractFile.cpp\n",
            "bedtools2/src/bed12ToBed6/bed12ToBed6.cpp\n",
            "bedtools2/src/bed12ToBed6/Makefile.frag\n",
            "bedtools2/src/unionBedGraphs/unionBedGraphs.cpp\n",
            "bedtools2/src/unionBedGraphs/unionBedGraphsMain.cpp\n",
            "bedtools2/src/unionBedGraphs/Makefile.frag\n",
            "bedtools2/src/unionBedGraphs/unionBedGraphs.h\n",
            "bedtools2/src/nekSandbox1/FormatGuess.h\n",
            "bedtools2/src/nekSandbox1/Makefile\n",
            "bedtools2/src/nekSandbox1/nekSandboxMain.cpp\n",
            "bedtools2/src/fastaFromBed/fastaFromBedMain.cpp\n",
            "bedtools2/src/fastaFromBed/fastaFromBed.cpp\n",
            "bedtools2/src/fastaFromBed/fastaFromBed.h\n",
            "bedtools2/src/fastaFromBed/Makefile.frag\n",
            "bedtools2/src/bamToFastq/bamToFastq.h\n",
            "bedtools2/src/bamToFastq/bamToFastq.cpp\n",
            "bedtools2/src/bamToFastq/bamToFastqMain.cpp\n",
            "bedtools2/src/bamToFastq/Makefile.frag\n",
            "bedtools2/src/groupBy/groupBy.cpp\n",
            "bedtools2/src/groupBy/groupBy.h\n",
            "bedtools2/src/groupBy/Makefile.frag\n",
            "bedtools2/src/groupBy/groupByHelp.cpp\n",
            "bedtools2/src/jaccard/jaccard.cpp\n",
            "bedtools2/src/jaccard/Makefile.frag\n",
            "bedtools2/src/jaccard/jaccardHelp.cpp\n",
            "bedtools2/src/jaccard/jaccard.h\n",
            "bedtools2/src/windowBed/windowBed.cpp\n",
            "bedtools2/src/windowBed/windowBed.h\n",
            "bedtools2/src/windowBed/windowMain.cpp\n",
            "bedtools2/src/windowBed/Makefile.frag\n",
            "bedtools2/src/utils/tabFile/\n",
            "bedtools2/src/utils/htslib/\n",
            "bedtools2/src/utils/NewChromsweep/\n",
            "bedtools2/src/utils/RecordOutputMgr/\n",
            "bedtools2/src/utils/bedFile/\n",
            "bedtools2/src/utils/Contexts/\n",
            "bedtools2/src/utils/.DS_Store\n",
            "bedtools2/src/utils/BamTools/\n",
            "bedtools2/src/utils/BlockedIntervals/\n",
            "bedtools2/src/utils/general/\n",
            "bedtools2/src/utils/driver/\n",
            "bedtools2/src/utils/Point/\n",
            "bedtools2/src/utils/bedGraphFile/\n",
            "bedtools2/src/utils/sequenceUtilities/\n",
            "bedtools2/src/utils/Fasta/\n",
            "bedtools2/src/utils/gzstream/\n",
            "bedtools2/src/utils/fileType/\n",
            "bedtools2/src/utils/KeyListOps/\n",
            "bedtools2/src/utils/BamTools-Ancillary/\n",
            "bedtools2/src/utils/GenomeFile/\n",
            "bedtools2/src/utils/version/\n",
            "bedtools2/src/utils/ToolBase/\n",
            "bedtools2/src/utils/lineFileUtilities/\n",
            "bedtools2/src/utils/bedFilePE/\n",
            "bedtools2/src/utils/chromsweep/\n",
            "bedtools2/src/utils/FileRecordTools/\n",
            "bedtools2/src/utils/BinTree/\n",
            "bedtools2/src/utils/VectorOps/\n",
            "bedtools2/src/utils/stringUtilities/\n",
            "bedtools2/src/utils/stringUtilities/stringUtilities.h\n",
            "bedtools2/src/utils/VectorOps/VectorOps.cpp\n",
            "bedtools2/src/utils/VectorOps/Makefile.frag\n",
            "bedtools2/src/utils/VectorOps/VectorOps.h\n",
            "bedtools2/src/utils/BinTree/BinTree.cpp\n",
            "bedtools2/src/utils/BinTree/BinTree.h\n",
            "bedtools2/src/utils/BinTree/Makefile.frag\n",
            "bedtools2/src/utils/FileRecordTools/Records/\n",
            "bedtools2/src/utils/FileRecordTools/FileRecordMergeMgr.h\n",
            "bedtools2/src/utils/FileRecordTools/FileRecordMgr.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Makefile.frag\n",
            "bedtools2/src/utils/FileRecordTools/FileRecordMgr.h\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/\n",
            "bedtools2/src/utils/FileRecordTools/FileRecordMergeMgr.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/InputStreamMgr.h\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/BamFileReader.h\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/BufferedStreamMgr.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/SingleLineDelimTextFileReader.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/SingleLineDelimTransferBuffer.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/SingleLineDelimTransferBuffer.h\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/InputStreamMgr.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/FileReader.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/BamFileReader.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/Makefile.frag\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/SingleLineDelimTextFileReader.h\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/FileReader.h\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/BufferedStreamMgr.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/BedGraphInterval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordMgr.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordList.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed12Interval.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/GffRecord.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed4Interval.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/BedGraphInterval.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordList.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/BamRecord.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordKeyVector.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/PlusFields.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/StrandQueue.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed5Interval.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Record.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/GffRecord.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed4Interval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/VcfRecord.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed3Interval.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordKeyList.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/VcfRecord.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/EmptyRecord.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed3Interval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordKeyVector.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed6Interval.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/BlockMgr.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordMgr.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed6Interval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/GffPlusRecord.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/GffPlusRecord.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/BedPlusInterval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/Makefile.frag\n",
            "bedtools2/src/utils/FileRecordTools/Records/StrandQueue.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Record.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed5Interval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/NoPosPlusRecord.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed12Interval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/BlockMgr.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordKeyList.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/PlusFields.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/BamRecord.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/NoPosPlusRecord.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/EmptyRecord.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/BedPlusInterval.h\n",
            "bedtools2/src/utils/chromsweep/chromsweep.cpp\n",
            "bedtools2/src/utils/chromsweep/Makefile.frag\n",
            "bedtools2/src/utils/chromsweep/chromsweep.h\n",
            "bedtools2/src/utils/bedFilePE/bedFilePE.cpp\n",
            "bedtools2/src/utils/bedFilePE/Makefile.frag\n",
            "bedtools2/src/utils/bedFilePE/bedFilePE.h\n",
            "bedtools2/src/utils/lineFileUtilities/lineFileUtilities.h\n",
            "bedtools2/src/utils/ToolBase/ToolBase.cpp\n",
            "bedtools2/src/utils/ToolBase/Makefile.frag\n",
            "bedtools2/src/utils/ToolBase/ToolBase.h\n",
            "bedtools2/src/utils/version/version.h\n",
            "bedtools2/src/utils/version/version.cpp\n",
            "bedtools2/src/utils/version/version_release.txt\n",
            "bedtools2/src/utils/version/Makefile.frag\n",
            "bedtools2/src/utils/GenomeFile/NewGenomeFile.cpp\n",
            "bedtools2/src/utils/GenomeFile/NewGenomeFile.h\n",
            "bedtools2/src/utils/GenomeFile/GenomeFile.cpp\n",
            "bedtools2/src/utils/GenomeFile/Makefile.frag\n",
            "bedtools2/src/utils/GenomeFile/GenomeFile.h\n",
            "bedtools2/src/utils/BamTools-Ancillary/BamAncillary.h\n",
            "bedtools2/src/utils/BamTools-Ancillary/BamAncillary.cpp\n",
            "bedtools2/src/utils/BamTools-Ancillary/Makefile.frag\n",
            "bedtools2/src/utils/KeyListOps/KeyListOps.h\n",
            "bedtools2/src/utils/KeyListOps/KeyListOps.cpp\n",
            "bedtools2/src/utils/KeyListOps/KeyListOpsMethods.cpp\n",
            "bedtools2/src/utils/KeyListOps/Makefile.frag\n",
            "bedtools2/src/utils/KeyListOps/KeyListOpsMethods.h\n",
            "bedtools2/src/utils/fileType/fileType.cpp\n",
            "bedtools2/src/utils/fileType/FileRecordTypeChecker.h\n",
            "bedtools2/src/utils/fileType/FileRecordTypeChecker.cpp\n",
            "bedtools2/src/utils/fileType/Makefile.frag\n",
            "bedtools2/src/utils/fileType/fileType.h\n",
            "bedtools2/src/utils/gzstream/gzstream.C\n",
            "bedtools2/src/utils/gzstream/README\n",
            "bedtools2/src/utils/gzstream/COPYING.LIB\n",
            "bedtools2/src/utils/gzstream/Makefile.frag\n",
            "bedtools2/src/utils/gzstream/version.txt\n",
            "bedtools2/src/utils/gzstream/gzstream.h\n",
            "bedtools2/src/utils/Fasta/Fasta.h\n",
            "bedtools2/src/utils/Fasta/split.cpp\n",
            "bedtools2/src/utils/Fasta/Fasta.cpp\n",
            "bedtools2/src/utils/Fasta/LargeFileSupport.h\n",
            "bedtools2/src/utils/Fasta/Makefile.frag\n",
            "bedtools2/src/utils/Fasta/split.h\n",
            "bedtools2/src/utils/sequenceUtilities/sequenceUtils.h\n",
            "bedtools2/src/utils/sequenceUtilities/Makefile.frag\n",
            "bedtools2/src/utils/sequenceUtilities/sequenceUtils.cpp\n",
            "bedtools2/src/utils/bedGraphFile/bedGraphFile.h\n",
            "bedtools2/src/utils/bedGraphFile/Makefile.frag\n",
            "bedtools2/src/utils/bedGraphFile/bedGraphFile.cpp\n",
            "bedtools2/src/utils/Point/Point.h\n",
            "bedtools2/src/utils/driver/BedtoolsDriver.h\n",
            "bedtools2/src/utils/driver/BedtoolsDriver.cpp\n",
            "bedtools2/src/utils/driver/Makefile.frag\n",
            "bedtools2/src/utils/general/QuickString.cpp\n",
            "bedtools2/src/utils/general/ParseTools.cpp\n",
            "bedtools2/src/utils/general/CommonHelp.h\n",
            "bedtools2/src/utils/general/FreeList.h\n",
            "bedtools2/src/utils/general/CompressionTools.cpp\n",
            "bedtools2/src/utils/general/Random.cpp\n",
            "bedtools2/src/utils/general/BedtoolsTypes.h\n",
            "bedtools2/src/utils/general/ErrorMsg.h\n",
            "bedtools2/src/utils/general/PushBackStreamBuf.h\n",
            "bedtools2/src/utils/general/Tokenizer.h\n",
            "bedtools2/src/utils/general/ParseTools.h\n",
            "bedtools2/src/utils/general/QuickString.h\n",
            "bedtools2/src/utils/general/Tokenizer.cpp\n",
            "bedtools2/src/utils/general/ErrorMsg.cpp\n",
            "bedtools2/src/utils/general/CommonHelp.cpp\n",
            "bedtools2/src/utils/general/Makefile.frag\n",
            "bedtools2/src/utils/general/BTlist.h\n",
            "bedtools2/src/utils/general/CompressionTools.h\n",
            "bedtools2/src/utils/general/PushBackStreamBuf.cpp\n",
            "bedtools2/src/utils/general/InflateStreamBuf.h\n",
            "bedtools2/src/utils/general/Random.h\n",
            "bedtools2/src/utils/BlockedIntervals/BlockedIntervals.cpp\n",
            "bedtools2/src/utils/BlockedIntervals/BlockedIntervals.h\n",
            "bedtools2/src/utils/BlockedIntervals/Makefile.frag\n",
            "bedtools2/src/utils/BamTools/mapping/\n",
            "bedtools2/src/utils/BamTools/include/\n",
            "bedtools2/src/utils/BamTools/.gitignore\n",
            "bedtools2/src/utils/BamTools/Makefile.frag\n",
            "bedtools2/src/utils/BamTools/include/BamAncillary.h\n",
            "bedtools2/src/utils/BamTools/include/shared/\n",
            "bedtools2/src/utils/BamTools/include/SamHeader.hpp\n",
            "bedtools2/src/utils/BamTools/include/BamReader.hpp\n",
            "bedtools2/src/utils/BamTools/include/api/\n",
            "bedtools2/src/utils/BamTools/include/BamWriter.hpp\n",
            "bedtools2/src/utils/BamTools/include/BamAlignment.mapping.hpp\n",
            "bedtools2/src/utils/BamTools/include/BamAlignment.hpp\n",
            "bedtools2/src/utils/BamTools/include/api/BamWriter.h\n",
            "bedtools2/src/utils/BamTools/include/api/BamMultiReader.h\n",
            "bedtools2/src/utils/BamTools/include/api/SamConstants.h\n",
            "bedtools2/src/utils/BamTools/include/api/BamAlignment.h\n",
            "bedtools2/src/utils/BamTools/include/api/BamReader.h\n",
            "bedtools2/src/utils/BamTools/include/api/BamAux.h\n",
            "bedtools2/src/utils/BamTools/include/shared/bamtools_global.h\n",
            "bedtools2/src/utils/BamTools/mapping/BamAlignment.map\n",
            "bedtools2/src/utils/BamTools/mapping/BamAlignment.py\n",
            "bedtools2/src/utils/Contexts/ContextBase.h\n",
            "bedtools2/src/utils/Contexts/ContextClosest.h\n",
            "bedtools2/src/utils/Contexts/ContextIntersect.h\n",
            "bedtools2/src/utils/Contexts/ContextIntersect.cpp\n",
            "bedtools2/src/utils/Contexts/ContextFisher.cpp\n",
            "bedtools2/src/utils/Contexts/ContextMerge.h\n",
            "bedtools2/src/utils/Contexts/ContextGroupBy.h\n",
            "bedtools2/src/utils/Contexts/ContextComplement.h\n",
            "bedtools2/src/utils/Contexts/ContextSubtract.h\n",
            "bedtools2/src/utils/Contexts/ContextCoverage.cpp\n",
            "bedtools2/src/utils/Contexts/ContextBase.cpp\n",
            "bedtools2/src/utils/Contexts/ContextSummary.h\n",
            "bedtools2/src/utils/Contexts/ContextFisher.h\n",
            "bedtools2/src/utils/Contexts/ContextSpacing.cpp\n",
            "bedtools2/src/utils/Contexts/ContextMap.h\n",
            "bedtools2/src/utils/Contexts/ContextSubtract.cpp\n",
            "bedtools2/src/utils/Contexts/ContextJaccard.h\n",
            "bedtools2/src/utils/Contexts/ContextComplement.cpp\n",
            "bedtools2/src/utils/Contexts/ContextCoverage.h\n",
            "bedtools2/src/utils/Contexts/ContextSample.cpp\n",
            "bedtools2/src/utils/Contexts/ContextClosest.cpp\n",
            "bedtools2/src/utils/Contexts/Makefile.frag\n",
            "bedtools2/src/utils/Contexts/ContextJaccard.cpp\n",
            "bedtools2/src/utils/Contexts/ContextSample.h\n",
            "bedtools2/src/utils/Contexts/ContextSummary.cpp\n",
            "bedtools2/src/utils/Contexts/ContextGroupBy.cpp\n",
            "bedtools2/src/utils/Contexts/ContextMap.cpp\n",
            "bedtools2/src/utils/Contexts/ContextSpacing.h\n",
            "bedtools2/src/utils/Contexts/ContextMerge.cpp\n",
            "bedtools2/src/utils/bedFile/bedFile.h\n",
            "bedtools2/src/utils/bedFile/Makefile.frag\n",
            "bedtools2/src/utils/bedFile/bedFile.cpp\n",
            "bedtools2/src/utils/RecordOutputMgr/RecordOutputMgr.h\n",
            "bedtools2/src/utils/RecordOutputMgr/Makefile.frag\n",
            "bedtools2/src/utils/RecordOutputMgr/RecordOutputMgr.cpp\n",
            "bedtools2/src/utils/NewChromsweep/NewChromsweep.h\n",
            "bedtools2/src/utils/NewChromsweep/CloseSweep.h\n",
            "bedtools2/src/utils/NewChromsweep/Makefile.frag\n",
            "bedtools2/src/utils/NewChromsweep/NewChromsweep.cpp\n",
            "bedtools2/src/utils/NewChromsweep/CloseSweep.cpp\n",
            "bedtools2/src/utils/htslib/htslib.mk\n",
            "bedtools2/src/utils/htslib/cram/\n",
            "bedtools2/src/utils/htslib/htslib/\n",
            "bedtools2/src/utils/htslib/faidx.c\n",
            "bedtools2/src/utils/htslib/vcfutils.c\n",
            "bedtools2/src/utils/htslib/bcf_sr_sort.h\n",
            "bedtools2/src/utils/htslib/faidx.5\n",
            "bedtools2/src/utils/htslib/regidx.c\n",
            "bedtools2/src/utils/htslib/htslib.pc.in\n",
            "bedtools2/src/utils/htslib/configure.ac\n",
            "bedtools2/src/utils/htslib/INSTALL\n",
            "bedtools2/src/utils/htslib/hts_os.c\n",
            "bedtools2/src/utils/htslib/.DS_Store\n",
            "bedtools2/src/utils/htslib/tbx.c\n",
            "bedtools2/src/utils/htslib/probaln.c\n",
            "bedtools2/src/utils/htslib/LICENSE\n",
            "bedtools2/src/utils/htslib/kfunc.c\n",
            "bedtools2/src/utils/htslib/test/\n",
            "bedtools2/src/utils/htslib/hfile_gcs.c\n",
            "bedtools2/src/utils/htslib/errmod.c\n",
            "bedtools2/src/utils/htslib/multipart.c\n",
            "bedtools2/src/utils/htslib/hfile_internal.h\n",
            "bedtools2/src/utils/htslib/Makefile\n",
            "bedtools2/src/utils/htslib/htsfile.c\n",
            "bedtools2/src/utils/htslib/realn.c\n",
            "bedtools2/src/utils/htslib/bgzip.c\n",
            "bedtools2/src/utils/htslib/hfile_net.c\n",
            "bedtools2/src/utils/htslib/tabix.1\n",
            "bedtools2/src/utils/htslib/textutils.c\n",
            "bedtools2/src/utils/htslib/hts.c\n",
            "bedtools2/src/utils/htslib/README\n",
            "bedtools2/src/utils/htslib/vcf_sweep.c\n",
            "bedtools2/src/utils/htslib/README.md\n",
            "bedtools2/src/utils/htslib/sam.5\n",
            "bedtools2/src/utils/htslib/bcf_sr_sort.c\n",
            "bedtools2/src/utils/htslib/synced_bcf_reader.c\n",
            "bedtools2/src/utils/htslib/htslib_vars.mk\n",
            "bedtools2/src/utils/htslib/sam.c\n",
            "bedtools2/src/utils/htslib/NEWS\n",
            "bedtools2/src/utils/htslib/.gitignore\n",
            "bedtools2/src/utils/htslib/vcf.5\n",
            "bedtools2/src/utils/htslib/bgzf.c\n",
            "bedtools2/src/utils/htslib/.gitattributes\n",
            "bedtools2/src/utils/htslib/hts_internal.h\n",
            "bedtools2/src/utils/htslib/vcf.c\n",
            "bedtools2/src/utils/htslib/m4/\n",
            "bedtools2/src/utils/htslib/plugin.c\n",
            "bedtools2/src/utils/htslib/version.sh\n",
            "bedtools2/src/utils/htslib/hfile_libcurl.c\n",
            "bedtools2/src/utils/htslib/kstring.c\n",
            "bedtools2/src/utils/htslib/os/\n",
            "bedtools2/src/utils/htslib/md5.c\n",
            "bedtools2/src/utils/htslib/hfile_s3.c\n",
            "bedtools2/src/utils/htslib/htsfile.1\n",
            "bedtools2/src/utils/htslib/thread_pool.c\n",
            "bedtools2/src/utils/htslib/hfile.c\n",
            "bedtools2/src/utils/htslib/config.mk.in\n",
            "bedtools2/src/utils/htslib/tabix.c\n",
            "bedtools2/src/utils/htslib/textutils_internal.h\n",
            "bedtools2/src/utils/htslib/knetfile.c\n",
            "bedtools2/src/utils/htslib/bgzip.1\n",
            "bedtools2/src/utils/htslib/thread_pool_internal.h\n",
            "bedtools2/src/utils/htslib/os/rand.c\n",
            "bedtools2/src/utils/htslib/os/lzma_stub.h\n",
            "bedtools2/src/utils/htslib/m4/hts_prog_cc_warnings.m4\n",
            "bedtools2/src/utils/htslib/test/ce.fa\n",
            "bedtools2/src/utils/htslib/test/c2.fa.fai\n",
            "bedtools2/src/utils/htslib/test/c2#pad.sam\n",
            "bedtools2/src/utils/htslib/test/tabix/\n",
            "bedtools2/src/utils/htslib/test/range.bam.bai\n",
            "bedtools2/src/utils/htslib/test/c2.fa\n",
            "bedtools2/src/utils/htslib/test/test-regidx.c\n",
            "bedtools2/src/utils/htslib/test/realn02_exp.sam\n",
            "bedtools2/src/utils/htslib/test/realn02.fa\n",
            "bedtools2/src/utils/htslib/test/test-bcf-sr\n",
            "bedtools2/src/utils/htslib/test/test-regidx\n",
            "bedtools2/src/utils/htslib/test/auxf#values_java.cram\n",
            "bedtools2/src/utils/htslib/test/ce#supp.sam\n",
            "bedtools2/src/utils/htslib/test/ce#5.sam\n",
            "bedtools2/src/utils/htslib/test/ce#1.sam\n",
            "bedtools2/src/utils/htslib/test/thrash_threads6.c\n",
            "bedtools2/src/utils/htslib/test/c1#pad3.sam\n",
            "bedtools2/src/utils/htslib/test/xx#tlen2.sam\n",
            "bedtools2/src/utils/htslib/test/ce#large_seq.sam\n",
            "bedtools2/src/utils/htslib/test/.DS_Store\n",
            "bedtools2/src/utils/htslib/test/test-logging.pl\n",
            "bedtools2/src/utils/htslib/test/realn02.sam\n",
            "bedtools2/src/utils/htslib/test/c1#pad2.sam\n",
            "bedtools2/src/utils/htslib/test/test-bcf-sr.c\n",
            "bedtools2/src/utils/htslib/test/ce#5b.sam\n",
            "bedtools2/src/utils/htslib/test/ce#2.sam\n",
            "bedtools2/src/utils/htslib/test/xx#unsorted.sam\n",
            "bedtools2/src/utils/htslib/test/sam\n",
            "bedtools2/src/utils/htslib/test/ce#tag_depadded.sam\n",
            "bedtools2/src/utils/htslib/test/test-vcf-api.c\n",
            "bedtools2/src/utils/htslib/test/test_view\n",
            "bedtools2/src/utils/htslib/test/thrash_threads2.c\n",
            "bedtools2/src/utils/htslib/test/realn01.sam\n",
            "bedtools2/src/utils/htslib/test/c1#pad1.sam\n",
            "bedtools2/src/utils/htslib/test/cross_validate.sh\n",
            "bedtools2/src/utils/htslib/test/test-vcf-sweep.c\n",
            "bedtools2/src/utils/htslib/test/fieldarith\n",
            "bedtools2/src/utils/htslib/test/md#1.sam\n",
            "bedtools2/src/utils/htslib/test/realn01_exp.sam\n",
            "bedtools2/src/utils/htslib/test/compare_sam.pl\n",
            "bedtools2/src/utils/htslib/test/xx#minimal.sam\n",
            "bedtools2/src/utils/htslib/test/realn01.fa.fai\n",
            "bedtools2/src/utils/htslib/test/realn01_exp-a.sam\n",
            "bedtools2/src/utils/htslib/test/xx#repeated.sam\n",
            "bedtools2/src/utils/htslib/test/test_realn\n",
            "bedtools2/src/utils/htslib/test/xx#tlen.sam\n",
            "bedtools2/src/utils/htslib/test/xx#pair.sam\n",
            "bedtools2/src/utils/htslib/test/fieldarith.c\n",
            "bedtools2/src/utils/htslib/test/xx#large_aux2.sam\n",
            "bedtools2/src/utils/htslib/test/bgziptest.txt\n",
            "bedtools2/src/utils/htslib/test/auxf.fa\n",
            "bedtools2/src/utils/htslib/test/bgziptest.txt.gz.gzi\n",
            "bedtools2/src/utils/htslib/test/thrash_threads5.c\n",
            "bedtools2/src/utils/htslib/test/realn02-r.sam\n",
            "bedtools2/src/utils/htslib/test/hts_endian.c\n",
            "bedtools2/src/utils/htslib/test/fieldarith.sam\n",
            "bedtools2/src/utils/htslib/test/auxf#values.sam\n",
            "bedtools2/src/utils/htslib/test/hfile\n",
            "bedtools2/src/utils/htslib/test/realn01_exp-e.sam\n",
            "bedtools2/src/utils/htslib/test/c1#clip.sam\n",
            "bedtools2/src/utils/htslib/test/thrash_threads1.c\n",
            "bedtools2/src/utils/htslib/test/test.pl\n",
            "bedtools2/src/utils/htslib/test/test_realn.c\n",
            "bedtools2/src/utils/htslib/test/test_bgzf\n",
            "bedtools2/src/utils/htslib/test/noroundtrip-out.vcf\n",
            "bedtools2/src/utils/htslib/test/bgziptest.txt.gz\n",
            "bedtools2/src/utils/htslib/test/realn02_exp-a.sam\n",
            "bedtools2/src/utils/htslib/test/test-vcf-sweep\n",
            "bedtools2/src/utils/htslib/test/formatmissing.vcf\n",
            "bedtools2/src/utils/htslib/test/xx#large_aux.sam\n",
            "bedtools2/src/utils/htslib/test/hts_endian\n",
            "bedtools2/src/utils/htslib/test/range.out\n",
            "bedtools2/src/utils/htslib/test/test-bcf-translate\n",
            "bedtools2/src/utils/htslib/test/auxf.fa.fai\n",
            "bedtools2/src/utils/htslib/test/fastqs.fq\n",
            "bedtools2/src/utils/htslib/test/formatcols.vcf\n",
            "bedtools2/src/utils/htslib/test/test-vcf-api.out\n",
            "bedtools2/src/utils/htslib/test/test-bcf-sr.pl\n",
            "bedtools2/src/utils/htslib/test/xx.fa.fai\n",
            "bedtools2/src/utils/htslib/test/sam.c\n",
            "bedtools2/src/utils/htslib/test/test-bcf-translate.c\n",
            "bedtools2/src/utils/htslib/test/xx#blank.sam\n",
            "bedtools2/src/utils/htslib/test/test_view.c\n",
            "bedtools2/src/utils/htslib/test/thrash_threads4.c\n",
            "bedtools2/src/utils/htslib/test/c1.fa.fai\n",
            "bedtools2/src/utils/htslib/test/test-vcf-sweep.out\n",
            "bedtools2/src/utils/htslib/test/test-bcf-translate.out\n",
            "bedtools2/src/utils/htslib/test/xx#large_aux_java.cram\n",
            "bedtools2/src/utils/htslib/test/ce#5b_java.cram\n",
            "bedtools2/src/utils/htslib/test/c1#bounds.sam\n",
            "bedtools2/src/utils/htslib/test/realn02_exp-e.sam\n",
            "bedtools2/src/utils/htslib/test/faidx.fa\n",
            "bedtools2/src/utils/htslib/test/test-vcf-hdr-in.vcf\n",
            "bedtools2/src/utils/htslib/test/c1#noseq.sam\n",
            "bedtools2/src/utils/htslib/test/test-vcf-hdr.out\n",
            "bedtools2/src/utils/htslib/test/c1#unknown.sam\n",
            "bedtools2/src/utils/htslib/test/ce#tag_padded.sam\n",
            "bedtools2/src/utils/htslib/test/formatmissing-out.vcf\n",
            "bedtools2/src/utils/htslib/test/ce#unmap.sam\n",
            "bedtools2/src/utils/htslib/test/xx#rg.sam\n",
            "bedtools2/src/utils/htslib/test/range.cram\n",
            "bedtools2/src/utils/htslib/test/md.fa\n",
            "bedtools2/src/utils/htslib/test/xx#triplet.sam\n",
            "bedtools2/src/utils/htslib/test/xx.fa\n",
            "bedtools2/src/utils/htslib/test/ce#unmap1.sam\n",
            "bedtools2/src/utils/htslib/test/realn02.fa.fai\n",
            "bedtools2/src/utils/htslib/test/range.bam\n",
            "bedtools2/src/utils/htslib/test/fastqs_README.txt\n",
            "bedtools2/src/utils/htslib/test/noroundtrip.vcf\n",
            "bedtools2/src/utils/htslib/test/hfile.c\n",
            "bedtools2/src/utils/htslib/test/c1.fa\n",
            "bedtools2/src/utils/htslib/test/thread_pool.md\n",
            "bedtools2/src/utils/htslib/test/ce#1000.sam\n",
            "bedtools2/src/utils/htslib/test/ce.fa.fai\n",
            "bedtools2/src/utils/htslib/test/test_bgzf.c\n",
            "bedtools2/src/utils/htslib/test/ce#unmap2.sam\n",
            "bedtools2/src/utils/htslib/test/realn01.fa\n",
            "bedtools2/src/utils/htslib/test/range.cram.crai\n",
            "bedtools2/src/utils/htslib/test/test-vcf-api\n",
            "bedtools2/src/utils/htslib/test/thrash_threads3.c\n",
            "bedtools2/src/utils/htslib/test/md.fa.fai\n",
            "bedtools2/src/utils/htslib/test/tabix/tabix.tst\n",
            "bedtools2/src/utils/htslib/test/tabix/vcf_file.2.3199812.out\n",
            "bedtools2/src/utils/htslib/test/tabix/test-tabix.sh\n",
            "bedtools2/src/utils/htslib/test/tabix/large_chr.20.1.2147483647.out\n",
            "bedtools2/src/utils/htslib/test/tabix/bed_file.Y.100200.out\n",
            "bedtools2/src/utils/htslib/test/tabix/large_chr.vcf\n",
            "bedtools2/src/utils/htslib/test/tabix/vcf_file.vcf\n",
            "bedtools2/src/utils/htslib/test/tabix/gff_file.X.2934832.2935190.out\n",
            "bedtools2/src/utils/htslib/test/tabix/vcf_file.1.3000151.out\n",
            "bedtools2/src/utils/htslib/test/tabix/gff_file.gff\n",
            "bedtools2/src/utils/htslib/test/tabix/bed_file.bed\n",
            "bedtools2/src/utils/htslib/htslib/sam.h\n",
            "bedtools2/src/utils/htslib/htslib/synced_bcf_reader.h\n",
            "bedtools2/src/utils/htslib/htslib/vcf_sweep.h\n",
            "bedtools2/src/utils/htslib/htslib/vcf.h\n",
            "bedtools2/src/utils/htslib/htslib/khash_str2int.h\n",
            "bedtools2/src/utils/htslib/htslib/bgzf.h\n",
            "bedtools2/src/utils/htslib/htslib/cram.h\n",
            "bedtools2/src/utils/htslib/htslib/kstring.h\n",
            "bedtools2/src/utils/htslib/htslib/knetfile.h\n",
            "bedtools2/src/utils/htslib/htslib/hts_log.h\n",
            "bedtools2/src/utils/htslib/htslib/hfile.h\n",
            "bedtools2/src/utils/htslib/htslib/thread_pool.h\n",
            "bedtools2/src/utils/htslib/htslib/hts_defs.h\n",
            "bedtools2/src/utils/htslib/htslib/khash.h\n",
            "bedtools2/src/utils/htslib/htslib/regidx.h\n",
            "bedtools2/src/utils/htslib/htslib/faidx.h\n",
            "bedtools2/src/utils/htslib/htslib/vcfutils.h\n",
            "bedtools2/src/utils/htslib/htslib/kseq.h\n",
            "bedtools2/src/utils/htslib/htslib/kfunc.h\n",
            "bedtools2/src/utils/htslib/htslib/tbx.h\n",
            "bedtools2/src/utils/htslib/htslib/ksort.h\n",
            "bedtools2/src/utils/htslib/htslib/kbitset.h\n",
            "bedtools2/src/utils/htslib/htslib/hts_os.h\n",
            "bedtools2/src/utils/htslib/htslib/klist.h\n",
            "bedtools2/src/utils/htslib/htslib/hts.h\n",
            "bedtools2/src/utils/htslib/htslib/hts_endian.h\n",
            "bedtools2/src/utils/htslib/cram/cram_codecs.h\n",
            "bedtools2/src/utils/htslib/cram/sam_header.c\n",
            "bedtools2/src/utils/htslib/cram/cram_index.c\n",
            "bedtools2/src/utils/htslib/cram/open_trace_file.c\n",
            "bedtools2/src/utils/htslib/cram/mFILE.h\n",
            "bedtools2/src/utils/htslib/cram/cram_external.c\n",
            "bedtools2/src/utils/htslib/cram/cram.h\n",
            "bedtools2/src/utils/htslib/cram/misc.h\n",
            "bedtools2/src/utils/htslib/cram/files.c\n",
            "bedtools2/src/utils/htslib/cram/os.h\n",
            "bedtools2/src/utils/htslib/cram/rANS_static.h\n",
            "bedtools2/src/utils/htslib/cram/cram_samtools.c\n",
            "bedtools2/src/utils/htslib/cram/pooled_alloc.c\n",
            "bedtools2/src/utils/htslib/cram/string_alloc.c\n",
            "bedtools2/src/utils/htslib/cram/cram_io.h\n",
            "bedtools2/src/utils/htslib/cram/cram_decode.h\n",
            "bedtools2/src/utils/htslib/cram/cram_encode.h\n",
            "bedtools2/src/utils/htslib/cram/cram_stats.h\n",
            "bedtools2/src/utils/htslib/cram/mFILE.c\n",
            "bedtools2/src/utils/htslib/cram/cram_index.h\n",
            "bedtools2/src/utils/htslib/cram/sam_header.h\n",
            "bedtools2/src/utils/htslib/cram/open_trace_file.h\n",
            "bedtools2/src/utils/htslib/cram/cram_codecs.c\n",
            "bedtools2/src/utils/htslib/cram/rANS_byte.h\n",
            "bedtools2/src/utils/htslib/cram/string_alloc.h\n",
            "bedtools2/src/utils/htslib/cram/rANS_static.c\n",
            "bedtools2/src/utils/htslib/cram/pooled_alloc.h\n",
            "bedtools2/src/utils/htslib/cram/cram_samtools.h\n",
            "bedtools2/src/utils/htslib/cram/cram_stats.c\n",
            "bedtools2/src/utils/htslib/cram/cram_encode.c\n",
            "bedtools2/src/utils/htslib/cram/cram_structs.h\n",
            "bedtools2/src/utils/htslib/cram/cram_io.c\n",
            "bedtools2/src/utils/htslib/cram/cram_decode.c\n",
            "bedtools2/src/utils/tabFile/tabFile.cpp\n",
            "bedtools2/src/utils/tabFile/tabFile.h\n",
            "bedtools2/src/utils/tabFile/Makefile.frag\n",
            "bedtools2/src/genomeCoverageBed/genomeCoverageMain.cpp\n",
            "bedtools2/src/genomeCoverageBed/genomeCoverageBed.h\n",
            "bedtools2/src/genomeCoverageBed/Makefile.frag\n",
            "bedtools2/src/genomeCoverageBed/genomeCoverageBed.cpp\n",
            "bedtools2/src/maskFastaFromBed/maskFastaFromBed.cpp\n",
            "bedtools2/src/maskFastaFromBed/maskFastaFromBed.h\n",
            "bedtools2/src/maskFastaFromBed/maskFastaFromBedMain.cpp\n",
            "bedtools2/src/maskFastaFromBed/Makefile.frag\n",
            "bedtools2/src/regressTest/RegressTest.h\n",
            "bedtools2/src/regressTest/RegressTest.cpp\n",
            "bedtools2/src/regressTest/regressTestMain.cpp\n",
            "bedtools2/src/regressTest/Makefile.frag\n",
            "bedtools2/src/bedToIgv/bedToIgv.cpp\n",
            "bedtools2/src/bedToIgv/Makefile.frag\n",
            "bedtools2/src/nucBed/nucBed.h\n",
            "bedtools2/src/nucBed/nucBedMain.cpp\n",
            "bedtools2/src/nucBed/LargeFileSupport.h\n",
            "bedtools2/src/nucBed/Makefile.frag\n",
            "bedtools2/src/nucBed/nucBed.cpp\n",
            "bedtools2/src/reldist/reldist.cpp\n",
            "bedtools2/src/reldist/Makefile.frag\n",
            "bedtools2/src/reldist/reldistMain.cpp\n",
            "bedtools2/src/reldist/reldist.h\n",
            "bedtools2/src/getOverlap/getOverlap.cpp\n",
            "bedtools2/src/getOverlap/Makefile.frag\n",
            "bedtools2/src/randomBed/randomBedMain.cpp\n",
            "bedtools2/src/randomBed/randomBed.cpp\n",
            "bedtools2/src/randomBed/Makefile.frag\n",
            "bedtools2/src/randomBed/randomBed.h\n",
            "bedtools2/src/annotateBed/annotateBed.cpp\n",
            "bedtools2/src/annotateBed/annotateBed.h\n",
            "bedtools2/src/annotateBed/annotateMain.cpp\n",
            "bedtools2/src/annotateBed/Makefile.frag\n",
            "bedtools2/src/slopBed/slopBed.h\n",
            "bedtools2/src/slopBed/slopBed.cpp\n",
            "bedtools2/src/slopBed/Makefile.frag\n",
            "bedtools2/src/slopBed/slopBedMain.cpp\n",
            "bedtools2/src/mapFile/mapHelp.cpp\n",
            "bedtools2/src/mapFile/mapFile.cpp\n",
            "bedtools2/src/mapFile/Makefile.frag\n",
            "bedtools2/src/mapFile/mapFile.h\n",
            "bedtools2/src/pairToPair/pairToPair.h\n",
            "bedtools2/src/pairToPair/pairToPairMain.cpp\n",
            "bedtools2/src/pairToPair/pairToPair.cpp\n",
            "bedtools2/src/pairToPair/Makefile.frag\n",
            "bedtools2/src/linksBed/linksBed.cpp\n",
            "bedtools2/src/linksBed/Makefile.frag\n",
            "bedtools2/src/linksBed/linksBed.h\n",
            "bedtools2/src/linksBed/linksMain.cpp\n",
            "bedtools2/src/fisher/fisher.h\n",
            "bedtools2/src/fisher/kfunc.h\n",
            "bedtools2/src/fisher/kfunc.cpp\n",
            "bedtools2/src/fisher/Makefile.frag\n",
            "bedtools2/src/fisher/fisher.cpp\n",
            "bedtools2/src/fisher/fisherHelp.cpp\n",
            "bedtools2/src/expand/Makefile.frag\n",
            "bedtools2/src/expand/expand.cpp\n",
            "bedtools2/src/complementFile/complementHelp.cpp\n",
            "bedtools2/src/complementFile/complementFile.h\n",
            "bedtools2/src/complementFile/Makefile.frag\n",
            "bedtools2/src/complementFile/complementFile.cpp\n",
            "bedtools2/src/bedToBam/bedToBam.cpp\n",
            "bedtools2/src/bedToBam/Makefile.frag\n",
            "bedtools2/src/split/splitBed.h\n",
            "bedtools2/src/split/splitBed.cpp\n",
            "bedtools2/src/split/splitBedMain.cpp\n",
            "bedtools2/src/split/Makefile.frag\n",
            "bedtools2/src/sortBed/sortMain.cpp\n",
            "bedtools2/src/sortBed/sortBed.cpp\n",
            "bedtools2/src/sortBed/sortBed.h\n",
            "bedtools2/src/sortBed/Makefile.frag\n",
            "bedtools2/src/spacingFile/spacingHelp.cpp\n",
            "bedtools2/src/spacingFile/spacingFile.h\n",
            "bedtools2/src/spacingFile/Makefile.frag\n",
            "bedtools2/src/spacingFile/spacingFile.cpp\n",
            "bedtools2/data/gerp.chr1.bed.gz\n",
            "bedtools2/data/simpleRepeats.chr1.bed.gz\n",
            "bedtools2/data/aluY.chr1.bed.gz\n",
            "bedtools2/data/refseq.chr1.exons.bed.gz\n",
            "bedtools2/data/knownGene.hg18.chr21.short.bed\n",
            "bedtools2/data/knownGene.hg18.chr21.bed\n",
            "bedtools2/scripts/makeBashScripts.py\n",
            "bedtools2/scripts/vcfsort2.sh\n",
            "bedtools2/scripts/vcfsort.sh\n",
            "bedtools2/genomes/human.hg38.genome\n",
            "bedtools2/genomes/mouse.mm10.genome\n",
            "bedtools2/genomes/mouse.mm8.genome\n",
            "bedtools2/genomes/human.hg18.genome\n",
            "bedtools2/genomes/human.hg19.genome\n",
            "bedtools2/genomes/mouse.mm9.genome\n",
            "bedtools2/test/split/\n",
            "bedtools2/test/bedtobam/\n",
            "bedtools2/test/cluster/\n",
            "bedtools2/test/expand/\n",
            "bedtools2/test/shuffle/\n",
            "bedtools2/test/fisher/\n",
            "bedtools2/test/complement/\n",
            "bedtools2/test/.DS_Store\n",
            "bedtools2/test/intersect/\n",
            "bedtools2/test/sort/\n",
            "bedtools2/test/subtract/\n",
            "bedtools2/test/merge/\n",
            "bedtools2/test/general/\n",
            "bedtools2/test/slop/\n",
            "bedtools2/test/reldist/\n",
            "bedtools2/test/shift/\n",
            "bedtools2/test/genomecov/\n",
            "bedtools2/test/jaccard/\n",
            "bedtools2/test/groupby/\n",
            "bedtools2/test/bamtofastq/\n",
            "bedtools2/test/bed12tobed6/\n",
            "bedtools2/test/getfasta/\n",
            "bedtools2/test/map/\n",
            "bedtools2/test/bigchroms/\n",
            "bedtools2/test/spacing/\n",
            "bedtools2/test/sample/\n",
            "bedtools2/test/flank/\n",
            "bedtools2/test/negativecontrol/\n",
            "bedtools2/test/multicov/\n",
            "bedtools2/test/closest/\n",
            "bedtools2/test/test.sh\n",
            "bedtools2/test/coverage/\n",
            "bedtools2/test/bamtobed/\n",
            "bedtools2/test/makewindows/\n",
            "bedtools2/test/makewindows/test-makewindows.sh\n",
            "bedtools2/test/makewindows/a.33bp.bed\n",
            "bedtools2/test/makewindows/a.20bp.bed\n",
            "bedtools2/test/makewindows/input.bed\n",
            "bedtools2/test/makewindows/a.19bp.bed\n",
            "bedtools2/test/bamtobed/two_blocks.sam\n",
            "bedtools2/test/bamtobed/three_blocks.sam\n",
            "bedtools2/test/bamtobed/test-bamtobed.sh\n",
            "bedtools2/test/bamtobed/numeric_tag.sam\n",
            "bedtools2/test/bamtobed/extra-long-header.sam.gz\n",
            "bedtools2/test/bamtobed/two_blocks_w_D.sam\n",
            "bedtools2/test/bamtobed/one_block.sam\n",
            "bedtools2/test/coverage/two_blocks.sam\n",
            "bedtools2/test/coverage/chr_0-100.bed\n",
            "bedtools2/test/coverage/three_blocks.bam.bai\n",
            "bedtools2/test/coverage/three_blocks_match.bam.bai\n",
            "bedtools2/test/coverage/sam-w-del.sam\n",
            "bedtools2/test/coverage/three_blocks_nomatch.bed\n",
            "bedtools2/test/coverage/two_blocks.bam.bai\n",
            "bedtools2/test/coverage/three_blocks.sam\n",
            "bedtools2/test/coverage/one_block.bam.bai\n",
            "bedtools2/test/coverage/test-coverage.sh\n",
            "bedtools2/test/coverage/x.bed\n",
            "bedtools2/test/coverage/y.bed\n",
            "bedtools2/test/coverage/three_blocks_nomatch.bed12\n",
            "bedtools2/test/coverage/a.bed\n",
            "bedtools2/test/coverage/c.bed\n",
            "bedtools2/test/coverage/three_blocks_match.bam\n",
            "bedtools2/test/coverage/sam-w-del.bam.bai\n",
            "bedtools2/test/coverage/b.bed\n",
            "bedtools2/test/coverage/three_blocks_match.bed\n",
            "bedtools2/test/coverage/one_block.sam\n",
            "bedtools2/test/coverage/three_blocks_match_1bp.bed\n",
            "bedtools2/test/closest/d_d1F.bed\n",
            "bedtools2/test/closest/d_q1.bed\n",
            "bedtools2/test/closest/d_iu.bed\n",
            "bedtools2/test/closest/d_d1R.bed\n",
            "bedtools2/test/closest/d_q2.bed\n",
            "bedtools2/test/closest/tfbs.bed\n",
            "bedtools2/test/closest/null_b.bed\n",
            "bedtools2/test/closest/bug281_a.medium.bed\n",
            "bedtools2/test/closest/null_c.bed\n",
            "bedtools2/test/closest/null_a.bed\n",
            "bedtools2/test/closest/d_id.bed\n",
            "bedtools2/test/closest/a2.bed\n",
            "bedtools2/test/closest/test-closest.sh\n",
            "bedtools2/test/closest/kclosest/\n",
            "bedtools2/test/closest/close-a.bed\n",
            "bedtools2/test/closest/b.names.bed\n",
            "bedtools2/test/closest/close-b.bed\n",
            "bedtools2/test/closest/strand-test-d.bed\n",
            "bedtools2/test/closest/b-one-bp-closer.bed\n",
            "bedtools2/test/closest/islands.2.bed\n",
            "bedtools2/test/closest/bug157_b.bed\n",
            "bedtools2/test/closest/shores.bed\n",
            "bedtools2/test/closest/strand-test-a.bed\n",
            "bedtools2/test/closest/strand-test-c.bed\n",
            "bedtools2/test/closest/bug157_a.bed\n",
            "bedtools2/test/closest/strand-test-b.bed\n",
            "bedtools2/test/closest/bug281_b.bed\n",
            "bedtools2/test/closest/dmr.bed\n",
            "bedtools2/test/closest/a.bed\n",
            "bedtools2/test/closest/b.bed\n",
            "bedtools2/test/closest/a.names.bed\n",
            "bedtools2/test/closest/mdb3.bed\n",
            "bedtools2/test/closest/mdb2.bed\n",
            "bedtools2/test/closest/b2.bed\n",
            "bedtools2/test/closest/mdb1.bed\n",
            "bedtools2/test/closest/mq1.bed\n",
            "bedtools2/test/closest/d.bed\n",
            "bedtools2/test/closest/d4.bed\n",
            "bedtools2/test/closest/sortAndNaming/\n",
            "bedtools2/test/closest/islands.bed\n",
            "bedtools2/test/closest/bug281_b.flip.bed\n",
            "bedtools2/test/closest/bug281_a.flip.medium.bed\n",
            "bedtools2/test/closest/d_d2R.bed\n",
            "bedtools2/test/closest/d_d2F.bed\n",
            "bedtools2/test/closest/sortAndNaming/num_all.bed\n",
            "bedtools2/test/closest/sortAndNaming/db1_noChr.bed\n",
            "bedtools2/test/closest/sortAndNaming/alpha_all.bed\n",
            "bedtools2/test/closest/sortAndNaming/test-sort-and-naming.sh\n",
            "bedtools2/test/closest/sortAndNaming/q1_gls.bed\n",
            "bedtools2/test/closest/sortAndNaming/db1_num.bed\n",
            "bedtools2/test/closest/sortAndNaming/db2_numBackwards.bed\n",
            "bedtools2/test/closest/sortAndNaming/num_missing.bed\n",
            "bedtools2/test/closest/sortAndNaming/q1_num.bed\n",
            "bedtools2/test/closest/sortAndNaming/db2_num.bed\n",
            "bedtools2/test/closest/sortAndNaming/db1_leadingZero.txt\n",
            "bedtools2/test/closest/sortAndNaming/sq1.bed\n",
            "bedtools2/test/closest/sortAndNaming/db1_gls.bed\n",
            "bedtools2/test/closest/sortAndNaming/alpha_missing.bed\n",
            "bedtools2/test/closest/sortAndNaming/sdb1.bed\n",
            "bedtools2/test/closest/sortAndNaming/db3_numBackwards.bed\n",
            "bedtools2/test/closest/sortAndNaming/q1a_num.bed\n",
            "bedtools2/test/closest/kclosest/test-kclosest.sh\n",
            "bedtools2/test/closest/kclosest/bug471_b.bed\n",
            "bedtools2/test/closest/kclosest/bug471_a.bed\n",
            "bedtools2/test/closest/kclosest/d2.bed\n",
            "bedtools2/test/closest/kclosest/q2.bed\n",
            "bedtools2/test/closest/kclosest/d3.bed\n",
            "bedtools2/test/closest/kclosest/d1.bed\n",
            "bedtools2/test/closest/kclosest/q1.bed\n",
            "bedtools2/test/multicov/two_blocks.sam\n",
            "bedtools2/test/multicov/test-multicov.sh\n",
            "bedtools2/test/multicov/test-multi.bed\n",
            "bedtools2/test/multicov/test-multi.2.sam\n",
            "bedtools2/test/multicov/multicov.bed\n",
            "bedtools2/test/multicov/test-multi.sam\n",
            "bedtools2/test/multicov/one_block.sam\n",
            "bedtools2/test/negativecontrol/test-negativecontrol.sh\n",
            "bedtools2/test/flank/test-flank.sh\n",
            "bedtools2/test/flank/a.bed\n",
            "bedtools2/test/flank/tiny.genome\n",
            "bedtools2/test/sample/test-sample.sh\n",
            "bedtools2/test/sample/human.hg19.genome\n",
            "bedtools2/test/spacing/a.bed\n",
            "bedtools2/test/spacing/test-spacing.sh\n",
            "bedtools2/test/bigchroms/test-bigchroms.sh\n",
            "bedtools2/test/bigchroms/big4.bed\n",
            "bedtools2/test/bigchroms/make-big-chrom.py\n",
            "bedtools2/test/bigchroms/big4c.bed\n",
            "bedtools2/test/bigchroms/bigx.bed\n",
            "bedtools2/test/bigchroms/abig.bed\n",
            "bedtools2/test/map/values.bam\n",
            "bedtools2/test/map/.Rhistory\n",
            "bedtools2/test/map/a.vsorted.bed\n",
            "bedtools2/test/map/three_blocks_nomatch.bed\n",
            "bedtools2/test/map/b.vsorted.bed\n",
            "bedtools2/test/map/test.vcf\n",
            "bedtools2/test/map/values.bed\n",
            "bedtools2/test/map/test.gff\n",
            "bedtools2/test/map/values5.bed\n",
            "bedtools2/test/map/ivls.bed\n",
            "bedtools2/test/map/values4.bed\n",
            "bedtools2/test/map/t.bed\n",
            "bedtools2/test/map/ivls2.bed\n",
            "bedtools2/test/map/values3.bed\n",
            "bedtools2/test/map/.RData\n",
            "bedtools2/test/map/values2.bed\n",
            "bedtools2/test/map/d.bed\n",
            "bedtools2/test/map/genome\n",
            "bedtools2/test/map/three_blocks_match.bed\n",
            "bedtools2/test/map/test-map.sh\n",
            "bedtools2/test/map/bug262_a.bed\n",
            "bedtools2/test/map/bug262_b.bed\n",
            "bedtools2/test/map/fullFields.bam\n",
            "bedtools2/test/getfasta/test.iupac.bed\n",
            "bedtools2/test/getfasta/rna.fasta\n",
            "bedtools2/test/getfasta/t.fa.gz.fai\n",
            "bedtools2/test/getfasta/t.fa\n",
            "bedtools2/test/getfasta/t.fa.gz\n",
            "bedtools2/test/getfasta/test.iupac.fa\n",
            "bedtools2/test/getfasta/blocks.bed\n",
            "bedtools2/test/getfasta/t_fH.fa.fai\n",
            "bedtools2/test/getfasta/t.bed\n",
            "bedtools2/test/getfasta/t.fa.fai\n",
            "bedtools2/test/getfasta/b.bed\n",
            "bedtools2/test/getfasta/test-getfasta.sh\n",
            "bedtools2/test/getfasta/t_fH.fa\n",
            "bedtools2/test/getfasta/t.fa.gz.gzi\n",
            "bedtools2/test/getfasta/rna.fasta.fai\n",
            "bedtools2/test/getfasta/c\n",
            "bedtools2/test/getfasta/rna.bed\n",
            "bedtools2/test/bed12tobed6/two_blocks.bed\n",
            "bedtools2/test/bed12tobed6/bed6.bed\n",
            "bedtools2/test/bed12tobed6/three_blocks.bed\n",
            "bedtools2/test/bed12tobed6/test-bed12tobed6.sh\n",
            "bedtools2/test/bed12tobed6/one_blocks.bed\n",
            "bedtools2/test/bamtofastq/test-bamtofastq.sh\n",
            "bedtools2/test/bamtofastq/test2.sam\n",
            "bedtools2/test/bamtofastq/golden.fq\n",
            "bedtools2/test/bamtofastq/test.bam\n",
            "bedtools2/test/bamtofastq/golden.fq2\n",
            "bedtools2/test/bamtofastq/test.sam\n",
            "bedtools2/test/groupby/gdc.bam\n",
            "bedtools2/test/groupby/values3_case.header.bed\n",
            "bedtools2/test/groupby/test-groupby.sh\n",
            "bedtools2/test/groupby/values3.unmarked_header.bed\n",
            "bedtools2/test/groupby/a_vcfSVtest.vcf\n",
            "bedtools2/test/groupby/values3.7fields.header.bed\n",
            "bedtools2/test/groupby/values3.unmarked_header.bed.2\n",
            "bedtools2/test/groupby/values3.no_header.bed\n",
            "bedtools2/test/groupby/values3.header.bed\n",
            "bedtools2/test/groupby/noPosvalues.header.bed\n",
            "bedtools2/test/groupby/test.bed\n",
            "bedtools2/test/groupby/bug569_problem.txt\n",
            "bedtools2/test/jaccard/a645.bed\n",
            "bedtools2/test/jaccard/short.bed\n",
            "bedtools2/test/jaccard/bMixedStrands.bed\n",
            "bedtools2/test/jaccard/a.bed\n",
            "bedtools2/test/jaccard/c.bed\n",
            "bedtools2/test/jaccard/three_blocks_match.bam\n",
            "bedtools2/test/jaccard/b.bed\n",
            "bedtools2/test/jaccard/b645.bed\n",
            "bedtools2/test/jaccard/e.bed\n",
            "bedtools2/test/jaccard/test-jaccard.sh\n",
            "bedtools2/test/jaccard/long.bed\n",
            "bedtools2/test/jaccard/three_blocks_match.bed\n",
            "bedtools2/test/jaccard/aMixedStrands.bed\n",
            "bedtools2/test/jaccard/a.bam\n",
            "bedtools2/test/genomecov/two_blocks.sam\n",
            "bedtools2/test/genomecov/test_ref.fa.fai\n",
            "bedtools2/test/genomecov/test-genomecov.sh\n",
            "bedtools2/test/genomecov/chip.sam\n",
            "bedtools2/test/genomecov/y.bam\n",
            "bedtools2/test/genomecov/test_ref.fa\n",
            "bedtools2/test/genomecov/sam-w-del.sam\n",
            "bedtools2/test/genomecov/mk-deep.py\n",
            "bedtools2/test/genomecov/three_blocks_nomatch.bed\n",
            "bedtools2/test/genomecov/three_blocks.sam\n",
            "bedtools2/test/genomecov/empty.bam\n",
            "bedtools2/test/genomecov/y.bed\n",
            "bedtools2/test/genomecov/empty.cram\n",
            "bedtools2/test/genomecov/genome.txt\n",
            "bedtools2/test/genomecov/three_blocks_match.bed\n",
            "bedtools2/test/genomecov/pair-chip.sam\n",
            "bedtools2/test/genomecov/one_block.sam\n",
            "bedtools2/test/genomecov/three_blocks_match_1bp.bed\n",
            "bedtools2/test/shift/test-shift.sh\n",
            "bedtools2/test/shift/issue_807.bed\n",
            "bedtools2/test/shift/a.bed\n",
            "bedtools2/test/shift/b.bed\n",
            "bedtools2/test/shift/tiny.genome\n",
            "bedtools2/test/shift/huge.genome\n",
            "bedtools2/test/shift/issue_807.genomesize\n",
            "bedtools2/test/reldist/issue_711.a.bed\n",
            "bedtools2/test/reldist/issue_711.b.bed\n",
            "bedtools2/test/reldist/test-reldist.sh\n",
            "bedtools2/test/slop/a.bed\n",
            "bedtools2/test/slop/tiny.genome\n",
            "bedtools2/test/slop/test-slop.sh\n",
            "bedtools2/test/general/hg19.fa.fai\n",
            "bedtools2/test/general/empty.bed\n",
            "bedtools2/test/general/a.chromheader.bed\n",
            "bedtools2/test/general/test-general.sh\n",
            "bedtools2/test/general/non-empty.bed.gz\n",
            "bedtools2/test/general/t.bed\n",
            "bedtools2/test/general/a.trackheader.bed\n",
            "bedtools2/test/merge/bamCol8Mean.txt\n",
            "bedtools2/test/merge/mergeBugTest.bed\n",
            "bedtools2/test/merge/vcfSVtest.2.vcf\n",
            "bedtools2/test/merge/expFormat.bed\n",
            "bedtools2/test/merge/bamCol9Mean.txt\n",
            "bedtools2/test/merge/unsorted.bed\n",
            "bedtools2/test/merge/bug254_d.bed\n",
            "bedtools2/test/merge/bug254_e.bed\n",
            "bedtools2/test/merge/bamCol6Collapse.txt\n",
            "bedtools2/test/merge/bamCol3Collapse.txt\n",
            "bedtools2/test/merge/vcfSVtest.vcf\n",
            "bedtools2/test/merge/bug254_c.bed\n",
            "bedtools2/test/merge/bamCol11Collapse.txt\n",
            "bedtools2/test/merge/a.full.bed\n",
            "bedtools2/test/merge/a.bed\n",
            "bedtools2/test/merge/test-merge.sh\n",
            "bedtools2/test/merge/precisionTest.bed\n",
            "bedtools2/test/merge/b.bed\n",
            "bedtools2/test/merge/a.names.bed\n",
            "bedtools2/test/merge/testA.vcf\n",
            "bedtools2/test/merge/chained.bed.gz\n",
            "bedtools2/test/merge/bamCol1Collapse.txt\n",
            "bedtools2/test/merge/precisionTest2.bed\n",
            "bedtools2/test/merge/bamCol10Collapse.txt\n",
            "bedtools2/test/merge/bamCol5Mean.txt\n",
            "bedtools2/test/merge/a.gff\n",
            "bedtools2/test/merge/bamCol7Collapse.txt\n",
            "bedtools2/test/merge/bamCol4Mean.txt\n",
            "bedtools2/test/merge/a.full.bam\n",
            "bedtools2/test/merge/mixedStrands.bed\n",
            "bedtools2/test/merge/fullFields.bam\n",
            "bedtools2/test/subtract/test-subtract.sh\n",
            "bedtools2/test/subtract/a.bed\n",
            "bedtools2/test/subtract/c.bed\n",
            "bedtools2/test/subtract/b.bed\n",
            "bedtools2/test/subtract/b2.bed\n",
            "bedtools2/test/subtract/d.bed\n",
            "bedtools2/test/sort/names.txt\n",
            "bedtools2/test/sort/a.bed\n",
            "bedtools2/test/sort/test-sort.sh\n",
            "bedtools2/test/sort/b.bed\n",
            "bedtools2/test/intersect/two_blocks.sam\n",
            "bedtools2/test/intersect/a_with_bothUnmapped.bam\n",
            "bedtools2/test/intersect/gdc.gff\n",
            "bedtools2/test/intersect/bug223_sv1_b.vcf\n",
            "bedtools2/test/intersect/bed3.bed\n",
            "bedtools2/test/intersect/bug150_a.bed\n",
            "bedtools2/test/intersect/test_ref.fa.fai\n",
            "bedtools2/test/intersect/x.bam\n",
            "bedtools2/test/intersect/exons.issue750.bed\n",
            "bedtools2/test/intersect/performanceTest.sh\n",
            "bedtools2/test/intersect/bed6.strand.bed\n",
            "bedtools2/test/intersect/queryUnmappedMateMappedCoordsInvalid.bam\n",
            "bedtools2/test/intersect/bug223_sv1_a.vcf\n",
            "bedtools2/test/intersect/bug150_b.bed\n",
            "bedtools2/test/intersect/headerOnly.vcf\n",
            "bedtools2/test/intersect/gdc_one.bed\n",
            "bedtools2/test/intersect/three_blocks_match.bam.bai\n",
            "bedtools2/test/intersect/bed6.strand2.bed\n",
            "bedtools2/test/intersect/null_b.bed\n",
            "bedtools2/test/intersect/a_withLargeHeader.bed\n",
            "bedtools2/test/intersect/human.hg19.vSort.genome\n",
            "bedtools2/test/intersect/gdc.bam\n",
            "bedtools2/test/intersect/bed4.bed\n",
            "bedtools2/test/intersect/chromOrderB.bed\n",
            "bedtools2/test/intersect/bedplus.bed\n",
            "bedtools2/test/intersect/test_ref.fa\n",
            "bedtools2/test/intersect/bed5.bed\n",
            "bedtools2/test/intersect/oneRecordNoNewline.bed\n",
            "bedtools2/test/intersect/cram_test (2)/\n",
            "bedtools2/test/intersect/aWithHeaderVsB.txt\n",
            "bedtools2/test/intersect/null_c.bed\n",
            "bedtools2/test/intersect/issue-919.region.bed\n",
            "bedtools2/test/intersect/j1.bed\n",
            "bedtools2/test/intersect/b.issue311.vcf\n",
            "bedtools2/test/intersect/null_a.bed\n",
            "bedtools2/test/intersect/oneUnmapped.bam\n",
            "bedtools2/test/intersect/chromOrderA.bed\n",
            "bedtools2/test/intersect/bed6.bed\n",
            "bedtools2/test/intersect/issue_773_b.bed\n",
            "bedtools2/test/intersect/new_test-intersect.sh\n",
            "bedtools2/test/intersect/chromsOutOfOrder.bed\n",
            "bedtools2/test/intersect/jim.bed\n",
            "bedtools2/test/intersect/three_blocks_nomatch.bed\n",
            "bedtools2/test/intersect/cram_test (1)/\n",
            "bedtools2/test/intersect/b_vcfSVtest.vcf\n",
            "bedtools2/test/intersect/perfData (2)/\n",
            "bedtools2/test/intersect/bug223_f.vcf\n",
            "bedtools2/test/intersect/a.issue311.bed\n",
            "bedtools2/test/intersect/three_blocks.sam\n",
            "bedtools2/test/intersect/gdc_exp\n",
            "bedtools2/test/intersect/bug44_b.bed\n",
            "bedtools2/test/intersect/bug223_d.vcf\n",
            "bedtools2/test/intersect/perfData (1)/\n",
            "bedtools2/test/intersect/bug223_e.vcf\n",
            "bedtools2/test/intersect/a_testZeroLen.bed\n",
            "bedtools2/test/intersect/a_bgzipped.bed.gz\n",
            "bedtools2/test/intersect/x.bed\n",
            "bedtools2/test/intersect/a_withLargeHeader_bgzipped.bed.gz\n",
            "bedtools2/test/intersect/dosLineChar_a.bed\n",
            "bedtools2/test/intersect/bug44_a.vcf.gz\n",
            "bedtools2/test/intersect/b.cram\n",
            "bedtools2/test/intersect/y.bed\n",
            "bedtools2/test/intersect/test-intersect.sh\n",
            "bedtools2/test/intersect/a_vcfSVtest.vcf\n",
            "bedtools2/test/intersect/blocks.bed12\n",
            "bedtools2/test/intersect/issue-919.bed\n",
            "bedtools2/test/intersect/recordsOutOfOrder.bed\n",
            "bedtools2/test/intersect/a.bed\n",
            "bedtools2/test/intersect/jim.vcf\n",
            "bedtools2/test/intersect/c.bed\n",
            "bedtools2/test/intersect/three_blocks_match.bam\n",
            "bedtools2/test/intersect/dosLineCharWithExtraTab_b.bed\n",
            "bedtools2/test/intersect/a.cram\n",
            "bedtools2/test/intersect/b.bed\n",
            "bedtools2/test/intersect/two_blocks_partial.bam.bai\n",
            "bedtools2/test/intersect/foo\n",
            "bedtools2/test/intersect/split.issue750.bam\n",
            "bedtools2/test/intersect/mapped_and_unmapped.sam\n",
            "bedtools2/test/intersect/two_blocks_partial.bam\n",
            "bedtools2/test/intersect/e.bed\n",
            "bedtools2/test/intersect/a_withLargeHeader_gzipped.bed.gz\n",
            "bedtools2/test/intersect/svlen.vcf\n",
            "bedtools2/test/intersect/d.bed\n",
            "bedtools2/test/intersect/splitBug155_a.bed\n",
            "bedtools2/test/intersect/multi_intersect/\n",
            "bedtools2/test/intersect/notexthdr.bam\n",
            "bedtools2/test/intersect/large_b.bed\n",
            "bedtools2/test/intersect/nonamecheck_a.bed\n",
            "bedtools2/test/intersect/issue_773_y.bed\n",
            "bedtools2/test/intersect/sortAndNaming/\n",
            "bedtools2/test/intersect/split.issue750.sam\n",
            "bedtools2/test/intersect/ultra-long-bed4.bed\n",
            "bedtools2/test/intersect/a_gzipped.bed.gz\n",
            "bedtools2/test/intersect/splitBug155_b.bed\n",
            "bedtools2/test/intersect/bug167_strandSweep.bed\n",
            "bedtools2/test/intersect/large_a.bed\n",
            "bedtools2/test/intersect/nonamecheck_b.bed\n",
            "bedtools2/test/intersect/b.issue311.gff\n",
            "bedtools2/test/intersect/two_blocks_partial.bed\n",
            "bedtools2/test/intersect/three_blocks_match.bed\n",
            "bedtools2/test/intersect/human.hg19.genome\n",
            "bedtools2/test/intersect/bug187_a.bed\n",
            "bedtools2/test/intersect/strand_with_star.b.bed\n",
            "bedtools2/test/intersect/b_testZeroLen.bed\n",
            "bedtools2/test/intersect/one_block.sam\n",
            "bedtools2/test/intersect/bug187_b.bed\n",
            "bedtools2/test/intersect/three_blocks_match_1bp.bed\n",
            "bedtools2/test/intersect/strand_with_star.a.bed\n",
            "bedtools2/test/intersect/aVSb.bam\n",
            "bedtools2/test/intersect/a.bam\n",
            "bedtools2/test/intersect/sortAndNaming/num_all.bed\n",
            "bedtools2/test/intersect/sortAndNaming/diffs.txt\n",
            "bedtools2/test/intersect/sortAndNaming/db1_noChr.bed\n",
            "bedtools2/test/intersect/sortAndNaming/alpha_all.bed\n",
            "bedtools2/test/intersect/sortAndNaming/test-sort-and-naming.sh\n",
            "bedtools2/test/intersect/sortAndNaming/exp1\n",
            "bedtools2/test/intersect/sortAndNaming/q1_gls.bed\n",
            "bedtools2/test/intersect/sortAndNaming/db1_num.bed\n",
            "bedtools2/test/intersect/sortAndNaming/db2_numBackwards.bed\n",
            "bedtools2/test/intersect/sortAndNaming/num_missing.bed\n",
            "bedtools2/test/intersect/sortAndNaming/q1_num.bed\n",
            "bedtools2/test/intersect/sortAndNaming/db2_num.bed\n",
            "bedtools2/test/intersect/sortAndNaming/db1_leadingZero.txt\n",
            "bedtools2/test/intersect/sortAndNaming/sq1.bed\n",
            "bedtools2/test/intersect/sortAndNaming/db1_gls.bed\n",
            "bedtools2/test/intersect/sortAndNaming/alpha_missing.bed\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/\n",
            "bedtools2/test/intersect/sortAndNaming/sdb1.bed\n",
            "bedtools2/test/intersect/sortAndNaming/db3_numBackwards.bed\n",
            "bedtools2/test/intersect/sortAndNaming/q1a_num.bed\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/curr5v5.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/2.21_500Kvs500K.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/q500K.bed\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/dbchr1.bed\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/2.21_chr1.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/diffs.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/curr500Kvs500K.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/qchr1.bed\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/2.21_5v5.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/currChr1.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/db500K.bed\n",
            "bedtools2/test/intersect/multi_intersect/query3.bed\n",
            "bedtools2/test/intersect/multi_intersect/query2.bed\n",
            "bedtools2/test/intersect/multi_intersect/d8.bed\n",
            "bedtools2/test/intersect/multi_intersect/d9.bed\n",
            "bedtools2/test/intersect/multi_intersect/query.bed\n",
            "bedtools2/test/intersect/multi_intersect/g.bed\n",
            "bedtools2/test/intersect/multi_intersect/test-multi_intersect.sh\n",
            "bedtools2/test/intersect/multi_intersect/d4.bed\n",
            "bedtools2/test/intersect/multi_intersect/d5.bed\n",
            "bedtools2/test/intersect/multi_intersect/d7.bed\n",
            "bedtools2/test/intersect/multi_intersect/d6.bed\n",
            "bedtools2/test/intersect/multi_intersect/d2.bed\n",
            "bedtools2/test/intersect/multi_intersect/d3.bed\n",
            "bedtools2/test/intersect/multi_intersect/d1.bed\n",
            "bedtools2/test/intersect/perfData (1)/a10M.bed\n",
            "bedtools2/test/intersect/perfData (2)/a10M.bed\n",
            "bedtools2/test/intersect/cram_test (1)/test_ref.fa.fai\n",
            "bedtools2/test/intersect/cram_test (1)/test_ref.fa\n",
            "bedtools2/test/intersect/cram_test (1)/bak/\n",
            "bedtools2/test/intersect/cram_test (1)/b.cram\n",
            "bedtools2/test/intersect/cram_test (1)/b.sam\n",
            "bedtools2/test/intersect/cram_test (1)/a.cram\n",
            "bedtools2/test/intersect/cram_test (1)/a.sam\n",
            "bedtools2/test/intersect/cram_test (1)/b.bam\n",
            "bedtools2/test/intersect/cram_test (1)/a.bam\n",
            "bedtools2/test/intersect/cram_test (1)/bak/test_b.cram\n",
            "bedtools2/test/intersect/cram_test (1)/bak/test_b.sam\n",
            "bedtools2/test/intersect/cram_test (1)/bak/test.sam\n",
            "bedtools2/test/intersect/cram_test (1)/bak/test.cram\n",
            "bedtools2/test/intersect/cram_test (2)/test_ref.fa.fai\n",
            "bedtools2/test/intersect/cram_test (2)/test_ref.fa\n",
            "bedtools2/test/intersect/cram_test (2)/bak/\n",
            "bedtools2/test/intersect/cram_test (2)/b.cram\n",
            "bedtools2/test/intersect/cram_test (2)/b.sam\n",
            "bedtools2/test/intersect/cram_test (2)/a.cram\n",
            "bedtools2/test/intersect/cram_test (2)/a.sam\n",
            "bedtools2/test/intersect/cram_test (2)/b.bam\n",
            "bedtools2/test/intersect/cram_test (2)/a.bam\n",
            "bedtools2/test/intersect/cram_test (2)/bak/test_b.cram\n",
            "bedtools2/test/intersect/cram_test (2)/bak/test_b.sam\n",
            "bedtools2/test/intersect/cram_test (2)/bak/test.sam\n",
            "bedtools2/test/intersect/cram_test (2)/bak/test.cram\n",
            "bedtools2/test/complement/test-complement.sh\n",
            "bedtools2/test/complement/issue_503.genome\n",
            "bedtools2/test/complement/issue_503.bed\n",
            "bedtools2/test/fisher/test-fisher.sh\n",
            "bedtools2/test/fisher/sim.py\n",
            "bedtools2/test/fisher/plot.py\n",
            "bedtools2/test/fisher/cmp.sh\n",
            "bedtools2/test/fisher/issue954_a.bed\n",
            "bedtools2/test/fisher/issue954_b.bed\n",
            "bedtools2/test/fisher/dm6.fai\n",
            "bedtools2/test/fisher/t.60.genome\n",
            "bedtools2/test/fisher/get.sh\n",
            "bedtools2/test/fisher/README.md\n",
            "bedtools2/test/fisher/a.bed\n",
            "bedtools2/test/fisher/b.bed\n",
            "bedtools2/test/fisher/tumor.gff\n",
            "bedtools2/test/fisher/t.500.genome\n",
            "bedtools2/test/fisher/a_merge.bed\n",
            "bedtools2/test/fisher/shuf.sh\n",
            "bedtools2/test/fisher/test.bed\n",
            "bedtools2/test/fisher/x\n",
            "bedtools2/test/fisher/issue954.genome\n",
            "bedtools2/test/fisher/y\n",
            "bedtools2/test/shuffle/incl.sizedif.bed\n",
            "bedtools2/test/shuffle/test-shuffle.sh\n",
            "bedtools2/test/shuffle/excl.bed\n",
            "bedtools2/test/shuffle/simrep.bed\n",
            "bedtools2/test/shuffle/incl.bed\n",
            "bedtools2/test/expand/expand.txt\n",
            "bedtools2/test/expand/test-expand.sh\n",
            "bedtools2/test/cluster/in.bed\n",
            "bedtools2/test/cluster/test-cluster.sh\n",
            "bedtools2/test/bedtobam/test-bedtobam.sh\n",
            "bedtools2/test/bedtobam/chrsize.tmp\n",
            "bedtools2/test/split/test-split.sh\n",
            "bedtools2/test/split/randData.bed\n",
            "/content/InSiTE/bedtools2\n",
            "Building BEDTools:\n",
            "=========================================================\n",
            "CXXFLAGS is [-g -Wall -O2 -std=c++11 ]\n",
            "DETECTED_VERSION = v2.31.0\n",
            "CURRENT_VERSION  = \n",
            "Updating version file.\n",
            "  * compiling src/bedtools.cpp\n",
            "  * compiling src/annotateBed/annotateMain.cpp\n",
            "  * compiling src/annotateBed/annotateBed.cpp\n",
            "  * compiling src/bamToBed/bamToBed.cpp\n",
            "\u001b[01m\u001b[Ksrc/bamToBed/bamToBed.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint bamtobed_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/bamToBed/bamToBed.cpp:88:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KuseAlignmentScore\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   88 |     bool \u001b[01;35m\u001b[KuseAlignmentScore\u001b[m\u001b[K = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/bamToBed/bamToBed.cpp:12\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:30:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   32 |                 \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/bamToFastq/bamToFastqMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/bamToFastq/bamToFastqMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint bamtofastq_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/bamToFastq/bamToFastqMain.cpp:38:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFastq2\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   38 |     bool \u001b[01;35m\u001b[KhaveFastq2\u001b[m\u001b[K    = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/bamToFastq/bamToFastq.cpp\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/bamToFastq/bamToFastq.h:15\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/bamToFastq/bamToFastq.cpp:15\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:30:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   32 |                 \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/bedToBam/bedToBam.cpp\n",
            "\u001b[01m\u001b[Ksrc/bedToBam/bedToBam.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint bedtobam_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/bedToBam/bedToBam.cpp:60:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveMapQual\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   60 |     bool \u001b[01;35m\u001b[KhaveMapQual\u001b[m\u001b[K     = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/bedpeToBam/bedpeToBam.cpp\n",
            "\u001b[01m\u001b[Ksrc/bedpeToBam/bedpeToBam.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint bedpetobam_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/bedpeToBam/bedpeToBam.cpp:61:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveMapQual\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   61 |     bool \u001b[01;35m\u001b[KhaveMapQual\u001b[m\u001b[K     = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/bedToIgv/bedToIgv.cpp\n",
            "  * compiling src/bed12ToBed6/bed12ToBed6.cpp\n",
            "  * compiling src/closestFile/closestHelp.cpp\n",
            "  * compiling src/closestFile/closestFile.cpp\n",
            "  * compiling src/clusterBed/clusterMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/clusterBed/clusterMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint cluster_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/clusterBed/clusterMain.cpp:38:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveMaxDistance\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   38 |     bool \u001b[01;35m\u001b[KhaveMaxDistance\u001b[m\u001b[K = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/clusterBed/clusterBed.cpp\n",
            "  * compiling src/complementFile/complementHelp.cpp\n",
            "  * compiling src/complementFile/complementFile.cpp\n",
            "  * compiling src/coverageFile/coverageHelp.cpp\n",
            "  * compiling src/coverageFile/coverageFile.cpp\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvirtual void CoverageFile::giveFinalReport(RecordOutputMgr*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:100:25:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint asprintf(char**, const char*, ...)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  100 |                 \u001b[01;35m\u001b[Kasprintf(&depthPctString, \"%0.7f\", depthPct)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid CoverageFile::doMean(RecordOutputMgr*, RecordKeyVector&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:181:17:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint asprintf(char**, const char*, ...)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  181 |         \u001b[01;35m\u001b[Kasprintf(&meanString, \"%0.7f\", mean)\u001b[m\u001b[K;\n",
            "      |         \u001b[01;35m\u001b[K~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid CoverageFile::doHist(RecordOutputMgr*, RecordKeyVector&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:210:25:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint asprintf(char**, const char*, ...)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  210 |                 \u001b[01;35m\u001b[Kasprintf(&coveredFractionString, \"%0.7f\", coveredFraction)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid CoverageFile::doDefault(RecordOutputMgr*, RecordKeyVector&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:230:17:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint asprintf(char**, const char*, ...)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  230 |         \u001b[01;35m\u001b[Kasprintf(&coveredFractionString, \"%0.7f\", coveredFraction)\u001b[m\u001b[K;\n",
            "      |         \u001b[01;35m\u001b[K~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/expand/expand.cpp\n",
            "  * compiling src/fastaFromBed/fastaFromBedMain.cpp\n",
            "  * compiling src/fastaFromBed/fastaFromBed.cpp\n",
            "  * compiling src/flankBed/flankBedMain.cpp\n",
            "  * compiling src/flankBed/flankBed.cpp\n",
            "  * compiling src/genomeCoverageBed/genomeCoverageMain.cpp\n",
            "  * compiling src/genomeCoverageBed/genomeCoverageBed.cpp\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/utils/GenomeFile/NewGenomeFile.h:20\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/genomeCoverageBed/genomeCoverageBed.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/genomeCoverageBed/genomeCoverageBed.cpp:13\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:30:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   32 |                 \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/getOverlap/getOverlap.cpp\n",
            "\u001b[01m\u001b[Ksrc/getOverlap/getOverlap.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint getoverlap_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/getOverlap/getOverlap.cpp:44:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveColumns\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   44 |     bool \u001b[01;35m\u001b[KhaveColumns\u001b[m\u001b[K = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/groupBy/groupByHelp.cpp\n",
            "  * compiling src/groupBy/groupBy.cpp\n",
            "  * compiling src/intersectFile/intersectHelp.cpp\n",
            "  * compiling src/intersectFile/intersectFile.cpp\n",
            "  * compiling src/fisher/fisherHelp.cpp\n",
            "  * compiling src/fisher/fisher.cpp\n",
            "  * compiling src/fisher/kfunc.cpp\n",
            "  * compiling src/jaccard/jaccardHelp.cpp\n",
            "  * compiling src/jaccard/jaccard.cpp\n",
            "  * compiling src/linksBed/linksMain.cpp\n",
            "  * compiling src/linksBed/linksBed.cpp\n",
            "  * compiling src/maskFastaFromBed/maskFastaFromBedMain.cpp\n",
            "  * compiling src/maskFastaFromBed/maskFastaFromBed.cpp\n",
            "  * compiling src/mapFile/mapHelp.cpp\n",
            "  * compiling src/mapFile/mapFile.cpp\n",
            "  * compiling src/mergeFile/mergeHelp.cpp\n",
            "  * compiling src/mergeFile/mergeFile.cpp\n",
            "  * compiling src/multiBamCov/multiBamCovMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/multiBamCov/multiBamCovMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint multibamcov_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/multiBamCov/multiBamCovMain.cpp:38:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveBed\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   38 |     bool \u001b[01;35m\u001b[KhaveBed\u001b[m\u001b[K           = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/multiBamCov/multiBamCovMain.cpp:39:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveBams\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   39 |     bool \u001b[01;35m\u001b[KhaveBams\u001b[m\u001b[K          = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/multiBamCov/multiBamCovMain.cpp:47:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFraction\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   47 |     bool \u001b[01;35m\u001b[KhaveFraction\u001b[m\u001b[K       = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/multiBamCov/multiBamCov.cpp\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/BamReader.hpp:4\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamMultiReader.h:4\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/multiBamCov/multiBamCov.h:16\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/multiBamCov/multiBamCov.cpp:13\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:30:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   32 |                 \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/multiIntersectBed/multiIntersectBedMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/multiIntersectBed/multiIntersectBedMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint multiintersect_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/multiIntersectBed/multiIntersectBedMain.cpp:45:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFiles\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   45 |     bool \u001b[01;35m\u001b[KhaveFiles\u001b[m\u001b[K         = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/multiIntersectBed/multiIntersectBedMain.cpp:47:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveGenome\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   47 |     bool \u001b[01;35m\u001b[KhaveGenome\u001b[m\u001b[K        = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/multiIntersectBed/multiIntersectBedMain.cpp:48:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFiller\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   48 |     bool \u001b[01;35m\u001b[KhaveFiller\u001b[m\u001b[K        = true;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/multiIntersectBed/multiIntersectBed.cpp\n",
            "  * compiling src/nucBed/nucBedMain.cpp\n",
            "  * compiling src/nucBed/nucBed.cpp\n",
            "  * compiling src/pairToBed/pairToBedMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/pairToBed/pairToBedMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint pairtobed_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/pairToBed/pairToBedMain.cpp:43:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFraction\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   43 |     bool \u001b[01;35m\u001b[KhaveFraction\u001b[m\u001b[K       = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/pairToBed/pairToBed.cpp\n",
            "\u001b[01m\u001b[Ksrc/pairToBed/pairToBed.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid BedIntersectPE::FindSpanningOverlaps(const BEDPE&, std::vector<BED>&, const string&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/pairToBed/pairToBed.cpp:257:12:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KspanLength\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  257 |     CHRPOS \u001b[01;35m\u001b[KspanLength\u001b[m\u001b[K = 0;\n",
            "      |            \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/pairToBed/pairToBed.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kbool BedIntersectPE::FindOneOrMoreSpanningOverlaps(const BEDPE&, const string&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/pairToBed/pairToBed.cpp:310:12:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KspanLength\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  310 |     CHRPOS \u001b[01;35m\u001b[KspanLength\u001b[m\u001b[K = 0;\n",
            "      |            \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/pairToBed/pairToBed.h:15\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/pairToBed/pairToBed.cpp:13\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:30:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   32 |                 \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/pairToPair/pairToPairMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/pairToPair/pairToPairMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint pairtopair_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/pairToPair/pairToPairMain.cpp:44:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFraction\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   44 |     bool \u001b[01;35m\u001b[KhaveFraction\u001b[m\u001b[K = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/pairToPair/pairToPair.cpp\n",
            "\u001b[01m\u001b[Ksrc/pairToPair/pairToPair.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid PairToPair::FindOverlaps(const BEDPE&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/pairToPair/pairToPair.cpp:112:14:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[Kfound1\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  112 |         bool \u001b[01;35m\u001b[Kfound1\u001b[m\u001b[K = false;\n",
            "      |              \u001b[01;35m\u001b[K^~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/pairToPair/pairToPair.cpp:113:14:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[Kfound2\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  113 |         bool \u001b[01;35m\u001b[Kfound2\u001b[m\u001b[K = false;\n",
            "      |              \u001b[01;35m\u001b[K^~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/randomBed/randomBedMain.cpp\n",
            "  * compiling src/randomBed/randomBed.cpp\n",
            "  * compiling src/regressTest/regressTestMain.cpp\n",
            "  * compiling src/regressTest/RegressTest.cpp\n",
            "\u001b[01m\u001b[Ksrc/regressTest/RegressTest.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kbool RegressTest::executeAndCompareCorrectness(const fileListType&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/regressTest/RegressTest.cpp:431:15:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint system(const char*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  431 |         \u001b[01;35m\u001b[Ksystem(diffCmd.c_str())\u001b[m\u001b[K;\n",
            "      |         \u001b[01;35m\u001b[K~~~~~~^~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/regressTest/RegressTest.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kbool RegressTest::startMemoryProfile(bool)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/regressTest/RegressTest.cpp:573:14:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kchar* fgets(char*, int, FILE*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  573 |         \u001b[01;35m\u001b[Kfgets(sLine, 4192, fp)\u001b[m\u001b[K;\n",
            "      |         \u001b[01;35m\u001b[K~~~~~^~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/regressTest/RegressTest.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kbool RegressTest::calcMemoryStats()\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/regressTest/RegressTest.cpp:622:22:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kchar* fgets(char*, int, FILE*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  622 |                 \u001b[01;35m\u001b[Kfgets(sLine, 4192, fp)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~^~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/reldist/reldistMain.cpp\n",
            "  * compiling src/reldist/reldist.cpp\n",
            "\u001b[01m\u001b[Ksrc/reldist/reldist.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid RelativeDistance::CalculateRelativeDistance()\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/reldist/reldist.cpp:129:25:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kcomparison of integer expressions of different signedness: ‘\u001b[01m\u001b[KCHRPOS\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong int\u001b[m\u001b[K’} and ‘\u001b[01m\u001b[Kstd::vector<long int>::size_type\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong unsigned int\u001b[m\u001b[K’} [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wsign-compare\u0007-Wsign-compare\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  129 |             if (\u001b[01;35m\u001b[Klow_idx != chrom_mids->size() - 1\u001b[m\u001b[K)\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/sampleFile/sampleHelp.cpp\n",
            "  * compiling src/sampleFile/sampleFile.cpp\n",
            "  * compiling src/shiftBed/shiftBedMain.cpp\n",
            "  * compiling src/shiftBed/shiftBed.cpp\n",
            "  * compiling src/shuffleBed/shuffleBedMain.cpp\n",
            "  * compiling src/shuffleBed/shuffleBed.cpp\n",
            "  * compiling src/slopBed/slopBedMain.cpp\n",
            "  * compiling src/slopBed/slopBed.cpp\n",
            "  * compiling src/sortBed/sortMain.cpp\n",
            "  * compiling src/sortBed/sortBed.cpp\n",
            "  * compiling src/spacingFile/spacingHelp.cpp\n",
            "  * compiling src/spacingFile/spacingFile.cpp\n",
            "  * compiling src/split/splitBedMain.cpp\n",
            "  * compiling src/split/splitBed.cpp\n",
            "  * compiling src/subtractFile/subtractHelp.cpp\n",
            "  * compiling src/subtractFile/subtractFile.cpp\n",
            "  * compiling src/summaryFile/summaryHelp.cpp\n",
            "  * compiling src/summaryFile/summaryFile.cpp\n",
            "  * compiling src/tagBam/tagBamMain.cpp\n",
            "  * compiling src/tagBam/tagBam.cpp\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/tagBam/tagBam.h:17\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/tagBam/tagBam.cpp:13\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:30:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   32 |                 \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/unionBedGraphs/unionBedGraphsMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/unionBedGraphs/unionBedGraphsMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint unionbedgraphs_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/unionBedGraphs/unionBedGraphsMain.cpp:46:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFiles\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   46 |     bool \u001b[01;35m\u001b[KhaveFiles\u001b[m\u001b[K         = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/unionBedGraphs/unionBedGraphsMain.cpp:48:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveGenome\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   48 |     bool \u001b[01;35m\u001b[KhaveGenome\u001b[m\u001b[K        = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/unionBedGraphs/unionBedGraphsMain.cpp:49:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFiller\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   49 |     bool \u001b[01;35m\u001b[KhaveFiller\u001b[m\u001b[K        = true;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/unionBedGraphs/unionBedGraphs.cpp\n",
            "  * compiling src/windowBed/windowMain.cpp\n",
            "  * compiling src/windowBed/windowBed.cpp\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/windowBed/windowBed.h:15\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/windowBed/windowBed.cpp:13\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:30:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   32 |                 \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/windowMaker/windowMakerMain.cpp\n",
            "  * compiling src/windowMaker/windowMaker.cpp\n",
            "  * compiling src/utils/FileRecordTools/FileRecordMgr.cpp\n",
            "  * compiling src/utils/FileRecordTools/FileRecordMergeMgr.cpp\n",
            "  * compiling src/utils/FileRecordTools/FileReaders/FileReader.cpp\n",
            "  * compiling src/utils/FileRecordTools/FileReaders/SingleLineDelimTextFileReader.cpp\n",
            "  * compiling src/utils/FileRecordTools/FileReaders/BamFileReader.cpp\n",
            "  * compiling src/utils/FileRecordTools/FileReaders/BufferedStreamMgr.cpp\n",
            "\u001b[01m\u001b[Ksrc/utils/FileRecordTools/FileReaders/BufferedStreamMgr.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kbool BufferedStreamMgr::readFileChunk()\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/FileRecordTools/FileReaders/BufferedStreamMgr.cpp:148:37:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kcomparison of integer expressions of different signedness: ‘\u001b[01m\u001b[Kint\u001b[m\u001b[K’ and ‘\u001b[01m\u001b[Ksize_t\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong unsigned int\u001b[m\u001b[K’} [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wsign-compare\u0007-Wsign-compare\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  148 |                 if (\u001b[01;35m\u001b[K_mainBufCurrLen < _useBufSize\u001b[m\u001b[K) {\n",
            "      |                     \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~^~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/utils/FileRecordTools/FileReaders/InputStreamMgr.cpp\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/utils/FileRecordTools/FileReaders/InputStreamMgr.h:14\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/utils/FileRecordTools/FileReaders/InputStreamMgr.cpp:8\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:30:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   32 |                 \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/utils/FileRecordTools/Records/Record.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/EmptyRecord.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/Bed3Interval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/Bed4Interval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/BedGraphInterval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/Bed5Interval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/Bed6Interval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/PlusFields.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/GffRecord.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/GffPlusRecord.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/NoPosPlusRecord.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/BedPlusInterval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/Bed12Interval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/BamRecord.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/VcfRecord.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/BlockMgr.cpp\n",
            "\u001b[01m\u001b[Ksrc/utils/FileRecordTools/Records/BlockMgr.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kint BlockMgr::getNonRedundantOverlap()\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/FileRecordTools/Records/BlockMgr.cpp:194:35:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kcomparison of integer expressions of different signedness: ‘\u001b[01m\u001b[Kint\u001b[m\u001b[K’ and ‘\u001b[01m\u001b[Kstd::vector<long int>::size_type\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong unsigned int\u001b[m\u001b[K’} [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wsign-compare\u0007-Wsign-compare\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  194 |                 for (int i = 1; \u001b[01;35m\u001b[Ki < _overlapStarts.size()\u001b[m\u001b[K; ++i)\n",
            "      |                                 \u001b[01;35m\u001b[K~~^~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/utils/FileRecordTools/Records/StrandQueue.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/RecordMgr.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/RecordList.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/RecordKeyList.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/RecordKeyVector.cpp\n",
            "  * compiling src/utils/bedFile/bedFile.cpp\n",
            "  * compiling src/utils/BinTree/BinTree.cpp\n",
            "  * compiling src/utils/version/version.cpp\n",
            "  * compiling src/utils/bedGraphFile/bedGraphFile.cpp\n",
            "  * compiling src/utils/chromsweep/chromsweep.cpp\n",
            "  * compiling src/utils/Contexts/ContextBase.cpp\n",
            "  * compiling src/utils/Contexts/ContextIntersect.cpp\n",
            "  * compiling src/utils/Contexts/ContextFisher.cpp\n",
            "  * compiling src/utils/Contexts/ContextMap.cpp\n",
            "  * compiling src/utils/Contexts/ContextSample.cpp\n",
            "  * compiling src/utils/Contexts/ContextSpacing.cpp\n",
            "  * compiling src/utils/Contexts/ContextMerge.cpp\n",
            "  * compiling src/utils/Contexts/ContextJaccard.cpp\n",
            "  * compiling src/utils/Contexts/ContextClosest.cpp\n",
            "  * compiling src/utils/Contexts/ContextSubtract.cpp\n",
            "  * compiling src/utils/Contexts/ContextCoverage.cpp\n",
            "  * compiling src/utils/Contexts/ContextComplement.cpp\n",
            "  * compiling src/utils/Contexts/ContextGroupBy.cpp\n",
            "  * compiling src/utils/Contexts/ContextSummary.cpp\n",
            "  * compiling src/utils/general/ParseTools.cpp\n",
            "  * compiling src/utils/general/PushBackStreamBuf.cpp\n",
            "  * compiling src/utils/general/CompressionTools.cpp\n",
            "  * compiling src/utils/general/Tokenizer.cpp\n",
            "  * compiling src/utils/general/CommonHelp.cpp\n",
            "  * compiling src/utils/general/ErrorMsg.cpp\n",
            "  * compiling src/utils/general/Random.cpp\n",
            "g++ -g -Wall -O2 -std=c++11  -Isrc/utils/gzstream  -c -o obj/gzstream.o src/utils/gzstream/gzstream.C\n",
            "  * compiling src/utils/fileType/fileType.cpp\n",
            "  * compiling src/utils/fileType/FileRecordTypeChecker.cpp\n",
            "\u001b[01m\u001b[Ksrc/utils/fileType/FileRecordTypeChecker.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kbool FileRecordTypeChecker::isTextDelimtedFormat(const char*, size_t)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/fileType/FileRecordTypeChecker.cpp:366:33:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kcomparison of integer expressions of different signedness: ‘\u001b[01m\u001b[Kint\u001b[m\u001b[K’ and ‘\u001b[01m\u001b[Ksize_t\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong unsigned int\u001b[m\u001b[K’} [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wsign-compare\u0007-Wsign-compare\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  366 |                 for (int j=0; \u001b[01;35m\u001b[Kj < len\u001b[m\u001b[K; j++) {\n",
            "      |                               \u001b[01;35m\u001b[K~~^~~~~\u001b[m\u001b[K\n",
            "  * compiling src/utils/bedFilePE/bedFilePE.cpp\n",
            "  * compiling src/utils/KeyListOps/KeyListOps.cpp\n",
            "  * compiling src/utils/KeyListOps/KeyListOpsMethods.cpp\n",
            "  * compiling src/utils/NewChromsweep/NewChromsweep.cpp\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/NewChromsweep/NewChromsweep.cpp:13\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/NewChromsweep/NewChromsweep.h:\u001b[m\u001b[K In constructor ‘\u001b[01m\u001b[KNewChromSweep::NewChromSweep(ContextIntersect*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/NewChromsweep/NewChromsweep.h:124:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[KNewChromSweep::_testLastQueryRec\u001b[m\u001b[K’ will be initialized after [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wreorder\u0007-Wreorder\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  124 |     bool \u001b[01;35m\u001b[K_testLastQueryRec\u001b[m\u001b[K;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/utils/NewChromsweep/NewChromsweep.h:93:14:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K  ‘\u001b[01m\u001b[Kbool NewChromSweep::_runToDbEnd\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wreorder\u0007-Wreorder\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   93 |         bool \u001b[01;35m\u001b[K_runToDbEnd\u001b[m\u001b[K;\n",
            "      |              \u001b[01;35m\u001b[K^~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/utils/NewChromsweep/NewChromsweep.cpp:17:1:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K  when initialized here [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wreorder\u0007-Wreorder\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   17 | \u001b[01;35m\u001b[KNewChromSweep\u001b[m\u001b[K::NewChromSweep(ContextIntersect *context)\n",
            "      | \u001b[01;35m\u001b[K^~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/utils/NewChromsweep/CloseSweep.cpp\n",
            "  * compiling src/utils/sequenceUtilities/sequenceUtils.cpp\n",
            "  * compiling src/utils/tabFile/tabFile.cpp\n",
            "  * compiling src/utils/BamTools-Ancillary/BamAncillary.cpp\n",
            "  * compiling src/utils/BlockedIntervals/BlockedIntervals.cpp\n",
            "  * compiling src/utils/Fasta/Fasta.cpp\n",
            "  * compiling src/utils/Fasta/split.cpp\n",
            "  * compiling src/utils/VectorOps/VectorOps.cpp\n",
            "  * compiling src/utils/GenomeFile/NewGenomeFile.cpp\n",
            "  * compiling src/utils/GenomeFile/GenomeFile.cpp\n",
            "\u001b[01m\u001b[Ksrc/utils/GenomeFile/GenomeFile.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid GenomeFile::loadGenomeFileIntoMap()\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/GenomeFile/GenomeFile.cpp:60:26:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[Kc2\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   60 |                     long \u001b[01;35m\u001b[Kc2\u001b[m\u001b[K;\n",
            "      |                          \u001b[01;35m\u001b[K^~\u001b[m\u001b[K\n",
            "  * compiling src/utils/RecordOutputMgr/RecordOutputMgr.cpp\n",
            "\u001b[01m\u001b[Ksrc/utils/RecordOutputMgr/RecordOutputMgr.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid RecordOutputMgr::printClosest(RecordKeyVector&, const std::vector<long int>*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/RecordOutputMgr/RecordOutputMgr.cpp:170:56:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kcomparison of integer expressions of different signedness: ‘\u001b[01m\u001b[Kint\u001b[m\u001b[K’ and ‘\u001b[01m\u001b[Ksize_t\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong unsigned int\u001b[m\u001b[K’} [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wsign-compare\u0007-Wsign-compare\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  170 |                 if (\u001b[01;35m\u001b[Kcontext->getNumClosestHitsWanted() > keyList.size()\u001b[m\u001b[K)\n",
            "      |                     \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/utils/RecordOutputMgr/RecordOutputMgr.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid RecordOutputMgr::reportOverlapSummary(RecordKeyVector&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/RecordOutputMgr/RecordOutputMgr.cpp:456:38:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kcomparison of integer expressions of different signedness: ‘\u001b[01m\u001b[Ksize_t\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong unsigned int\u001b[m\u001b[K’} and ‘\u001b[01m\u001b[Kint\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wsign-compare\u0007-Wsign-compare\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  456 |                 for (size_t i = 0; \u001b[01;35m\u001b[Ki < _context->getNumInputFiles()\u001b[m\u001b[K; i++)\n",
            "      |                                    \u001b[01;35m\u001b[K~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/utils/ToolBase/ToolBase.cpp\n",
            "  * compiling src/utils/driver/BedtoolsDriver.cpp\n",
            "cd src/utils/htslib && make config.h\n",
            "make[1]: Entering directory '/content/InSiTE/bedtools2/src/utils/htslib'\n",
            "echo '/* Default config.h generated by Makefile */' > config.h\n",
            "echo '#define HAVE_LIBBZ2 1' >> config.h\n",
            "echo '#define HAVE_LIBLZMA 1' >> config.h\n",
            "echo '#ifndef __APPLE__' >> config.h\n",
            "echo '#define HAVE_LZMA_H 1' >> config.h\n",
            "echo '#endif' >> config.h\n",
            "echo '#define HAVE_FSEEKO 1' >> config.h\n",
            "echo '#define HAVE_DRAND48 1' >> config.h\n",
            "make[1]: Leaving directory '/content/InSiTE/bedtools2/src/utils/htslib'\n",
            "cd src/utils/htslib && make lib-static\n",
            "make[1]: Entering directory '/content/InSiTE/bedtools2/src/utils/htslib'\n",
            "gcc -g -Wall -O2 -I.  -c -o kfunc.o kfunc.c\n",
            "gcc -g -Wall -O2 -I.  -c -o knetfile.o knetfile.c\n",
            "gcc -g -Wall -O2 -I.  -c -o kstring.o kstring.c\n",
            "gcc -g -Wall -O2 -I.  -c -o bcf_sr_sort.o bcf_sr_sort.c\n",
            "gcc -g -Wall -O2 -I.  -c -o bgzf.o bgzf.c\n",
            "gcc -g -Wall -O2 -I.  -c -o errmod.o errmod.c\n",
            "gcc -g -Wall -O2 -I.  -c -o faidx.o faidx.c\n",
            "gcc -g -Wall -O2 -I.  -c -o hfile.o hfile.c\n",
            "gcc -g -Wall -O2 -I.  -c -o hfile_net.o hfile_net.c\n",
            "echo '#define HTS_VERSION \"1.9\"' > version.h\n",
            "gcc -g -Wall -O2 -I.  -c -o hts.o hts.c\n",
            "\u001b[01m\u001b[Khts.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Khts_idx_init\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Khts.c:1397:59:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Koverflow in conversion from ‘\u001b[01m\u001b[Kuint32_t\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Kunsigned int\u001b[m\u001b[K’} to ‘\u001b[01m\u001b[Kint\u001b[m\u001b[K’ changes value from ‘\u001b[01m\u001b[Kidx->z.last_bin = 4294967295\u001b[m\u001b[K’ to ‘\u001b[01m\u001b[K-1\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Woverflow\u0007-Woverflow\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            " 1397 |     idx->z.save_bin = idx->z.save_tid = idx->z.last_tid = \u001b[01;35m\u001b[Kidx\u001b[m\u001b[K->z.last_bin = 0xffffffffu;\n",
            "      |                                                           \u001b[01;35m\u001b[K^~~\u001b[m\u001b[K\n",
            "gcc -g -Wall -O2 -I.  -c -o hts_os.o hts_os.c\n",
            "gcc -g -Wall -O2 -I.  -c -o md5.o md5.c\n",
            "gcc -g -Wall -O2 -I.  -c -o multipart.o multipart.c\n",
            "gcc -g -Wall -O2 -I.  -c -o probaln.o probaln.c\n",
            "gcc -g -Wall -O2 -I.  -c -o realn.o realn.c\n",
            "gcc -g -Wall -O2 -I.  -c -o regidx.o regidx.c\n",
            "gcc -g -Wall -O2 -I.  -c -o sam.o sam.c\n",
            "gcc -g -Wall -O2 -I.  -c -o synced_bcf_reader.o synced_bcf_reader.c\n",
            "gcc -g -Wall -O2 -I.  -c -o vcf_sweep.o vcf_sweep.c\n",
            "gcc -g -Wall -O2 -I.  -c -o tbx.o tbx.c\n",
            "gcc -g -Wall -O2 -I.  -c -o textutils.o textutils.c\n",
            "gcc -g -Wall -O2 -I.  -c -o thread_pool.o thread_pool.c\n",
            "gcc -g -Wall -O2 -I.  -c -o vcf.o vcf.c\n",
            "gcc -g -Wall -O2 -I.  -c -o vcfutils.o vcfutils.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_codecs.o cram/cram_codecs.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_decode.o cram/cram_decode.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_encode.o cram/cram_encode.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_external.o cram/cram_external.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_index.o cram/cram_index.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_io.o cram/cram_io.c\n",
            "In file included from \u001b[01m\u001b[K/usr/include/string.h:535\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Kcram/cram_io.c:53\u001b[m\u001b[K:\n",
            "In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n",
            "    inlined from ‘\u001b[01m\u001b[Kcram_populate_ref\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:2095:9\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ specified bound 4096 equals destination size [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n",
            "      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n",
            "    inlined from ‘\u001b[01m\u001b[Kfull_path\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:3748:13\u001b[m\u001b[K,\n",
            "    inlined from ‘\u001b[01m\u001b[Kcram_write_SAM_hdr\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:3823:17\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ specified bound 4096 equals destination size [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n",
            "      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n",
            "    inlined from ‘\u001b[01m\u001b[Kcram_dopen\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:4126:9\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ specified bound 20 equals destination size [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n",
            "      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_samtools.o cram/cram_samtools.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_stats.o cram/cram_stats.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/files.o cram/files.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/mFILE.o cram/mFILE.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/open_trace_file.o cram/open_trace_file.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/pooled_alloc.o cram/pooled_alloc.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/rANS_static.o cram/rANS_static.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/sam_header.o cram/sam_header.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/string_alloc.o cram/string_alloc.c\n",
            "In file included from \u001b[01m\u001b[K/usr/include/string.h:535\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Kcram/string_alloc.c:41\u001b[m\u001b[K:\n",
            "In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n",
            "    inlined from ‘\u001b[01m\u001b[Kstring_ndup\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/string_alloc.c:151:5\u001b[m\u001b[K,\n",
            "    inlined from ‘\u001b[01m\u001b[Kstring_dup\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/string_alloc.c:143:12\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ output truncated before terminating nul copying as many bytes from a string as its length [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n",
            "      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Kcram/string_alloc.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kstring_dup\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Kcram/string_alloc.c:143:12:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Klength computed here\n",
            "  143 |     return \u001b[01;36m\u001b[Kstring_ndup(a_str, instr, strlen(instr))\u001b[m\u001b[K;\n",
            "      |            \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "ar -rc libhts.a kfunc.o knetfile.o kstring.o bcf_sr_sort.o bgzf.o errmod.o faidx.o hfile.o hfile_net.o hts.o hts_os.o md5.o multipart.o probaln.o realn.o regidx.o sam.o synced_bcf_reader.o vcf_sweep.o tbx.o textutils.o thread_pool.o vcf.o vcfutils.o cram/cram_codecs.o cram/cram_decode.o cram/cram_encode.o cram/cram_external.o cram/cram_index.o cram/cram_io.o cram/cram_samtools.o cram/cram_stats.o cram/files.o cram/mFILE.o cram/open_trace_file.o cram/pooled_alloc.o cram/rANS_static.o cram/sam_header.o cram/string_alloc.o\n",
            "ranlib libhts.a\n",
            "make[1]: Leaving directory '/content/InSiTE/bedtools2/src/utils/htslib'\n",
            "- Building main bedtools binary.\n",
            "done.\n",
            "- Creating executables for old CLI.\n",
            "done.\n",
            "  * compiling src/htsutil.cpp\n",
            "/content/InSiTE\n",
            "--2023-11-07 18:37:36--  https://hgdownload.cse.ucsc.edu/goldenPath/hg38/bigZips/hg38.2bit\n",
            "Resolving hgdownload.cse.ucsc.edu (hgdownload.cse.ucsc.edu)... 128.114.119.163\n",
            "Connecting to hgdownload.cse.ucsc.edu (hgdownload.cse.ucsc.edu)|128.114.119.163|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 835393456 (797M)\n",
            "Saving to: ‘reference_datasets/genomes/hg38.2bit’\n",
            "\n",
            "hg38.2bit           100%[===================>] 796.69M  36.6MB/s    in 23s     \n",
            "\n",
            "2023-11-07 18:37:59 (35.2 MB/s) - ‘reference_datasets/genomes/hg38.2bit’ saved [835393456/835393456]\n",
            "\n",
            "--2023-11-07 18:38:00--  http://ftp.ncbi.nlm.nih.gov/genomes/archive/old_genbank/Eukaryotes/vertebrates_mammals/Homo_sapiens/GRCh38/seqs_for_alignment_pipelines/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.bowtie_index.tar.gz\n",
            "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.10, 130.14.250.11, 2607:f220:41e:250::12, ...\n",
            "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|130.14.250.10|:80... connected.\n",
            "HTTP request sent, awaiting response... 301 Moved Permanently\n",
            "Location: https://ftp.ncbi.nlm.nih.gov/genomes/archive/old_genbank/Eukaryotes/vertebrates_mammals/Homo_sapiens/GRCh38/seqs_for_alignment_pipelines/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.bowtie_index.tar.gz [following]\n",
            "--2023-11-07 18:38:00--  https://ftp.ncbi.nlm.nih.gov/genomes/archive/old_genbank/Eukaryotes/vertebrates_mammals/Homo_sapiens/GRCh38/seqs_for_alignment_pipelines/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.bowtie_index.tar.gz\n",
            "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|130.14.250.10|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 3749247988 (3.5G) [application/x-gzip]\n",
            "Saving to: ‘GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.bowtie_index.tar.gz’\n",
            "\n",
            "GCA_000001405.15_GR 100%[===================>]   3.49G  52.1MB/s    in 80s     \n",
            "\n",
            "2023-11-07 18:39:20 (44.5 MB/s) - ‘GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.bowtie_index.tar.gz’ saved [3749247988/3749247988]\n",
            "\n",
            "GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.bowtie_index.1.bt2\n",
            "GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.bowtie_index.2.bt2\n",
            "GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.bowtie_index.3.bt2\n",
            "GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.bowtie_index.4.bt2\n",
            "GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.bowtie_index.rev.1.bt2\n",
            "GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.bowtie_index.rev.2.bt2\n",
            "Retrieving folder list\n",
            "Processing file 1n6NYU8gDpUWTHa-8Ra1FRhb4BuP2Y0Oz refseq.codingexons.bed\n",
            "Processing file 1mg6r73Jd2aErj6PP8PwuudOEnT4Vq3TY refseq.exons.bed\n",
            "Processing file 1S8LvPGiist74BQWrUpZ_1PJdWYwp1ne0 refseq.introns.bed\n",
            "Processing file 1K97jy3nFYM-lH71mqI1SrUb2p2lYnHNJ refseq.transcripts.bed\n",
            "Retrieving folder list completed\n",
            "Building directory structure\n",
            "Building directory structure completed\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1n6NYU8gDpUWTHa-8Ra1FRhb4BuP2Y0Oz\n",
            "To: /content/InSiTE/annotations/refseq.codingexons.bed\n",
            "100% 99.8M/99.8M [00:00<00:00, 118MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1mg6r73Jd2aErj6PP8PwuudOEnT4Vq3TY\n",
            "To: /content/InSiTE/annotations/refseq.exons.bed\n",
            "100% 133M/133M [00:01<00:00, 110MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1S8LvPGiist74BQWrUpZ_1PJdWYwp1ne0\n",
            "To: /content/InSiTE/annotations/refseq.introns.bed\n",
            "100% 125M/125M [00:01<00:00, 91.6MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1K97jy3nFYM-lH71mqI1SrUb2p2lYnHNJ\n",
            "To: /content/InSiTE/annotations/refseq.transcripts.bed\n",
            "100% 29.3M/29.3M [00:00<00:00, 106MB/s]\n",
            "Download completed\n"
          ]
        }
      ],
      "source": [
        "#%%capture\n",
        "#@title 1. Install\n",
        "\n",
        "# check number of cores\n",
        "import multiprocessing\n",
        "cores = str(multiprocessing.cpu_count())\n",
        "print(\"Number of cores: \" + cores)\n",
        "\n",
        "#@markdown Please execute each cell by pressing the *Play* button on\n",
        "#@markdown the left.\n",
        "\n",
        "# download github \"https://github.com/bryanjjones/InSiTE\" and install dependencies\n",
        "!git clone https://github.com/bryanjjones/InSiTE.git\n",
        "!pip install -r InSiTE/requirements.txt\n",
        "\n",
        "# move to InSiTE directory\n",
        "%cd InSiTE/\n",
        "\n",
        "# install bowtie\n",
        "!wget https://sourceforge.net/projects/bowtie-bio/files/bowtie2/2.5.1/bowtie2-2.5.1-linux-x86_64.zip -P bins/\n",
        "!unzip bins/bowtie2-2.5.1-linux-x86_64.zip -d bins/\n",
        "!mv bins/bowtie2-2.5.1-linux-x86_64 bins/bowtie2\n",
        "\n",
        "# install TwoBitToFa\n",
        "# look here: http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/\n",
        "#!wget http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/twoBitToFa\n",
        "#path_to_twobittofa = \"twoBitToFa\"\n",
        "\n",
        "# install bbmap\n",
        "!wget https://sourceforge.net/projects/bbmap/files/BBMap_39.03.tar.gz\n",
        "!tar -xvzf BBMap_39.03.tar.gz\n",
        "!mv bbmap bins/bbmap\n",
        "#bbmerge_location = \"bbmap/bbmerge-auto.sh\"\n",
        "\n",
        "# install bedtools\n",
        "!wget https://github.com/arq5x/bedtools2/releases/download/v2.31.0/bedtools-2.31.0.tar.gz\n",
        "!tar -xvzf bedtools-2.31.0.tar.gz\n",
        "# add to path\n",
        "%cd bedtools2\n",
        "!make\n",
        "%cd ..\n",
        "import os\n",
        "os.environ['PATH'] += ':/content/InSiTE/bedtools2/bin'\n",
        "\n",
        "# download genome files\n",
        "!mkdir reference_datasets/genomes\n",
        "#!gsutil -m cp -r gs://genomics-public-data/references/GRCh38 reference_datasets/genomes/\n",
        "!wget https://hgdownload.cse.ucsc.edu/goldenPath/hg38/bigZips/hg38.2bit -P reference_datasets/genomes/\n",
        "\n",
        "# download reference index\n",
        "!mkdir reference_datasets/genomes/GRCh38.fna.bowtie_index\n",
        "!wget ftp.ncbi.nlm.nih.gov/genomes/archive/old_genbank/Eukaryotes/vertebrates_mammals/Homo_sapiens/GRCh38/seqs_for_alignment_pipelines/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.bowtie_index.tar.gz\n",
        "!tar -xvzf GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.bowtie_index.tar.gz -C reference_datasets/genomes/GRCh38.fna.bowtie_index/\n",
        "\n",
        "# download annotations\n",
        "!gdown --folder https://drive.google.com/drive/folders/1WCWZyEOAJxNd9g36fohQii_G7OVUDQ9c\n",
        "!mv annotations reference_datasets/annotations"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "cellView": "form",
        "id": "dublwDfDVRiw",
        "outputId": "5ce9a846-d197-4688-a497-4717e20cb714",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Mounted at /content/gdrive\n"
          ]
        }
      ],
      "source": [
        "#@title 2. Link Google Drive\n",
        "\n",
        "#@markdown Running this code will create a popup asking to authorize Colab to access your Google Drive. <br>\n",
        "#@markdown *Note that you'll have to re-authorize Colab access with every new runtime.*\n",
        "\n",
        "from google.colab import drive\n",
        "drive.mount('/content/gdrive', force_remount=True)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "cellView": "form",
        "id": "8ndmUn1kcVgx"
      },
      "outputs": [],
      "source": [
        "#@title 2. Upload Vector file\n",
        "vector_file = False #@param {type:\"boolean\"}\n",
        "#@markdown This will eliminate reads that align to plasmid and not integrated transposon\n",
        "if vector_file:\n",
        "  from google.colab import files\n",
        "  uploaded = files.upload()\n",
        "  tmp = list(uploaded.keys())\n",
        "  vector_file = tmp[0]"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "oOmap33hVJ0j",
        "cellView": "form",
        "outputId": "d1a43dd4-ba21-4809-d74d-8846def80dd2",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "python scripts/InSiTE.py --fastq=/content/gdrive/MyDrive/data/30-933683439/NGS3_R2_001.fastq.gz --pairs=/content/gdrive/MyDrive/data/30-933683439/NGS3_R1_001.fastq.gz --twobitgenomelocation=reference_datasets/genomes/hg38.2bit --bowtielocation=bins/bowtie2/bowtie2 --bowtiethreads=8\n",
            "Merging paired reads files /content/gdrive/MyDrive/data/30-933683439/NGS3_R2_001.fastq.gz and /content/gdrive/MyDrive/data/30-933683439/NGS3_R1_001.fastq.gz using bbmerge.\n",
            "./bins/bbmap/bbmerge-auto.sh in1=/content/gdrive/MyDrive/data/30-933683439/NGS3_R2_001.fastq.gz in2=/content/gdrive/MyDrive/data/30-933683439/NGS3_R1_001.fastq.gz out=/content/gdrive/MyDrive/data/30-933683439/NGS3_R2_001_merge.fastq outu1=/content/gdrive/MyDrive/data/30-933683439/NGS3_R2_001_unmerge.fastq outu2=/content/gdrive/MyDrive/data/30-933683439/NGS3_R1_001_unmerge.fastq reads=-1\n"
          ]
        }
      ],
      "source": [
        "#@title 3. Run analysis\n",
        "import os\n",
        "import re\n",
        "#@markdown Select a folder to run all paired reads with the same conditions. Leave blank to run specific files:\n",
        "folder = \"/content/gdrive/MyDrive/data/30-933683439/\" #@param{type:\"string\"}\n",
        "#@markdown fastq input file/s (Google Drive is found in /content/gdrive/MyDrive/...):\n",
        "fastq_file = \"/content/gdrive/MyDrive/data/30-933683439/NGS1_R1_001.fastq.gz\" #@param {type:\"string\"}\n",
        "paired_fastq_file = \"/content/gdrive/MyDrive/data/30-933683439/NGS1_R2_001.fastq.gz\" #@param {type:\"string\"}\n",
        "#@markdown Limit number of reads:\n",
        "number_reads = -1 #@param {type:\"integer\"}\n",
        "#@markdown barcode sequence to trim off of reads:\n",
        "barcode = \"NNNNNN\"  #@param {type:\"string\"}\n",
        "#@markdown number of nucleotides upstream of integration site to return:\n",
        "lwindow = 50 #@param {type:\"integer\"}\n",
        "#@markdown number of nucleotides downstream of integration site to return:\n",
        "rwindow = 50 #@param {type:\"integer\"}\n",
        "#@markdown minimum length of a read to try mapping. default (25) will usually avoid any false positives in read sets of 200k reads\n",
        "min = 25 #@param {type:\"integer\"}\n",
        "#@markdown 5' primer sequence to remove from reads:\n",
        "primer5 = \"\" #@param {type:\"string\"}\n",
        "#@markdown 3' primer sequence to remove from reads:\n",
        "primer3 = \"\"  #@param {type:\"string\"}\n",
        "#@markdown additional (non-genomic) nts to trim off of 3' end of reads:\n",
        "trim5 = 0 #@param {type:\"integer\"}\n",
        "#@markdown additional (non-genomic) nts to trim off of 5' end of reads:\n",
        "trim3 = 0 #@param {type:\"integer\"}\n",
        "#@markdown select feature to map reads to:\n",
        "feature = \"All\" #@param [\"All\", \"gene\", \"exon\", \"intron\"]\n",
        "#@markdown whether to map distance of each read, or only whether reads overlap with feature. Same number of distance variables must be given as features.\n",
        "dist = False #@param {type:\"boolean\"}\n",
        "close = 100 #@param {type:\"integer\"}\n",
        "#@markdown do not map insertion sites to genome annotations\n",
        "no_annotate = False #@param {type:\"boolean\"}\n",
        "#@markdown use random integration sites matched to given query set instead of actual query set\n",
        "random_sites = False #@param {type:\"boolean\"}\n",
        "#@markdown use random sequences for mapping\n",
        "random_nt = False #@param {type:\"boolean\"}\n",
        "#@markdown do not get sequences from either entrez or local TwoBit genome around locations indicated by genome_location_csv\n",
        "no_seqs = False #@param {type:\"boolean\"}\n",
        "#compress_reads = True #@param {type:\"boolean\"}\n",
        "\n",
        "paired_files = {}\n",
        "if folder:\n",
        "  results_dir = folder\n",
        "  for file in os.listdir(folder):\n",
        "    if '.fastq' in file and 'merge' not in file and 'trimmed' not in file:\n",
        "      try:\n",
        "        paired_files[re.sub(r'_R\\d_', '', file)].append(os.path.join(folder, file))\n",
        "      except KeyError:\n",
        "        paired_files[re.sub(r'_R\\d_', '', file)] = [os.path.join(folder, file)]\n",
        "else:\n",
        "  results_dir = os.path.dirname(fastq_file)\n",
        "  paired_files['OnlyOne'] = [fastq_file, paired_fastq_file]\n",
        "\n",
        "#run analysis\n",
        "for key, value in paired_files.items():\n",
        "  if len(value) == 1 or len(value) > 2:\n",
        "    print('Wrong number of files linked to '+value[0])\n",
        "    continue\n",
        "  fastq_file = value[0]\n",
        "  paired_fastq_file = value[1]\n",
        "  command = \"python scripts/InSiTE.py --fastq=\" + fastq_file\n",
        "  if paired_fastq_file:\n",
        "    command += \" --pairs=\" + paired_fastq_file\n",
        "  #command += \" --chromosome_ids=reference_datasets/chromosomes.csv\"\n",
        "  #command += \" --bowtieindex=reference_datasets/genomes/GRCh38.fna.bowtie_index/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.bowtie_index\"\n",
        "  #command += \" --twobitlocation=scripts/TwoBitToFa\"\n",
        "  command += \" --twobitgenomelocation=reference_datasets/genomes/hg38.2bit\"\n",
        "  command += \" --bowtielocation=bins/bowtie2/bowtie2\"\n",
        "  command += \" --bowtiethreads=\" + cores\n",
        "  #command += \" --bbmergelocation=$bbmerge_location\"\n",
        "  #command += \" --bbduklocation=bbmap/bbduk.sh\"\n",
        "  if number_reads != -1:\n",
        "    command += \" --nreads=\" + str(number_reads)\n",
        "  if vector_file:\n",
        "    command += \" --vectors=\" + vector_file\n",
        "  if random_sites:\n",
        "    command += \" --random_is\"\n",
        "  if random_nt:\n",
        "    command += \" --random_nt\"\n",
        "  if no_seqs:\n",
        "    command += \" --no_seqs\"\n",
        "  #if compress_reads:\n",
        "  #  command += \" -z\"\n",
        "  if no_annotate:\n",
        "    command += \" --no_annotate\"\n",
        "  if barcode != \"NNNNNN\":\n",
        "    command += \" --barcode=\" + barcode\n",
        "  if lwindow != 50:\n",
        "    command += \" --lwindow=\" + str(lwindow)\n",
        "  if rwindow != 50:\n",
        "    command += \" --rwindow=\" + str(rwindow)\n",
        "  if min != 25:\n",
        "    command += \" --min=\" + str(min)\n",
        "  if primer5:\n",
        "    command += \" --primer5=\" + primer5\n",
        "  if primer3:\n",
        "    command += \" --primer3=\" + primer3\n",
        "  if trim5 != 0:\n",
        "    command += \" --trim5=\" + str(trim5)\n",
        "  if trim3 != 0:\n",
        "    command += \" --trim3=\" + str(trim3)\n",
        "  if feature != \"All\":\n",
        "    command += \" --feature=\" + feature\n",
        "  if dist:\n",
        "    command += \" --dist\"\n",
        "  if close != 100:\n",
        "    command += \" --close=\" + str(close)\n",
        "\n",
        "  # run analysis\n",
        "  print(command)\n",
        "  !$command\n",
        "  for file in os.listdir(results_dir):\n",
        "    if 'trimmed.fastq' in file or 'merged_reads.fastq' in file or '.bam' in file or ('.sam' in file and 'abundant' not in file):\n",
        "      os.remove(os.path.join(results_dir, file))\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {
        "id": "9jydOzAIsoZY",
        "cellView": "form",
        "outputId": "44bd0000-a8cb-4949-ca08-3958e9600b67",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 260
        }
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content/gdrive/MyDrive/data/30-933683439\n",
            "/content/gdrive/.shortcut-targets-by-id/1hnxHMtb_w2FWKQqzWKlHMj4-nOKoVfLB/data/30-933683439\n",
            "rm: cannot remove 'results.zip': No such file or directory\n",
            "  adding: NGS1_R1_001_distances.csv (deflated 57%)\n",
            "  adding: NGS1_R1_001_IS_annotations.csv (deflated 69%)\n",
            "  adding: NGS1_R1_001_IS_mappings.csv (deflated 80%)\n",
            "  adding: NGS2_R1_001_distances.csv (deflated 57%)\n",
            "  adding: NGS2_R1_001_IS_annotations.csv (deflated 68%)\n",
            "  adding: NGS2_R1_001_IS_mappings.csv (deflated 79%)\n",
            "  adding: NGS1_R1_001IS_logo.svg (stored 0%)\n",
            "  adding: NGS2_R1_001IS_logo.svg (stored 0%)\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "\n",
              "    async function download(id, filename, size) {\n",
              "      if (!google.colab.kernel.accessAllowed) {\n",
              "        return;\n",
              "      }\n",
              "      const div = document.createElement('div');\n",
              "      const label = document.createElement('label');\n",
              "      label.textContent = `Downloading \"${filename}\": `;\n",
              "      div.appendChild(label);\n",
              "      const progress = document.createElement('progress');\n",
              "      progress.max = size;\n",
              "      div.appendChild(progress);\n",
              "      document.body.appendChild(div);\n",
              "\n",
              "      const buffers = [];\n",
              "      let downloaded = 0;\n",
              "\n",
              "      const channel = await google.colab.kernel.comms.open(id);\n",
              "      // Send a message to notify the kernel that we're ready.\n",
              "      channel.send({})\n",
              "\n",
              "      for await (const message of channel.messages) {\n",
              "        // Send a message to notify the kernel that we're ready.\n",
              "        channel.send({})\n",
              "        if (message.buffers) {\n",
              "          for (const buffer of message.buffers) {\n",
              "            buffers.push(buffer);\n",
              "            downloaded += buffer.byteLength;\n",
              "            progress.value = downloaded;\n",
              "          }\n",
              "        }\n",
              "      }\n",
              "      const blob = new Blob(buffers, {type: 'application/binary'});\n",
              "      const a = document.createElement('a');\n",
              "      a.href = window.URL.createObjectURL(blob);\n",
              "      a.download = filename;\n",
              "      div.appendChild(a);\n",
              "      a.click();\n",
              "      div.remove();\n",
              "    }\n",
              "  "
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "download(\"download_ea1e1ab1-909a-4550-84b3-ac8fd91bbf6c\", \"results.zip\", 44772473)"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content/InSiTE\n",
            "Also look in your google drive folder\n"
          ]
        }
      ],
      "source": [
        "#@title Download Results\n",
        "#get dir of fastq_file\n",
        "results_dir = os.path.dirname(fastq_file)\n",
        "print(results_dir)\n",
        "# download results\n",
        "%cd $results_dir\n",
        "!rm results.zip\n",
        "from google.colab import files\n",
        "!zip results.zip *.csv *.svg\n",
        "files.download(\"results.zip\")\n",
        "%cd /content/InSiTE\n",
        "print('Also look in your google drive folder')"
      ]
    }
  ],
  "metadata": {
    "colab": {
      "provenance": [],
      "machine_shape": "hm",
      "authorship_tag": "ABX9TyPbP62IJnxS1yElh4+zNf4E",
      "include_colab_link": true
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}